// @angi/one-design-system/dist/components/es/Progress/index.js
import { jsx as s, jsxs as R, Fragment as E } from "react/jsx-runtime";
import { forwardRef as S } from "react";
import { a as p } from "../_chunks/ProgressBar-CJx2i27x.js";
import { c as P } from "../_chunks/clsx-OuTLNxxd.js";
import { c as o } from "../_chunks/createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import '../assets/Progress.css';var _ = "t3vuh60", A = "t3vuh61", N = o({ defaultClassName: "t3vuh62", variantClassNames: { size: { sm: "t3vuh63", lg: "t3vuh64" }, variant: { solid: "t3vuh65", outline: "t3vuh66" }, radius: { none: "t3vuh67", sm: "t3vuh68" } }, defaultVariants: { size: "sm", variant: "solid", radius: "none" }, compoundVariants: [] }), V = o({ defaultClassName: "t3vuh6a t3vuh69", variantClassNames: { radius: { none: "t3vuh6b", sm: "t3vuh6c" }, animation: { true: "t3vuh6d" } }, defaultVariants: { radius: "none", animation: !1 }, compoundVariants: [] });
const g = 0, F = 100, O = 0, I = ["sm", "lg"], T = "sm", z = ["solid", "outline"], U = "solid", $ = ["none", "sm"], x = "none", D = S(
  ({
    min: n = g,
    max: i = F,
    value: l = O,
    label: t,
    valueText: e,
    size: m = T,
    variant: u = U,
    radius: r = x,
    animation: c = !1,
    className: d,
    ...v
  }, f) => {
    const h = Object.fromEntries(
      Object.entries(v).filter(([, a]) => a !== void 0)
    );
    return /* @__PURE__ */ s(
      p,
      {
        ref: f,
        value: l,
        minValue: n,
        maxValue: i,
        ...t && { "aria-label": t },
        className: P(_, d),
        ...h,
        children: ({ percentage: a }) => /* @__PURE__ */ R(E, { children: [
          e && /* @__PURE__ */ s("span", { "aria-live": "polite", className: A, children: e }),
          /* @__PURE__ */ s("div", { className: N({ size: m, variant: u, radius: r }), children: /* @__PURE__ */ s(
            "div",
            {
              className: V({ radius: r, animation: c }),
              style: { width: `${a}%` }
            }
          ) })
        ] })
      }
    );
  }
);
D.displayName = "Progress";
export {
  F as DEFAULT_PROGRESS_MAX,
  g as DEFAULT_PROGRESS_MIN,
  x as DEFAULT_PROGRESS_RADIUS,
  T as DEFAULT_PROGRESS_SIZE,
  O as DEFAULT_PROGRESS_VALUE,
  U as DEFAULT_PROGRESS_VARIANT,
  D as Progress,
  $ as ProgressRadiuses,
  I as ProgressSizes,
  z as ProgressVariants
};
