// @angi/one-design-system/dist/components/es/CheckboxFieldset/index.js
import { jsxs as h, jsx as a } from "react/jsx-runtime";
import { forwardRef as C, useId as k } from "react";
import { c as w } from "../_chunks/Label-DLRS1VtT.js";
import { f as y } from "../_chunks/Form-sh8c7lqF.js";
import { u as F, C as _ } from "../_chunks/useCheckboxGroupContext-R2g3qemh.js";
import { L } from "../_chunks/Legend-Cgz7gN8f.js";
import { H as V } from "../_chunks/Hint-CIDmUonJ.js";
import { V as D } from "../_chunks/ValidationMessage-Dc42Hn25.js";
import { C as R } from "../_chunks/CheckboxTile-DKQeLOJY.js";
import { C as S } from "../_chunks/Checkbox-DpPIhntn.js";
import { c as I } from "../_chunks/clsx-OuTLNxxd.js";
import { L as T } from "../_chunks/Label-BqoOfVfQ.js";
import { C as q } from "../_chunks/CheckboxItem--3cltbnE.js";
import { c as B } from "../_chunks/createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import '../assets/CheckboxFieldset.css';var E = "zz1h9l0";
const g = C(
  ({ value: i, label: e, disabled: l = !1, hideFocusRing: p = !1, className: r }, o) => {
    const t = k(), m = `${t}-checkbox`, s = `${t}-label`, c = F(), d = c.isSelected(i), v = c.isInvalid, n = l || c.isDisabled;
    return /* @__PURE__ */ h(
      "label",
      {
        className: I(E, r),
        "data-checked": d || void 0,
        "data-disabled": n || void 0,
        "data-invalid": v || void 0,
        children: [
          /* @__PURE__ */ a(
            q,
            {
              ref: o,
              value: i,
              id: m,
              "aria-labelledby": s,
              disabled: l,
              hideFocusRing: p
            }
          ),
          /* @__PURE__ */ a(T, { variant: "normal", ...e, id: s, asChild: !0, children: /* @__PURE__ */ a("span", { children: e.children }) })
        ]
      }
    );
  }
);
g.displayName = "_CheckboxCompact";
var G = "xwicp90", H = "xwicp91", M = "xwicp92", j = B({ defaultClassName: "xwicp93", variantClassNames: { variant: { tile: "xwicp94", text: "xwicp95", compact: "xwicp96" } }, defaultVariants: { variant: "tile" }, compoundVariants: [] });
const A = (i) => {
  switch (i.variant) {
    case "tile":
      return i.items.map((e) => /* @__PURE__ */ a(R, { ...e }, e.value));
    case "text":
      return i.items.map((e) => /* @__PURE__ */ a(
        S,
        {
          ...e,
          label: { variant: "normal", ...e.label }
        },
        e.value
      ));
    case "compact":
      return i.items.map((e) => /* @__PURE__ */ a(g, { ...e, hideFocusRing: !0 }, e.value));
  }
}, z = C((i, e) => {
  const {
    variant: l,
    items: p,
    legend: r,
    hint: o,
    validationMessage: t,
    className: m,
    required: s,
    disabled: c,
    readOnly: d,
    ...v
  } = i, n = k(), b = o ? `${n}-hint` : void 0, x = t ? `${n}-validation` : void 0, f = [b, x].filter(
    Boolean
  ), u = f.length > 0 ? f.join(" ") : void 0, N = t?.variant === "invalid", $ = r?.required ?? s ?? !1;
  return /* @__PURE__ */ h(
    _,
    {
      ref: e,
      className: I(G, m),
      invalid: N,
      ...s !== void 0 && { required: s },
      ...c !== void 0 && { disabled: c },
      ...d !== void 0 && { readOnly: d },
      ...u !== void 0 && {
        "aria-describedby": u
      },
      ...v,
      children: [
        (r || o) && /* @__PURE__ */ h("div", { className: H, children: [
          r && /* @__PURE__ */ a(L, { ...r, asChild: !0, required: $, children: /* @__PURE__ */ a(w, { children: r.children }) }),
          o && /* @__PURE__ */ a(y, { elementType: "div", slot: "description", id: b, children: /* @__PURE__ */ a(V, { ...o }) })
        ] }),
        t && /* @__PURE__ */ a("div", { className: M, children: /* @__PURE__ */ a(D, { id: x, ...t, children: t.children }) }),
        /* @__PURE__ */ a(
          "div",
          {
            className: j({ variant: i.variant }),
            children: A(i)
          }
        )
      ]
    }
  );
});
z.displayName = "CheckboxFieldset";
const re = ["tile", "text", "compact"], oe = "tile";
export {
  z as CheckboxFieldset,
  re as CheckboxFieldsetVariants,
  oe as DEFAULT_CHECKBOX_FIELDSET_VARIANT
};
