// @angi/one-design-system/dist/components/es/Modal/index.js
import { jsxs as ze, jsx as O } from "react/jsx-runtime";
import $, { useState as I, useEffect as k, useCallback as F, useMemo as z, useRef as N, useContext as M, createContext as B, forwardRef as A, useSyncExternalStore as rn } from "react";
import { c as W } from "../_chunks/clsx-OuTLNxxd.js";
import { a as sn } from "../_chunks/Button-CHUd1jP9.js";
import { e as me, c as ne, k as Ye, b as Y, g as on, $ as pe, a as ge, i as Ge, l as _e, m as Xe, n as an } from "../_chunks/utils-CjtkL41C.js";
import { a as qe } from "../_chunks/RSPContexts-Dm3BezC-.js";
import { $ as re } from "../_chunks/filterDOMProps-Cxz9cAAv.js";
import { $ as xe } from "../_chunks/focusSafely-OgqIqadr.js";
import { a as ln } from "../_chunks/usePress-D9Qp6Amx.js";
import { a as cn, b as un } from "../_chunks/FocusScope-DlOb-a2X.js";
import dn, { flushSync as fn } from "react-dom";
import { $ as hn } from "../_chunks/useControlledState-3TGZUXux.js";
import { d as mn } from "../_chunks/Label-DLRS1VtT.js";
import { $ as pn } from "../_chunks/context-qNvgpQEb.js";
import { a as gn, i as Je, j as bn } from "../_chunks/useFocusWithin-CAx2NHfT.js";
import { $ as De } from "../_chunks/useFocusable-CrxUcjB4.js";
import { h as vn } from "../_chunks/openLink-EIDcTMUl.js";
import { c as ie } from "../_chunks/createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import { themeContract as L } from "@angi/one-design-system-themes-vanilla-extract";
import { S as $n } from "../_chunks/xmark-CxuwVom6.js";
import { B as yn } from "../_chunks/Button-CG9o4a5a.js";
import { t as En } from "../_chunks/Text-B3tFn66X.js";
import { u as wn } from "../_chunks/useTranslations-Da7GR3Ot.js";
import '../assets/Modal.css';function ee(t, e) {
  if (!t) return !1;
  let n = window.getComputedStyle(t), r = /(auto|scroll)/.test(n.overflow + n.overflowX + n.overflowY);
  return r && e && (r = t.scrollHeight !== t.clientHeight || t.scrollWidth !== t.clientWidth), r;
}
function Ze(t, e) {
  let n = t;
  for (ee(n, e) && (n = n.parentElement); n && !ee(n, e); ) n = n.parentElement;
  return n || document.scrollingElement || document.documentElement;
}
const _n = /* @__PURE__ */ new Set([
  "checkbox",
  "radio",
  "range",
  "color",
  "file",
  "image",
  "button",
  "submit",
  "reset"
]);
function j(t) {
  return t instanceof HTMLInputElement && !_n.has(t.type) || t instanceof HTMLTextAreaElement || t instanceof HTMLElement && t.isContentEditable;
}
let x = typeof document < "u" && window.visualViewport;
function xn() {
  let t = me(), [e, n] = I(() => t ? {
    width: 0,
    height: 0
  } : Se());
  return k(() => {
    let r = () => {
      x && x.scale > 1 || n((o) => {
        let a = Se();
        return a.width === o.width && a.height === o.height ? o : a;
      });
    }, i, s = (o) => {
      x && x.scale > 1 || j(o.target) && (i = requestAnimationFrame(() => {
        (!document.activeElement || !j(document.activeElement)) && n((a) => {
          let l = {
            width: window.innerWidth,
            height: window.innerHeight
          };
          return l.width === a.width && l.height === a.height ? a : l;
        });
      }));
    };
    return window.addEventListener("blur", s, !0), x ? x.addEventListener("resize", r) : window.addEventListener("resize", r), () => {
      cancelAnimationFrame(i), window.removeEventListener("blur", s, !0), x ? x.removeEventListener("resize", r) : window.removeEventListener("resize", r);
    };
  }, []), e;
}
function Se() {
  return {
    // Multiply by the visualViewport scale to get the "natural" size, unaffected by pinch zooming.
    width: x ? x.width * x.scale : window.innerWidth,
    height: x ? x.height * x.scale : window.innerHeight
  };
}
function Qe(t, e = !0) {
  let [n, r] = I(!0), i = n && e;
  return ne(() => {
    if (i && t.current && "getAnimations" in t.current)
      for (let s of t.current.getAnimations()) s instanceof CSSTransition && s.cancel();
  }, [
    t,
    i
  ]), et(t, i, F(() => r(!1), [])), i;
}
function Te(t, e) {
  let [n, r] = I(e ? "open" : "closed");
  switch (n) {
    case "open":
      e || r("exiting");
      break;
    case "closed":
    case "exiting":
      e && r("open");
      break;
  }
  let i = n === "exiting";
  return et(t, i, F(() => {
    r((s) => s === "exiting" ? "closed" : s);
  }, [])), i;
}
function et(t, e, n) {
  ne(() => {
    if (e && t.current) {
      if (!("getAnimations" in t.current)) {
        n();
        return;
      }
      let r = t.current.getAnimations();
      if (r.length === 0) {
        n();
        return;
      }
      let i = !1;
      return Promise.all(r.map((s) => s.finished)).then(() => {
        i || fn(() => {
          n();
        });
      }).catch(() => {
      }), () => {
        i = !0;
      };
    }
  }, [
    t,
    e,
    n
  ]);
}
const Dn = Symbol.for("react-aria.i18n.locale"), Sn = Symbol.for("react-aria.i18n.strings");
let R;
class se {
  /** Returns a localized string for the given key and locale. */
  getStringForLocale(e, n) {
    let i = this.getStringsForLocale(n)[e];
    if (!i) throw new Error(`Could not find intl message ${e} in ${n} locale`);
    return i;
  }
  /** Returns all localized strings for the given locale. */
  getStringsForLocale(e) {
    let n = this.strings[e];
    return n || (n = Tn(e, this.strings, this.defaultLocale), this.strings[e] = n), n;
  }
  static getGlobalDictionaryForPackage(e) {
    if (typeof window > "u") return null;
    let n = window[Dn];
    if (R === void 0) {
      let i = window[Sn];
      if (!i) return null;
      R = {};
      for (let s in i) R[s] = new se({
        [n]: i[s]
      }, n);
    }
    let r = R?.[e];
    if (!r) throw new Error(`Strings for package "${e}" were not included by LocalizedStringProvider. Please add it to the list passed to createLocalizedStringDictionary.`);
    return r;
  }
  constructor(e, n = "en-US") {
    this.strings = Object.fromEntries(Object.entries(e).filter(([, r]) => r)), this.defaultLocale = n;
  }
}
function Tn(t, e, n = "en-US") {
  if (e[t]) return e[t];
  let r = On(t);
  if (e[r]) return e[r];
  for (let i in e)
    if (i.startsWith(r + "-")) return e[i];
  return e[n];
}
function On(t) {
  return Intl.Locale ? new Intl.Locale(t).language : t.split("-")[0];
}
const Oe = /* @__PURE__ */ new Map(), Ce = /* @__PURE__ */ new Map();
class Cn {
  /** Formats a localized string for the given key with the provided variables. */
  format(e, n) {
    let r = this.strings.getStringForLocale(e, this.locale);
    return typeof r == "function" ? r(n, this) : r;
  }
  plural(e, n, r = "cardinal") {
    let i = n["=" + e];
    if (i) return typeof i == "function" ? i() : i;
    let s = this.locale + ":" + r, o = Oe.get(s);
    o || (o = new Intl.PluralRules(this.locale, {
      type: r
    }), Oe.set(s, o));
    let a = o.select(e);
    return i = n[a] || n.other, typeof i == "function" ? i() : i;
  }
  number(e) {
    let n = Ce.get(this.locale);
    return n || (n = new Intl.NumberFormat(this.locale), Ce.set(this.locale, n)), n.format(e);
  }
  select(e, n) {
    let r = e[n] || e.other;
    return typeof r == "function" ? r() : r;
  }
  constructor(e, n) {
    this.locale = e, this.strings = n;
  }
}
const Pe = /* @__PURE__ */ new WeakMap();
function Pn(t) {
  let e = Pe.get(t);
  return e || (e = new se(t), Pe.set(t, e)), e;
}
function Ln(t, e) {
  return e && se.getGlobalDictionaryForPackage(e) || Pn(t);
}
function In(t, e) {
  let { locale: n } = pn(), r = Ln(t, e);
  return z(() => new Cn(n, r), [
    n,
    r
  ]);
}
function Mn({ children: t }) {
  let e = z(() => ({
    register: () => {
    }
  }), []);
  return /* @__PURE__ */ $.createElement(ln.Provider, {
    value: e
  }, t);
}
function An(t) {
  let { ref: e, onInteractOutside: n, isDisabled: r, onInteractOutsideStart: i } = t, s = N({
    isPointerDown: !1,
    ignoreEmulatedMouseEvents: !1
  }), o = De((l) => {
    n && J(l, e) && (i && i(l), s.current.isPointerDown = !0);
  }), a = De((l) => {
    n && n(l);
  });
  k(() => {
    let l = s.current;
    if (r) return;
    const u = e.current, c = gn(u);
    if (typeof PointerEvent < "u") {
      let m = (f) => {
        l.isPointerDown && J(f, e) && a(f), l.isPointerDown = !1;
      };
      return c.addEventListener("pointerdown", o, !0), c.addEventListener("click", m, !0), () => {
        c.removeEventListener("pointerdown", o, !0), c.removeEventListener("click", m, !0);
      };
    } else if (process.env.NODE_ENV === "test") {
      let m = (b) => {
        l.ignoreEmulatedMouseEvents ? l.ignoreEmulatedMouseEvents = !1 : l.isPointerDown && J(b, e) && a(b), l.isPointerDown = !1;
      }, f = (b) => {
        l.ignoreEmulatedMouseEvents = !0, l.isPointerDown && J(b, e) && a(b), l.isPointerDown = !1;
      };
      return c.addEventListener("mousedown", o, !0), c.addEventListener("mouseup", m, !0), c.addEventListener("touchstart", o, !0), c.addEventListener("touchend", f, !0), () => {
        c.removeEventListener("mousedown", o, !0), c.removeEventListener("mouseup", m, !0), c.removeEventListener("touchstart", o, !0), c.removeEventListener("touchend", f, !0);
      };
    }
  }, [
    e,
    r
  ]);
}
function J(t, e) {
  if (t.button > 0) return !1;
  if (t.target) {
    const n = t.target.ownerDocument;
    if (!n || !n.documentElement.contains(t.target) || t.target.closest("[data-react-aria-top-layer]")) return !1;
  }
  return e.current ? !t.composedPath().includes(e.current) : !1;
}
const T = [];
function Nn(t, e) {
  let { onClose: n, shouldCloseOnBlur: r, isOpen: i, isDismissable: s = !1, isKeyboardDismissDisabled: o = !1, shouldCloseOnInteractOutside: a } = t;
  k(() => {
    if (i && !T.includes(e))
      return T.push(e), () => {
        let h = T.indexOf(e);
        h >= 0 && T.splice(h, 1);
      };
  }, [
    i,
    e
  ]);
  let l = () => {
    T[T.length - 1] === e && n && n();
  }, u = (h) => {
    (!a || a(h.target)) && T[T.length - 1] === e && (h.stopPropagation(), h.preventDefault());
  }, c = (h) => {
    (!a || a(h.target)) && (T[T.length - 1] === e && (h.stopPropagation(), h.preventDefault()), l());
  }, m = (h) => {
    h.key === "Escape" && !o && !h.nativeEvent.isComposing && (h.stopPropagation(), h.preventDefault(), l());
  };
  An({
    ref: e,
    onInteractOutside: s && i ? c : void 0,
    onInteractOutsideStart: u
  });
  let { focusWithinProps: f } = Je({
    isDisabled: !r,
    onBlurWithin: (h) => {
      !h.relatedTarget || cn(h.relatedTarget) || (!a || a(h.relatedTarget)) && n?.();
    }
  }), b = (h) => {
    h.target === h.currentTarget && h.preventDefault();
  };
  return {
    overlayProps: {
      onKeyDown: m,
      ...f
    },
    underlayProps: {
      onPointerDown: b
    }
  };
}
const K = typeof document < "u" && window.visualViewport;
let Z = 0, ce;
function kn(t = {}) {
  let { isDisabled: e } = t;
  ne(() => {
    if (!e)
      return Z++, Z === 1 && (vn() ? ce = Hn() : ce = Rn()), () => {
        Z--, Z === 0 && ce();
      };
  }, [
    e
  ]);
}
function Rn() {
  let t = window.innerWidth - document.documentElement.clientWidth;
  return Ye(t > 0 && // Use scrollbar-gutter when supported because it also works for fixed positioned elements.
  ("scrollbarGutter" in document.documentElement.style ? ue(document.documentElement, "scrollbarGutter", "stable") : ue(document.documentElement, "paddingRight", `${t}px`)), ue(document.documentElement, "overflow", "hidden"));
}
function Hn() {
  let t, e = !1, n = (l) => {
    let u = l.target;
    t = ee(u) ? u : Ze(u, !0), e = !1;
    let c = u.ownerDocument.defaultView.getSelection();
    c && !c.isCollapsed && c.containsNode(u, !0) && (e = !0), "selectionStart" in u && "selectionEnd" in u && u.selectionStart < u.selectionEnd && u.ownerDocument.activeElement === u && (e = !0);
  }, r = document.createElement("style");
  r.textContent = `
@layer {
  * {
    overscroll-behavior: contain;
  }
}`.trim(), document.head.prepend(r);
  let i = (l) => {
    if (!(l.touches.length === 2 || e)) {
      if (!t || t === document.documentElement || t === document.body) {
        l.preventDefault();
        return;
      }
      t.scrollHeight === t.clientHeight && t.scrollWidth === t.clientWidth && l.preventDefault();
    }
  }, s = (l) => {
    let u = l.target, c = l.relatedTarget;
    if (c && j(c))
      c.focus({
        preventScroll: !0
      }), Le(c, j(u));
    else if (!c) {
      var m;
      let f = (m = u.parentElement) === null || m === void 0 ? void 0 : m.closest("[tabindex]");
      f?.focus({
        preventScroll: !0
      });
    }
  }, o = HTMLElement.prototype.focus;
  HTMLElement.prototype.focus = function(l) {
    let u = document.activeElement != null && j(document.activeElement);
    o.call(this, {
      ...l,
      preventScroll: !0
    }), (!l || !l.preventScroll) && Le(this, u);
  };
  let a = Ye(de(document, "touchstart", n, {
    passive: !1,
    capture: !0
  }), de(document, "touchmove", i, {
    passive: !1,
    capture: !0
  }), de(document, "blur", s, !0));
  return () => {
    a(), r.remove(), HTMLElement.prototype.focus = o;
  };
}
function ue(t, e, n) {
  let r = t.style[e];
  return t.style[e] = n, () => {
    t.style[e] = r;
  };
}
function de(t, e, n, r) {
  return t.addEventListener(e, n, r), () => {
    t.removeEventListener(e, n, r);
  };
}
function Le(t, e) {
  e || !K ? Ie(t) : K.addEventListener("resize", () => Ie(t), {
    once: !0
  });
}
function Ie(t) {
  let e = document.scrollingElement || document.documentElement, n = t;
  for (; n && n !== e; ) {
    let r = Ze(n);
    if (r !== document.documentElement && r !== document.body && r !== n) {
      let i = r.getBoundingClientRect(), s = n.getBoundingClientRect();
      if (s.top < i.top || s.bottom > i.top + n.clientHeight) {
        let o = i.bottom;
        K && (o = Math.min(o, K.offsetTop + K.height));
        let a = s.top - i.top - ((o - i.top) / 2 - s.height / 2);
        r.scrollTo({
          // Clamp to the valid range to prevent over-scrolling.
          top: Math.max(0, Math.min(r.scrollHeight - r.clientHeight, r.scrollTop + a)),
          behavior: "smooth"
        });
      }
    }
    n = r.parentElement;
  }
}
const Fn = /* @__PURE__ */ B({});
function Bn() {
  var t;
  return (t = M(Fn)) !== null && t !== void 0 ? t : {};
}
var tt = {};
tt = {
  dismiss: "تجاهل"
};
var nt = {};
nt = {
  dismiss: "Отхвърляне"
};
var rt = {};
rt = {
  dismiss: "Odstranit"
};
var it = {};
it = {
  dismiss: "Luk"
};
var st = {};
st = {
  dismiss: "Schließen"
};
var ot = {};
ot = {
  dismiss: "Απόρριψη"
};
var at = {};
at = {
  dismiss: "Dismiss"
};
var lt = {};
lt = {
  dismiss: "Descartar"
};
var ct = {};
ct = {
  dismiss: "Lõpeta"
};
var ut = {};
ut = {
  dismiss: "Hylkää"
};
var dt = {};
dt = {
  dismiss: "Rejeter"
};
var ft = {};
ft = {
  dismiss: "התעלם"
};
var ht = {};
ht = {
  dismiss: "Odbaci"
};
var mt = {};
mt = {
  dismiss: "Elutasítás"
};
var pt = {};
pt = {
  dismiss: "Ignora"
};
var gt = {};
gt = {
  dismiss: "閉じる"
};
var bt = {};
bt = {
  dismiss: "무시"
};
var vt = {};
vt = {
  dismiss: "Atmesti"
};
var $t = {};
$t = {
  dismiss: "Nerādīt"
};
var yt = {};
yt = {
  dismiss: "Lukk"
};
var Et = {};
Et = {
  dismiss: "Negeren"
};
var wt = {};
wt = {
  dismiss: "Zignoruj"
};
var _t = {};
_t = {
  dismiss: "Descartar"
};
var xt = {};
xt = {
  dismiss: "Dispensar"
};
var Dt = {};
Dt = {
  dismiss: "Revocare"
};
var St = {};
St = {
  dismiss: "Пропустить"
};
var Tt = {};
Tt = {
  dismiss: "Zrušiť"
};
var Ot = {};
Ot = {
  dismiss: "Opusti"
};
var Ct = {};
Ct = {
  dismiss: "Odbaci"
};
var Pt = {};
Pt = {
  dismiss: "Avvisa"
};
var Lt = {};
Lt = {
  dismiss: "Kapat"
};
var It = {};
It = {
  dismiss: "Скасувати"
};
var Mt = {};
Mt = {
  dismiss: "取消"
};
var At = {};
At = {
  dismiss: "關閉"
};
var Nt = {};
Nt = {
  "ar-AE": tt,
  "bg-BG": nt,
  "cs-CZ": rt,
  "da-DK": it,
  "de-DE": st,
  "el-GR": ot,
  "en-US": at,
  "es-ES": lt,
  "et-EE": ct,
  "fi-FI": ut,
  "fr-FR": dt,
  "he-IL": ft,
  "hr-HR": ht,
  "hu-HU": mt,
  "it-IT": pt,
  "ja-JP": gt,
  "ko-KR": bt,
  "lt-LT": vt,
  "lv-LV": $t,
  "nb-NO": yt,
  "nl-NL": Et,
  "pl-PL": wt,
  "pt-BR": _t,
  "pt-PT": xt,
  "ro-RO": Dt,
  "ru-RU": St,
  "sk-SK": Tt,
  "sl-SI": Ot,
  "sr-SP": Ct,
  "sv-SE": Pt,
  "tr-TR": Lt,
  "uk-UA": It,
  "zh-CN": Mt,
  "zh-TW": At
};
const Me = {
  border: 0,
  clip: "rect(0 0 0 0)",
  clipPath: "inset(50%)",
  height: "1px",
  margin: "-1px",
  overflow: "hidden",
  padding: 0,
  position: "absolute",
  width: "1px",
  whiteSpace: "nowrap"
};
function Un(t = {}) {
  let { style: e, isFocusable: n } = t, [r, i] = I(!1), { focusWithinProps: s } = Je({
    isDisabled: !n,
    onFocusWithinChange: (a) => i(a)
  }), o = z(() => r ? e : e ? {
    ...Me,
    ...e
  } : Me, [
    r
  ]);
  return {
    visuallyHiddenProps: {
      ...s,
      style: o
    }
  };
}
function Vn(t) {
  let { children: e, elementType: n = "div", isFocusable: r, style: i, ...s } = t, { visuallyHiddenProps: o } = Un(t);
  return /* @__PURE__ */ $.createElement(n, Y(s, o), e);
}
function jn(t) {
  return t && t.__esModule ? t.default : t;
}
function Kn(t) {
  let { onDismiss: e, ...n } = t, r = In(jn(Nt), "@react-aria/overlays"), i = mn(n, r.format("dismiss")), s = () => {
    e && e();
  };
  return /* @__PURE__ */ $.createElement(Vn, null, /* @__PURE__ */ $.createElement("button", {
    ...i,
    tabIndex: -1,
    onClick: s,
    style: {
      width: 1,
      height: 1
    }
  }));
}
const Wn = typeof HTMLElement < "u" && "inert" in HTMLElement.prototype;
let V = /* @__PURE__ */ new WeakMap(), D = [];
function zn(t, e) {
  let n = bn(t?.[0]), r = e instanceof n.Element ? {
    root: e
  } : e;
  var i;
  let s = (i = r?.root) !== null && i !== void 0 ? i : document.body, o = r?.shouldUseInert && Wn, a = new Set(t), l = /* @__PURE__ */ new Set(), u = (d) => o && d instanceof n.HTMLElement ? d.inert : d.getAttribute("aria-hidden") === "true", c = (d, p) => {
    o && d instanceof n.HTMLElement ? d.inert = p : p ? d.setAttribute("aria-hidden", "true") : (d.removeAttribute("aria-hidden"), d instanceof n.HTMLElement && (d.inert = !1));
  }, m = (d) => {
    for (let y of d.querySelectorAll("[data-live-announcer], [data-react-aria-top-layer]")) a.add(y);
    let p = (y) => {
      if (l.has(y) || a.has(y) || y.parentElement && l.has(y.parentElement) && y.parentElement.getAttribute("role") !== "row") return NodeFilter.FILTER_REJECT;
      for (let C of a)
        if (y.contains(C)) return NodeFilter.FILTER_SKIP;
      return NodeFilter.FILTER_ACCEPT;
    }, g = document.createTreeWalker(d, NodeFilter.SHOW_ELEMENT, {
      acceptNode: p
    }), U = p(d);
    if (U === NodeFilter.FILTER_ACCEPT && f(d), U !== NodeFilter.FILTER_REJECT) {
      let y = g.nextNode();
      for (; y != null; )
        f(y), y = g.nextNode();
    }
  }, f = (d) => {
    var p;
    let g = (p = V.get(d)) !== null && p !== void 0 ? p : 0;
    u(d) && g === 0 || (g === 0 && c(d, !0), l.add(d), V.set(d, g + 1));
  };
  D.length && D[D.length - 1].disconnect(), m(s);
  let b = new MutationObserver((d) => {
    for (let p of d)
      if (p.type === "childList" && ![
        ...a,
        ...l
      ].some((g) => g.contains(p.target)))
        for (let g of p.addedNodes)
          (g instanceof HTMLElement || g instanceof SVGElement) && (g.dataset.liveAnnouncer === "true" || g.dataset.reactAriaTopLayer === "true") ? a.add(g) : g instanceof Element && m(g);
  });
  b.observe(s, {
    childList: !0,
    subtree: !0
  });
  let h = {
    visibleNodes: a,
    hiddenNodes: l,
    observe() {
      b.observe(s, {
        childList: !0,
        subtree: !0
      });
    },
    disconnect() {
      b.disconnect();
    }
  };
  return D.push(h), () => {
    b.disconnect();
    for (let d of l) {
      let p = V.get(d);
      p != null && (p === 1 ? (c(d, !1), V.delete(d)) : V.set(d, p - 1));
    }
    h === D[D.length - 1] ? (D.pop(), D.length && D[D.length - 1].observe()) : D.splice(D.indexOf(h), 1);
  };
}
const kt = /* @__PURE__ */ $.createContext(null);
function Yn(t) {
  let e = me(), { portalContainer: n = e ? null : document.body, isExiting: r } = t, [i, s] = I(!1), o = z(() => ({
    contain: i,
    setContain: s
  }), [
    i,
    s
  ]), { getContainer: a } = Bn();
  if (!t.portalContainer && a && (n = a()), !n) return null;
  let l = t.children;
  return t.disableFocusManagement || (l = /* @__PURE__ */ $.createElement(un, {
    restoreFocus: !0,
    contain: (t.shouldContainFocus || i) && !r
  }, l)), l = /* @__PURE__ */ $.createElement(kt.Provider, {
    value: o
  }, /* @__PURE__ */ $.createElement(Mn, null, l)), /* @__PURE__ */ dn.createPortal(l, n);
}
function Rt() {
  let t = M(kt), e = t?.setContain;
  ne(() => {
    e?.(!0);
  }, [
    e
  ]);
}
function Gn(t, e, n) {
  let { overlayProps: r, underlayProps: i } = Nn({
    ...t,
    isOpen: e.isOpen,
    onClose: e.close
  }, n);
  return kn({
    isDisabled: !e.isOpen
  }), Rt(), k(() => {
    if (e.isOpen && n.current) return zn([
      n.current
    ], {
      shouldUseInert: !0
    });
  }, [
    e.isOpen,
    n
  ]), {
    modalProps: Y(r),
    underlayProps: i
  };
}
function Xn(t, e) {
  let { role: n = "dialog" } = t, r = on();
  r = t["aria-label"] ? void 0 : r;
  let i = N(!1);
  return k(() => {
    if (e.current && !e.current.contains(document.activeElement)) {
      xe(e.current);
      let s = setTimeout(() => {
        (document.activeElement === e.current || document.activeElement === document.body) && (i.current = !0, e.current && (e.current.blur(), xe(e.current)), i.current = !1);
      }, 500);
      return () => {
        clearTimeout(s);
      };
    }
  }, [
    e
  ]), Rt(), {
    dialogProps: {
      ...re(t, {
        labelable: !0
      }),
      role: n,
      tabIndex: -1,
      "aria-labelledby": t["aria-labelledby"] || r,
      // Prevent blur events from reaching useOverlay, which may cause
      // popovers to close. Since focus is contained within the dialog,
      // we don't want this to occur due to the above useEffect.
      onBlur: (s) => {
        i.current && s.stopPropagation();
      }
    },
    titleProps: {
      id: r
    }
  };
}
function qn(t) {
  let [e, n] = hn(t.isOpen, t.defaultOpen || !1, t.onOpenChange);
  const r = F(() => {
    n(!0);
  }, [
    n
  ]), i = F(() => {
    n(!1);
  }, [
    n
  ]), s = F(() => {
    n(!e);
  }, [
    n,
    e
  ]);
  return {
    isOpen: e,
    setOpen: n,
    open: r,
    close: i,
    toggle: s
  };
}
const Jn = /* @__PURE__ */ B(null), oe = /* @__PURE__ */ B(null), Zn = /* @__PURE__ */ A(function(e, n) {
  let r = e["aria-labelledby"];
  [e, n] = pe(e, n, Jn);
  let { dialogProps: i, titleProps: s } = Xn({
    ...e,
    // Only pass aria-labelledby from props, not context.
    // Context is used as a fallback below.
    "aria-labelledby": r
  }, n), o = M(oe);
  !i["aria-label"] && !i["aria-labelledby"] && (e["aria-labelledby"] ? i["aria-labelledby"] = e["aria-labelledby"] : process.env.NODE_ENV !== "production" && console.warn('If a Dialog does not contain a <Heading slot="title">, it must have an aria-label or aria-labelledby attribute for accessibility.'));
  let a = ge({
    defaultClassName: "react-aria-Dialog",
    className: e.className,
    style: e.style,
    children: e.children,
    values: {
      close: o?.close || (() => {
      })
    }
  }), l = re(e, {
    global: !0
  });
  return /* @__PURE__ */ $.createElement("section", {
    ...Y(l, a, i),
    ref: n,
    slot: e.slot || void 0
  }, /* @__PURE__ */ $.createElement(Ge, {
    values: [
      [
        qe,
        {
          slots: {
            [_e]: {},
            title: {
              ...s,
              level: 2
            }
          }
        }
      ],
      [
        sn,
        {
          slots: {
            [_e]: {},
            close: {
              onPress: () => o?.close()
            }
          }
        }
      ]
    ]
  }, a.children));
}), Qn = /* @__PURE__ */ A(function(e, n) {
  [e, n] = pe(e, n, qe);
  let { children: r, level: i = 3, className: s, ...o } = e, a = `h${i}`;
  return /* @__PURE__ */ $.createElement(a, {
    ...o,
    ref: n,
    className: s ?? "react-aria-Heading"
  }, r);
}), er = /* @__PURE__ */ B(null), be = /* @__PURE__ */ B(null), tr = /* @__PURE__ */ A(function(e, n) {
  if (M(be)) {
    if (process.env.NODE_ENV !== "production" && (e.onOpenChange || e.defaultOpen !== void 0 || e.isOpen !== void 0)) {
      const d = /* @__PURE__ */ new Set([
        "isDismissable",
        "isKeyboardDismissDisabled",
        "isOpen",
        "defaultOpen",
        "onOpenChange",
        "isEntering",
        "isExiting",
        "UNSTABLE_portalContainer",
        "shouldCloseOnInteractOutside"
      ]), p = Object.keys(e).filter((g) => d.has(g));
      console.warn(`This modal is already wrapped in a ModalOverlay, props [${p.join(", ")}] should be placed on the ModalOverlay instead.`);
    }
    return /* @__PURE__ */ $.createElement(Ae, {
      ...e,
      modalRef: n
    }, e.children);
  }
  let { isDismissable: i, isKeyboardDismissDisabled: s, isOpen: o, defaultOpen: a, onOpenChange: l, children: u, isEntering: c, isExiting: m, UNSTABLE_portalContainer: f, shouldCloseOnInteractOutside: b, ...h } = e;
  return /* @__PURE__ */ $.createElement(Ht, {
    isDismissable: i,
    isKeyboardDismissDisabled: s,
    isOpen: o,
    defaultOpen: a,
    onOpenChange: l,
    isEntering: c,
    isExiting: m,
    UNSTABLE_portalContainer: f,
    shouldCloseOnInteractOutside: b
  }, /* @__PURE__ */ $.createElement(Ae, {
    ...h,
    modalRef: n
  }, u));
});
function nr(t, e) {
  [t, e] = pe(t, e, er);
  let n = M(oe), r = qn(t), i = t.isOpen != null || t.defaultOpen != null || !n ? r : n;
  i === n && process.env.NODE_ENV !== "production" && (t.onOpenChange || t.defaultOpen !== void 0 || t.isOpen !== void 0) && console.warn("This modals state is controlled by a trigger, place onOpenChange on the trigger instead.");
  let s = Xe(e), o = N(null), a = Te(s, i.isOpen), l = Te(o, i.isOpen), u = a || l || t.isExiting || !1, c = me();
  return !i.isOpen && !u || c ? null : /* @__PURE__ */ $.createElement(rr, {
    ...t,
    state: i,
    isExiting: u,
    overlayRef: s,
    modalRef: o
  });
}
const Ht = /* @__PURE__ */ A(nr);
function rr({ UNSTABLE_portalContainer: t, ...e }) {
  let n = e.modalRef, { state: r } = e, { modalProps: i, underlayProps: s } = Gn(e, r, n), o = Qe(e.overlayRef) || e.isEntering || !1, a = ge({
    ...e,
    defaultClassName: "react-aria-ModalOverlay",
    values: {
      isEntering: o,
      isExiting: e.isExiting,
      state: r
    }
  }), l = xn(), u;
  if (typeof document < "u") {
    let m = ee(document.body) ? document.body : document.scrollingElement || document.documentElement, f = m.getBoundingClientRect().height % 1;
    u = m.scrollHeight - f;
  }
  let c = {
    ...a.style,
    "--visual-viewport-height": l.height + "px",
    "--page-height": u !== void 0 ? u + "px" : void 0
  };
  return /* @__PURE__ */ $.createElement(Yn, {
    isExiting: e.isExiting,
    portalContainer: t
  }, /* @__PURE__ */ $.createElement("div", {
    ...Y(re(e, {
      global: !0
    }), s),
    ...a,
    style: c,
    ref: e.overlayRef,
    "data-entering": o || void 0,
    "data-exiting": e.isExiting || void 0
  }, /* @__PURE__ */ $.createElement(Ge, {
    values: [
      [
        be,
        {
          modalProps: i,
          modalRef: n,
          isExiting: e.isExiting,
          isDismissable: e.isDismissable
        }
      ],
      [
        oe,
        r
      ]
    ]
  }, a.children)));
}
function Ae(t) {
  let { modalProps: e, modalRef: n, isExiting: r, isDismissable: i } = M(be), s = M(oe), o = z(() => an(t.modalRef, n), [
    t.modalRef,
    n
  ]), a = Xe(o), l = Qe(a), u = ge({
    ...t,
    defaultClassName: "react-aria-Modal",
    values: {
      isEntering: l,
      isExiting: r,
      state: s
    }
  });
  return /* @__PURE__ */ $.createElement("div", {
    ...Y(re(t, {
      global: !0
    }), e),
    ...u,
    ref: a,
    "data-entering": l || void 0,
    "data-exiting": r || void 0
  }, i && /* @__PURE__ */ $.createElement(Kn, {
    onDismiss: s.close
  }), u.children);
}
const Ft = B(null);
function Bt() {
  const t = M(Ft);
  if (!t)
    throw new Error("Modal compound components must be used within a Modal");
  return t;
}
const ir = { close: "Schließen" }, sr = { close: "Close" }, or = { close: "Fermer" }, ar = { close: "Chiudi" }, lr = { close: "Sluiten" }, cr = {
  de: ir,
  en: sr,
  fr: or,
  it: ar,
  nl: lr
};
var ur = "_1rhdav60", dr = "_1rhdav61";
const Ut = A(
  ({
    children: t,
    showCloseButton: e = !1,
    closeButtonLabel: n,
    className: r,
    ...i
  }, s) => {
    const { onClose: o } = Bt(), a = wn(cr), l = n ?? a.close;
    return /* @__PURE__ */ ze("div", { ref: s, className: W(ur, r), ...i, children: [
      /* @__PURE__ */ O(
        Qn,
        {
          slot: "title",
          className: W(En["heading.lg"], dr),
          children: t
        }
      ),
      e && /* @__PURE__ */ O(
        yn,
        {
          variant: "ghost",
          color: "neutral",
          size: "sm",
          iconOnly: !0,
          iconStart: /* @__PURE__ */ O($n, {}),
          "aria-label": l,
          onClick: o
        }
      )
    ] });
  }
);
Ut.displayName = "Modal.Header";
var fr = ie({ defaultClassName: "_1mel8e20", variantClassNames: { fullWidth: { true: "_1mel8e21", false: "_1mel8e22" } }, defaultVariants: { fullWidth: !1 }, compoundVariants: [] });
const Vt = A(
  ({ children: t, fullWidth: e = !1, className: n, ...r }, i) => {
    const { bodyRef: s } = Bt();
    return /* @__PURE__ */ O(
      "div",
      {
        ref: (o) => {
          typeof i == "function" ? i(o) : i && (i.current = o), s && (s.current = o);
        },
        className: W(fr({ fullWidth: e }), n),
        ...r,
        children: t
      }
    );
  }
);
Vt.displayName = "Modal.Body";
var hr = "_1p8y7pi0";
const jt = A(
  ({ children: t, className: e, ...n }, r) => /* @__PURE__ */ O("div", { ref: r, className: W(hr, e), ...n, children: t })
);
jt.displayName = "Modal.Footer";
function mr(t, e, n) {
  return Math.max(e, Math.min(t, n));
}
const E = {
  toVector(t, e) {
    return t === void 0 && (t = e), Array.isArray(t) ? t : [t, t];
  },
  add(t, e) {
    return [t[0] + e[0], t[1] + e[1]];
  },
  sub(t, e) {
    return [t[0] - e[0], t[1] - e[1]];
  },
  addTo(t, e) {
    t[0] += e[0], t[1] += e[1];
  },
  subTo(t, e) {
    t[0] -= e[0], t[1] -= e[1];
  }
};
function Ne(t, e, n) {
  return e === 0 || Math.abs(e) === 1 / 0 ? Math.pow(t, n * 5) : t * e * n / (e + n * t);
}
function ke(t, e, n, r = 0.15) {
  return r === 0 ? mr(t, e, n) : t < e ? -Ne(e - t, n - e, r) + e : t > n ? +Ne(t - n, n - e, r) + n : t;
}
function pr(t, [e, n], [r, i]) {
  const [[s, o], [a, l]] = t;
  return [ke(e, s, o, r), ke(n, a, l, i)];
}
function gr(t, e) {
  if (typeof t != "object" || t === null) return t;
  var n = t[Symbol.toPrimitive];
  if (n !== void 0) {
    var r = n.call(t, e);
    if (typeof r != "object") return r;
    throw new TypeError("@@toPrimitive must return a primitive value.");
  }
  return (e === "string" ? String : Number)(t);
}
function br(t) {
  var e = gr(t, "string");
  return typeof e == "symbol" ? e : String(e);
}
function _(t, e, n) {
  return e = br(e), e in t ? Object.defineProperty(t, e, {
    value: n,
    enumerable: !0,
    configurable: !0,
    writable: !0
  }) : t[e] = n, t;
}
function Re(t, e) {
  var n = Object.keys(t);
  if (Object.getOwnPropertySymbols) {
    var r = Object.getOwnPropertySymbols(t);
    e && (r = r.filter(function(i) {
      return Object.getOwnPropertyDescriptor(t, i).enumerable;
    })), n.push.apply(n, r);
  }
  return n;
}
function v(t) {
  for (var e = 1; e < arguments.length; e++) {
    var n = arguments[e] != null ? arguments[e] : {};
    e % 2 ? Re(Object(n), !0).forEach(function(r) {
      _(t, r, n[r]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(n)) : Re(Object(n)).forEach(function(r) {
      Object.defineProperty(t, r, Object.getOwnPropertyDescriptor(n, r));
    });
  }
  return t;
}
const Kt = {
  pointer: {
    start: "down",
    change: "move",
    end: "up"
  },
  mouse: {
    start: "down",
    change: "move",
    end: "up"
  },
  touch: {
    start: "start",
    change: "move",
    end: "end"
  },
  gesture: {
    start: "start",
    change: "change",
    end: "end"
  }
};
function He(t) {
  return t ? t[0].toUpperCase() + t.slice(1) : "";
}
const vr = ["enter", "leave"];
function $r(t = !1, e) {
  return t && !vr.includes(e);
}
function yr(t, e = "", n = !1) {
  const r = Kt[t], i = r && r[e] || e;
  return "on" + He(t) + He(i) + ($r(n, i) ? "Capture" : "");
}
const Er = ["gotpointercapture", "lostpointercapture"];
function wr(t) {
  let e = t.substring(2).toLowerCase();
  const n = !!~e.indexOf("passive");
  n && (e = e.replace("passive", ""));
  const r = Er.includes(e) ? "capturecapture" : "capture", i = !!~e.indexOf(r);
  return i && (e = e.replace("capture", "")), {
    device: e,
    capture: i,
    passive: n
  };
}
function _r(t, e = "") {
  const n = Kt[t], r = n && n[e] || e;
  return t + r;
}
function ae(t) {
  return "touches" in t;
}
function Wt(t) {
  return ae(t) ? "touch" : "pointerType" in t ? t.pointerType : "mouse";
}
function xr(t) {
  return Array.from(t.touches).filter((e) => {
    var n, r;
    return e.target === t.currentTarget || ((n = t.currentTarget) === null || n === void 0 || (r = n.contains) === null || r === void 0 ? void 0 : r.call(n, e.target));
  });
}
function Dr(t) {
  return t.type === "touchend" || t.type === "touchcancel" ? t.changedTouches : t.targetTouches;
}
function zt(t) {
  return ae(t) ? Dr(t)[0] : t;
}
function Sr(t) {
  return xr(t).map((e) => e.identifier);
}
function fe(t) {
  const e = zt(t);
  return ae(t) ? e.identifier : e.pointerId;
}
function Fe(t) {
  const e = zt(t);
  return [e.clientX, e.clientY];
}
function Tr(t) {
  const e = {};
  if ("buttons" in t && (e.buttons = t.buttons), "shiftKey" in t) {
    const {
      shiftKey: n,
      altKey: r,
      metaKey: i,
      ctrlKey: s
    } = t;
    Object.assign(e, {
      shiftKey: n,
      altKey: r,
      metaKey: i,
      ctrlKey: s
    });
  }
  return e;
}
function te(t, ...e) {
  return typeof t == "function" ? t(...e) : t;
}
function Or() {
}
function Cr(...t) {
  return t.length === 0 ? Or : t.length === 1 ? t[0] : function() {
    let e;
    for (const n of t)
      e = n.apply(this, arguments) || e;
    return e;
  };
}
function Be(t, e) {
  return Object.assign({}, e, t || {});
}
const Pr = 32;
class Lr {
  constructor(e, n, r) {
    this.ctrl = e, this.args = n, this.key = r, this.state || (this.state = {}, this.computeValues([0, 0]), this.computeInitial(), this.init && this.init(), this.reset());
  }
  get state() {
    return this.ctrl.state[this.key];
  }
  set state(e) {
    this.ctrl.state[this.key] = e;
  }
  get shared() {
    return this.ctrl.state.shared;
  }
  get eventStore() {
    return this.ctrl.gestureEventStores[this.key];
  }
  get timeoutStore() {
    return this.ctrl.gestureTimeoutStores[this.key];
  }
  get config() {
    return this.ctrl.config[this.key];
  }
  get sharedConfig() {
    return this.ctrl.config.shared;
  }
  get handler() {
    return this.ctrl.handlers[this.key];
  }
  reset() {
    const {
      state: e,
      shared: n,
      ingKey: r,
      args: i
    } = this;
    n[r] = e._active = e.active = e._blocked = e._force = !1, e._step = [!1, !1], e.intentional = !1, e._movement = [0, 0], e._distance = [0, 0], e._direction = [0, 0], e._delta = [0, 0], e._bounds = [[-1 / 0, 1 / 0], [-1 / 0, 1 / 0]], e.args = i, e.axis = void 0, e.memo = void 0, e.elapsedTime = e.timeDelta = 0, e.direction = [0, 0], e.distance = [0, 0], e.overflow = [0, 0], e._movementBound = [!1, !1], e.velocity = [0, 0], e.movement = [0, 0], e.delta = [0, 0], e.timeStamp = 0;
  }
  start(e) {
    const n = this.state, r = this.config;
    n._active || (this.reset(), this.computeInitial(), n._active = !0, n.target = e.target, n.currentTarget = e.currentTarget, n.lastOffset = r.from ? te(r.from, n) : n.offset, n.offset = n.lastOffset, n.startTime = n.timeStamp = e.timeStamp);
  }
  computeValues(e) {
    const n = this.state;
    n._values = e, n.values = this.config.transform(e);
  }
  computeInitial() {
    const e = this.state;
    e._initial = e._values, e.initial = e.values;
  }
  compute(e) {
    const {
      state: n,
      config: r,
      shared: i
    } = this;
    n.args = this.args;
    let s = 0;
    if (e && (n.event = e, r.preventDefault && e.cancelable && n.event.preventDefault(), n.type = e.type, i.touches = this.ctrl.pointerIds.size || this.ctrl.touchIds.size, i.locked = !!document.pointerLockElement, Object.assign(i, Tr(e)), i.down = i.pressed = i.buttons % 2 === 1 || i.touches > 0, s = e.timeStamp - n.timeStamp, n.timeStamp = e.timeStamp, n.elapsedTime = n.timeStamp - n.startTime), n._active) {
      const w = n._delta.map(Math.abs);
      E.addTo(n._distance, w);
    }
    this.axisIntent && this.axisIntent(e);
    const [o, a] = n._movement, [l, u] = r.threshold, {
      _step: c,
      values: m
    } = n;
    if (r.hasCustomTransform ? (c[0] === !1 && (c[0] = Math.abs(o) >= l && m[0]), c[1] === !1 && (c[1] = Math.abs(a) >= u && m[1])) : (c[0] === !1 && (c[0] = Math.abs(o) >= l && Math.sign(o) * l), c[1] === !1 && (c[1] = Math.abs(a) >= u && Math.sign(a) * u)), n.intentional = c[0] !== !1 || c[1] !== !1, !n.intentional) return;
    const f = [0, 0];
    if (r.hasCustomTransform) {
      const [w, P] = m;
      f[0] = c[0] !== !1 ? w - c[0] : 0, f[1] = c[1] !== !1 ? P - c[1] : 0;
    } else
      f[0] = c[0] !== !1 ? o - c[0] : 0, f[1] = c[1] !== !1 ? a - c[1] : 0;
    this.restrictToAxis && !n._blocked && this.restrictToAxis(f);
    const b = n.offset, h = n._active && !n._blocked || n.active;
    h && (n.first = n._active && !n.active, n.last = !n._active && n.active, n.active = i[this.ingKey] = n._active, e && (n.first && ("bounds" in r && (n._bounds = te(r.bounds, n)), this.setup && this.setup()), n.movement = f, this.computeOffset()));
    const [d, p] = n.offset, [[g, U], [y, C]] = n._bounds;
    n.overflow = [d < g ? -1 : d > U ? 1 : 0, p < y ? -1 : p > C ? 1 : 0], n._movementBound[0] = n.overflow[0] ? n._movementBound[0] === !1 ? n._movement[0] : n._movementBound[0] : !1, n._movementBound[1] = n.overflow[1] ? n._movementBound[1] === !1 ? n._movement[1] : n._movementBound[1] : !1;
    const le = n._active ? r.rubberband || [0, 0] : [0, 0];
    if (n.offset = pr(n._bounds, n.offset, le), n.delta = E.sub(n.offset, b), this.computeMovement(), h && (!n.last || s > Pr)) {
      n.delta = E.sub(n.offset, b);
      const w = n.delta.map(Math.abs);
      E.addTo(n.distance, w), n.direction = n.delta.map(Math.sign), n._direction = n._delta.map(Math.sign), !n.first && s > 0 && (n.velocity = [w[0] / s, w[1] / s], n.timeDelta = s);
    }
  }
  emit() {
    const e = this.state, n = this.shared, r = this.config;
    if (e._active || this.clean(), (e._blocked || !e.intentional) && !e._force && !r.triggerAllEvents) return;
    const i = this.handler(v(v(v({}, n), e), {}, {
      [this.aliasKey]: e.values
    }));
    i !== void 0 && (e.memo = i);
  }
  clean() {
    this.eventStore.clean(), this.timeoutStore.clean();
  }
}
function Ir([t, e], n) {
  const r = Math.abs(t), i = Math.abs(e);
  if (r > i && r > n)
    return "x";
  if (i > r && i > n)
    return "y";
}
class Mr extends Lr {
  constructor(...e) {
    super(...e), _(this, "aliasKey", "xy");
  }
  reset() {
    super.reset(), this.state.axis = void 0;
  }
  init() {
    this.state.offset = [0, 0], this.state.lastOffset = [0, 0];
  }
  computeOffset() {
    this.state.offset = E.add(this.state.lastOffset, this.state.movement);
  }
  computeMovement() {
    this.state.movement = E.sub(this.state.offset, this.state.lastOffset);
  }
  axisIntent(e) {
    const n = this.state, r = this.config;
    if (!n.axis && e) {
      const i = typeof r.axisThreshold == "object" ? r.axisThreshold[Wt(e)] : r.axisThreshold;
      n.axis = Ir(n._movement, i);
    }
    n._blocked = (r.lockDirection || !!r.axis) && !n.axis || !!r.axis && r.axis !== n.axis;
  }
  restrictToAxis(e) {
    if (this.config.axis || this.config.lockDirection)
      switch (this.state.axis) {
        case "x":
          e[1] = 0;
          break;
        case "y":
          e[0] = 0;
          break;
      }
  }
}
const Ue = (t) => t, Ve = 0.15, ve = {
  enabled(t = !0) {
    return t;
  },
  eventOptions(t, e, n) {
    return v(v({}, n.shared.eventOptions), t);
  },
  preventDefault(t = !1) {
    return t;
  },
  triggerAllEvents(t = !1) {
    return t;
  },
  rubberband(t = 0) {
    switch (t) {
      case !0:
        return [Ve, Ve];
      case !1:
        return [0, 0];
      default:
        return E.toVector(t);
    }
  },
  from(t) {
    if (typeof t == "function") return t;
    if (t != null) return E.toVector(t);
  },
  transform(t, e, n) {
    const r = t || n.shared.transform;
    if (this.hasCustomTransform = !!r, process.env.NODE_ENV === "development") {
      const i = r || Ue;
      return (s) => {
        const o = i(s);
        return (!isFinite(o[0]) || !isFinite(o[1])) && console.warn(`[@use-gesture]: config.transform() must produce a valid result, but it was: [${o[0]},${[1]}]`), o;
      };
    }
    return r || Ue;
  },
  threshold(t) {
    return E.toVector(t, 0);
  }
};
process.env.NODE_ENV === "development" && Object.assign(ve, {
  domTarget(t) {
    if (t !== void 0)
      throw Error("[@use-gesture]: `domTarget` option has been renamed to `target`.");
    return NaN;
  },
  lockDirection(t) {
    if (t !== void 0)
      throw Error("[@use-gesture]: `lockDirection` option has been merged with `axis`. Use it as in `{ axis: 'lock' }`");
    return NaN;
  },
  initial(t) {
    if (t !== void 0)
      throw Error("[@use-gesture]: `initial` option has been renamed to `from`.");
    return NaN;
  }
});
const Ar = 0, G = v(v({}, ve), {}, {
  axis(t, e, {
    axis: n
  }) {
    if (this.lockDirection = n === "lock", !this.lockDirection) return n;
  },
  axisThreshold(t = Ar) {
    return t;
  },
  bounds(t = {}) {
    if (typeof t == "function")
      return (s) => G.bounds(t(s));
    if ("current" in t)
      return () => t.current;
    if (typeof HTMLElement == "function" && t instanceof HTMLElement)
      return t;
    const {
      left: e = -1 / 0,
      right: n = 1 / 0,
      top: r = -1 / 0,
      bottom: i = 1 / 0
    } = t;
    return [[e, n], [r, i]];
  }
}), je = {
  ArrowRight: (t, e = 1) => [t * e, 0],
  ArrowLeft: (t, e = 1) => [-1 * t * e, 0],
  ArrowUp: (t, e = 1) => [0, -1 * t * e],
  ArrowDown: (t, e = 1) => [0, t * e]
};
class Nr extends Mr {
  constructor(...e) {
    super(...e), _(this, "ingKey", "dragging");
  }
  reset() {
    super.reset();
    const e = this.state;
    e._pointerId = void 0, e._pointerActive = !1, e._keyboardActive = !1, e._preventScroll = !1, e._delayed = !1, e.swipe = [0, 0], e.tap = !1, e.canceled = !1, e.cancel = this.cancel.bind(this);
  }
  setup() {
    const e = this.state;
    if (e._bounds instanceof HTMLElement) {
      const n = e._bounds.getBoundingClientRect(), r = e.currentTarget.getBoundingClientRect(), i = {
        left: n.left - r.left + e.offset[0],
        right: n.right - r.right + e.offset[0],
        top: n.top - r.top + e.offset[1],
        bottom: n.bottom - r.bottom + e.offset[1]
      };
      e._bounds = G.bounds(i);
    }
  }
  cancel() {
    const e = this.state;
    e.canceled || (e.canceled = !0, e._active = !1, setTimeout(() => {
      this.compute(), this.emit();
    }, 0));
  }
  setActive() {
    this.state._active = this.state._pointerActive || this.state._keyboardActive;
  }
  clean() {
    this.pointerClean(), this.state._pointerActive = !1, this.state._keyboardActive = !1, super.clean();
  }
  pointerDown(e) {
    const n = this.config, r = this.state;
    if (e.buttons != null && (Array.isArray(n.pointerButtons) ? !n.pointerButtons.includes(e.buttons) : n.pointerButtons !== -1 && n.pointerButtons !== e.buttons)) return;
    const i = this.ctrl.setEventIds(e);
    n.pointerCapture && e.target.setPointerCapture(e.pointerId), !(i && i.size > 1 && r._pointerActive) && (this.start(e), this.setupPointer(e), r._pointerId = fe(e), r._pointerActive = !0, this.computeValues(Fe(e)), this.computeInitial(), n.preventScrollAxis && Wt(e) !== "mouse" ? (r._active = !1, this.setupScrollPrevention(e)) : n.delay > 0 ? (this.setupDelayTrigger(e), n.triggerAllEvents && (this.compute(e), this.emit())) : this.startPointerDrag(e));
  }
  startPointerDrag(e) {
    const n = this.state;
    n._active = !0, n._preventScroll = !0, n._delayed = !1, this.compute(e), this.emit();
  }
  pointerMove(e) {
    const n = this.state, r = this.config;
    if (!n._pointerActive) return;
    const i = fe(e);
    if (n._pointerId !== void 0 && i !== n._pointerId) return;
    const s = Fe(e);
    if (document.pointerLockElement === e.target ? n._delta = [e.movementX, e.movementY] : (n._delta = E.sub(s, n._values), this.computeValues(s)), E.addTo(n._movement, n._delta), this.compute(e), n._delayed && n.intentional) {
      this.timeoutStore.remove("dragDelay"), n.active = !1, this.startPointerDrag(e);
      return;
    }
    if (r.preventScrollAxis && !n._preventScroll)
      if (n.axis)
        if (n.axis === r.preventScrollAxis || r.preventScrollAxis === "xy") {
          n._active = !1, this.clean();
          return;
        } else {
          this.timeoutStore.remove("startPointerDrag"), this.startPointerDrag(e);
          return;
        }
      else
        return;
    this.emit();
  }
  pointerUp(e) {
    this.ctrl.setEventIds(e);
    try {
      this.config.pointerCapture && e.target.hasPointerCapture(e.pointerId) && e.target.releasePointerCapture(e.pointerId);
    } catch {
      process.env.NODE_ENV === "development" && console.warn("[@use-gesture]: If you see this message, it's likely that you're using an outdated version of `@react-three/fiber`. \n\nPlease upgrade to the latest version.");
    }
    const n = this.state, r = this.config;
    if (!n._active || !n._pointerActive) return;
    const i = fe(e);
    if (n._pointerId !== void 0 && i !== n._pointerId) return;
    this.state._pointerActive = !1, this.setActive(), this.compute(e);
    const [s, o] = n._distance;
    if (n.tap = s <= r.tapsThreshold && o <= r.tapsThreshold, n.tap && r.filterTaps)
      n._force = !0;
    else {
      const [a, l] = n._delta, [u, c] = n._movement, [m, f] = r.swipe.velocity, [b, h] = r.swipe.distance, d = r.swipe.duration;
      if (n.elapsedTime < d) {
        const p = Math.abs(a / n.timeDelta), g = Math.abs(l / n.timeDelta);
        p > m && Math.abs(u) > b && (n.swipe[0] = Math.sign(a)), g > f && Math.abs(c) > h && (n.swipe[1] = Math.sign(l));
      }
    }
    this.emit();
  }
  pointerClick(e) {
    !this.state.tap && e.detail > 0 && (e.preventDefault(), e.stopPropagation());
  }
  setupPointer(e) {
    const n = this.config, r = n.device;
    if (process.env.NODE_ENV === "development")
      try {
        if (r === "pointer" && n.preventScrollDelay === void 0) {
          const i = "uv" in e ? e.sourceEvent.currentTarget : e.currentTarget;
          window.getComputedStyle(i).touchAction === "auto" && console.warn("[@use-gesture]: The drag target has its `touch-action` style property set to `auto`. It is recommended to add `touch-action: 'none'` so that the drag gesture behaves correctly on touch-enabled devices. For more information read this: https://use-gesture.netlify.app/docs/extras/#touch-action.\n\nThis message will only show in development mode. It won't appear in production. If this is intended, you can ignore it.", i);
        }
      } catch {
      }
    n.pointerLock && e.currentTarget.requestPointerLock(), n.pointerCapture || (this.eventStore.add(this.sharedConfig.window, r, "change", this.pointerMove.bind(this)), this.eventStore.add(this.sharedConfig.window, r, "end", this.pointerUp.bind(this)), this.eventStore.add(this.sharedConfig.window, r, "cancel", this.pointerUp.bind(this)));
  }
  pointerClean() {
    this.config.pointerLock && document.pointerLockElement === this.state.currentTarget && document.exitPointerLock();
  }
  preventScroll(e) {
    this.state._preventScroll && e.cancelable && e.preventDefault();
  }
  setupScrollPrevention(e) {
    this.state._preventScroll = !1, kr(e);
    const n = this.eventStore.add(this.sharedConfig.window, "touch", "change", this.preventScroll.bind(this), {
      passive: !1
    });
    this.eventStore.add(this.sharedConfig.window, "touch", "end", n), this.eventStore.add(this.sharedConfig.window, "touch", "cancel", n), this.timeoutStore.add("startPointerDrag", this.startPointerDrag.bind(this), this.config.preventScrollDelay, e);
  }
  setupDelayTrigger(e) {
    this.state._delayed = !0, this.timeoutStore.add("dragDelay", () => {
      this.state._step = [0, 0], this.startPointerDrag(e);
    }, this.config.delay);
  }
  keyDown(e) {
    const n = je[e.key];
    if (n) {
      const r = this.state, i = e.shiftKey ? 10 : e.altKey ? 0.1 : 1;
      this.start(e), r._delta = n(this.config.keyboardDisplacement, i), r._keyboardActive = !0, E.addTo(r._movement, r._delta), this.compute(e), this.emit();
    }
  }
  keyUp(e) {
    e.key in je && (this.state._keyboardActive = !1, this.setActive(), this.compute(e), this.emit());
  }
  bind(e) {
    const n = this.config.device;
    e(n, "start", this.pointerDown.bind(this)), this.config.pointerCapture && (e(n, "change", this.pointerMove.bind(this)), e(n, "end", this.pointerUp.bind(this)), e(n, "cancel", this.pointerUp.bind(this)), e("lostPointerCapture", "", this.pointerUp.bind(this))), this.config.keys && (e("key", "down", this.keyDown.bind(this)), e("key", "up", this.keyUp.bind(this))), this.config.filterTaps && e("click", "", this.pointerClick.bind(this), {
      capture: !0,
      passive: !1
    });
  }
}
function kr(t) {
  "persist" in t && typeof t.persist == "function" && t.persist();
}
const X = typeof window < "u" && window.document && window.document.createElement;
function Yt() {
  return X && "ontouchstart" in window;
}
function Rr() {
  return Yt() || X && window.navigator.maxTouchPoints > 1;
}
function Hr() {
  return X && "onpointerdown" in window;
}
function Fr() {
  return X && "exitPointerLock" in window.document;
}
function Br() {
  try {
    return "constructor" in GestureEvent;
  } catch {
    return !1;
  }
}
const S = {
  isBrowser: X,
  gesture: Br(),
  touch: Yt(),
  touchscreen: Rr(),
  pointer: Hr(),
  pointerLock: Fr()
}, Ur = 250, Vr = 180, jr = 0.5, Kr = 50, Wr = 250, zr = 10, Ke = {
  mouse: 0,
  touch: 0,
  pen: 8
}, Gt = v(v({}, G), {}, {
  device(t, e, {
    pointer: {
      touch: n = !1,
      lock: r = !1,
      mouse: i = !1
    } = {}
  }) {
    return this.pointerLock = r && S.pointerLock, S.touch && n ? "touch" : this.pointerLock ? "mouse" : S.pointer && !i ? "pointer" : S.touch ? "touch" : "mouse";
  },
  preventScrollAxis(t, e, {
    preventScroll: n
  }) {
    if (this.preventScrollDelay = typeof n == "number" ? n : n || n === void 0 && t ? Ur : void 0, !(!S.touchscreen || n === !1))
      return t || (n !== void 0 ? "y" : void 0);
  },
  pointerCapture(t, e, {
    pointer: {
      capture: n = !0,
      buttons: r = 1,
      keys: i = !0
    } = {}
  }) {
    return this.pointerButtons = r, this.keys = i, !this.pointerLock && this.device === "pointer" && n;
  },
  threshold(t, e, {
    filterTaps: n = !1,
    tapsThreshold: r = 3,
    axis: i = void 0
  }) {
    const s = E.toVector(t, n ? r : i ? 1 : 0);
    return this.filterTaps = n, this.tapsThreshold = r, s;
  },
  swipe({
    velocity: t = jr,
    distance: e = Kr,
    duration: n = Wr
  } = {}) {
    return {
      velocity: this.transform(E.toVector(t)),
      distance: this.transform(E.toVector(e)),
      duration: n
    };
  },
  delay(t = 0) {
    switch (t) {
      case !0:
        return Vr;
      case !1:
        return 0;
      default:
        return t;
    }
  },
  axisThreshold(t) {
    return t ? v(v({}, Ke), t) : Ke;
  },
  keyboardDisplacement(t = zr) {
    return t;
  }
});
process.env.NODE_ENV === "development" && Object.assign(Gt, {
  useTouch(t) {
    if (t !== void 0)
      throw Error("[@use-gesture]: `useTouch` option has been renamed to `pointer.touch`. Use it as in `{ pointer: { touch: true } }`.");
    return NaN;
  },
  experimental_preventWindowScrollY(t) {
    if (t !== void 0)
      throw Error("[@use-gesture]: `experimental_preventWindowScrollY` option has been renamed to `preventScroll`.");
    return NaN;
  },
  swipeVelocity(t) {
    if (t !== void 0)
      throw Error("[@use-gesture]: `swipeVelocity` option has been renamed to `swipe.velocity`. Use it as in `{ swipe: { velocity: 0.5 } }`.");
    return NaN;
  },
  swipeDistance(t) {
    if (t !== void 0)
      throw Error("[@use-gesture]: `swipeDistance` option has been renamed to `swipe.distance`. Use it as in `{ swipe: { distance: 50 } }`.");
    return NaN;
  },
  swipeDuration(t) {
    if (t !== void 0)
      throw Error("[@use-gesture]: `swipeDuration` option has been renamed to `swipe.duration`. Use it as in `{ swipe: { duration: 250 } }`.");
    return NaN;
  }
});
v(v({}, ve), {}, {
  device(t, e, {
    shared: n,
    pointer: {
      touch: r = !1
    } = {}
  }) {
    if (n.target && !S.touch && S.gesture) return "gesture";
    if (S.touch && r) return "touch";
    if (S.touchscreen) {
      if (S.pointer) return "pointer";
      if (S.touch) return "touch";
    }
  },
  bounds(t, e, {
    scaleBounds: n = {},
    angleBounds: r = {}
  }) {
    const i = (o) => {
      const a = Be(te(n, o), {
        min: -1 / 0,
        max: 1 / 0
      });
      return [a.min, a.max];
    }, s = (o) => {
      const a = Be(te(r, o), {
        min: -1 / 0,
        max: 1 / 0
      });
      return [a.min, a.max];
    };
    return typeof n != "function" && typeof r != "function" ? [i(), s()] : (o) => [i(o), s(o)];
  },
  threshold(t, e, n) {
    return this.lockDirection = n.axis === "lock", E.toVector(t, this.lockDirection ? [0.1, 3] : 0);
  },
  modifierKey(t) {
    return t === void 0 ? "ctrlKey" : t;
  },
  pinchOnWheel(t = !0) {
    return t;
  }
});
v(v({}, G), {}, {
  mouseOnly: (t = !0) => t
});
v(v({}, G), {}, {
  mouseOnly: (t = !0) => t
});
const Xt = /* @__PURE__ */ new Map(), he = /* @__PURE__ */ new Map();
function Yr(t) {
  Xt.set(t.key, t.engine), he.set(t.key, t.resolver);
}
const Gr = {
  key: "drag",
  engine: Nr,
  resolver: Gt
};
function Xr(t, e) {
  if (t == null) return {};
  var n = {}, r = Object.keys(t), i, s;
  for (s = 0; s < r.length; s++)
    i = r[s], !(e.indexOf(i) >= 0) && (n[i] = t[i]);
  return n;
}
function qr(t, e) {
  if (t == null) return {};
  var n = Xr(t, e), r, i;
  if (Object.getOwnPropertySymbols) {
    var s = Object.getOwnPropertySymbols(t);
    for (i = 0; i < s.length; i++)
      r = s[i], !(e.indexOf(r) >= 0) && Object.prototype.propertyIsEnumerable.call(t, r) && (n[r] = t[r]);
  }
  return n;
}
const Jr = {
  target(t) {
    if (t)
      return () => "current" in t ? t.current : t;
  },
  enabled(t = !0) {
    return t;
  },
  window(t = S.isBrowser ? window : void 0) {
    return t;
  },
  eventOptions({
    passive: t = !0,
    capture: e = !1
  } = {}) {
    return {
      passive: t,
      capture: e
    };
  },
  transform(t) {
    return t;
  }
}, Zr = ["target", "eventOptions", "window", "enabled", "transform"];
function Q(t = {}, e) {
  const n = {};
  for (const [r, i] of Object.entries(e))
    switch (typeof i) {
      case "function":
        if (process.env.NODE_ENV === "development") {
          const s = i.call(n, t[r], r, t);
          Number.isNaN(s) || (n[r] = s);
        } else
          n[r] = i.call(n, t[r], r, t);
        break;
      case "object":
        n[r] = Q(t[r], i);
        break;
      case "boolean":
        i && (n[r] = t[r]);
        break;
    }
  return n;
}
function Qr(t, e, n = {}) {
  const r = t, {
    target: i,
    eventOptions: s,
    window: o,
    enabled: a,
    transform: l
  } = r, u = qr(r, Zr);
  if (n.shared = Q({
    target: i,
    eventOptions: s,
    window: o,
    enabled: a,
    transform: l
  }, Jr), e) {
    const c = he.get(e);
    n[e] = Q(v({
      shared: n.shared
    }, u), c);
  } else
    for (const c in u) {
      const m = he.get(c);
      if (m)
        n[c] = Q(v({
          shared: n.shared
        }, u[c]), m);
      else if (process.env.NODE_ENV === "development" && !["drag", "pinch", "scroll", "wheel", "move", "hover"].includes(c)) {
        if (c === "domTarget")
          throw Error("[@use-gesture]: `domTarget` option has been renamed to `target`.");
        console.warn(`[@use-gesture]: Unknown config key \`${c}\` was used. Please read the documentation for further information.`);
      }
    }
  return n;
}
class qt {
  constructor(e, n) {
    _(this, "_listeners", /* @__PURE__ */ new Set()), this._ctrl = e, this._gestureKey = n;
  }
  add(e, n, r, i, s) {
    const o = this._listeners, a = _r(n, r), l = this._gestureKey ? this._ctrl.config[this._gestureKey].eventOptions : {}, u = v(v({}, l), s);
    e.addEventListener(a, i, u);
    const c = () => {
      e.removeEventListener(a, i, u), o.delete(c);
    };
    return o.add(c), c;
  }
  clean() {
    this._listeners.forEach((e) => e()), this._listeners.clear();
  }
}
class ei {
  constructor() {
    _(this, "_timeouts", /* @__PURE__ */ new Map());
  }
  add(e, n, r = 140, ...i) {
    this.remove(e), this._timeouts.set(e, window.setTimeout(n, r, ...i));
  }
  remove(e) {
    const n = this._timeouts.get(e);
    n && window.clearTimeout(n);
  }
  clean() {
    this._timeouts.forEach((e) => void window.clearTimeout(e)), this._timeouts.clear();
  }
}
class ti {
  constructor(e) {
    _(this, "gestures", /* @__PURE__ */ new Set()), _(this, "_targetEventStore", new qt(this)), _(this, "gestureEventStores", {}), _(this, "gestureTimeoutStores", {}), _(this, "handlers", {}), _(this, "config", {}), _(this, "pointerIds", /* @__PURE__ */ new Set()), _(this, "touchIds", /* @__PURE__ */ new Set()), _(this, "state", {
      shared: {
        shiftKey: !1,
        metaKey: !1,
        ctrlKey: !1,
        altKey: !1
      }
    }), ni(this, e);
  }
  setEventIds(e) {
    if (ae(e))
      return this.touchIds = new Set(Sr(e)), this.touchIds;
    if ("pointerId" in e)
      return e.type === "pointerup" || e.type === "pointercancel" ? this.pointerIds.delete(e.pointerId) : e.type === "pointerdown" && this.pointerIds.add(e.pointerId), this.pointerIds;
  }
  applyHandlers(e, n) {
    this.handlers = e, this.nativeHandlers = n;
  }
  applyConfig(e, n) {
    this.config = Qr(e, n, this.config);
  }
  clean() {
    this._targetEventStore.clean();
    for (const e of this.gestures)
      this.gestureEventStores[e].clean(), this.gestureTimeoutStores[e].clean();
  }
  effect() {
    return this.config.shared.target && this.bind(), () => this._targetEventStore.clean();
  }
  bind(...e) {
    const n = this.config.shared, r = {};
    let i;
    if (!(n.target && (i = n.target(), !i))) {
      if (n.enabled) {
        for (const o of this.gestures) {
          const a = this.config[o], l = We(r, a.eventOptions, !!i);
          if (a.enabled) {
            const u = Xt.get(o);
            new u(this, e, o).bind(l);
          }
        }
        const s = We(r, n.eventOptions, !!i);
        for (const o in this.nativeHandlers)
          s(o, "", (a) => this.nativeHandlers[o](v(v({}, this.state.shared), {}, {
            event: a,
            args: e
          })), void 0, !0);
      }
      for (const s in r)
        r[s] = Cr(...r[s]);
      if (!i) return r;
      for (const s in r) {
        const {
          device: o,
          capture: a,
          passive: l
        } = wr(s);
        this._targetEventStore.add(i, o, "", r[s], {
          capture: a,
          passive: l
        });
      }
    }
  }
}
function H(t, e) {
  t.gestures.add(e), t.gestureEventStores[e] = new qt(t, e), t.gestureTimeoutStores[e] = new ei();
}
function ni(t, e) {
  e.drag && H(t, "drag"), e.wheel && H(t, "wheel"), e.scroll && H(t, "scroll"), e.move && H(t, "move"), e.pinch && H(t, "pinch"), e.hover && H(t, "hover");
}
const We = (t, e, n) => (r, i, s, o = {}, a = !1) => {
  var l, u;
  const c = (l = o.capture) !== null && l !== void 0 ? l : e.capture, m = (u = o.passive) !== null && u !== void 0 ? u : e.passive;
  let f = a ? r : yr(r, i, c);
  n && m && (f += "Passive"), t[f] = t[f] || [], t[f].push(s);
};
function ri(t, e = {}, n, r) {
  const i = $.useMemo(() => new ti(t), []);
  if (i.applyHandlers(t, r), i.applyConfig(e, n), $.useEffect(i.effect.bind(i)), $.useEffect(() => i.clean.bind(i), []), e.target === void 0)
    return i.bind.bind(i);
}
function ii(t, e) {
  return Yr(Gr), ri({
    drag: t
  }, e || {}, "drag");
}
const si = 0.3, oi = 0.5, ai = 0.08, li = "transform 300ms cubic-bezier(0.32, 0.72, 0, 1)", Jt = 200, ci = `transform ${Jt}ms ease-in`;
function ui() {
  return typeof window > "u" ? !1 : window.matchMedia("(prefers-reduced-motion: reduce)").matches;
}
function di(t) {
  if (typeof window > "u") return () => {
  };
  const e = window.matchMedia("(prefers-reduced-motion: reduce)");
  return e.addEventListener("change", t), () => e.removeEventListener("change", t);
}
function fi() {
  return !1;
}
function hi() {
  return rn(
    di,
    ui,
    fi
  );
}
function mi(t, e) {
  return t < 0 ? t * e : t;
}
function pi({
  enabled: t,
  open: e,
  onDismiss: n,
  modalRef: r,
  dismissThresholdPercent: i = si,
  velocityThreshold: s = oi,
  rubberBandFactor: o = ai
}) {
  const [a, l] = I(0), [u, c] = I(!1), [m, f] = I(!1), b = N(null), h = hi(), d = N(!1), p = N(n);
  p.current = n, k(() => {
    e && (l(0), c(!1), f(!1), d.current = !1);
  }, [e]), k(() => {
    if (!t) return;
    const w = () => {
      d.current && (d.current = !1, c(!1), l(0));
    };
    return document.addEventListener("pointerup", w), () => {
      document.removeEventListener("pointerup", w);
    };
  }, [t]);
  const g = F(
    (w) => {
      const P = b.current;
      if (!P || !(P.scrollHeight > P.clientHeight)) return !1;
      const q = w.target;
      return q !== null && P.contains(q);
    },
    []
  ), y = ii(
    ({
      down: w,
      movement: [, P],
      velocity: [, $e],
      direction: [, q],
      cancel: Qt,
      canceled: en,
      first: ye,
      last: tn,
      event: Ee
    }) => {
      if (t) {
        if (en) {
          d.current = !1, c(!1), l(0);
          return;
        }
        if (ye && Ee && g(Ee)) {
          Qt();
          return;
        }
        if (ye && f(!1), w)
          d.current = !0, c(!0), l(mi(P, o));
        else if (tn) {
          d.current = !1, c(!1);
          const we = r.current?.offsetHeight ?? 0, nn = we * i;
          $e > s && q > 0 || P > nn ? (f(!0), l(we), setTimeout(() => {
            p.current();
          }, Jt)) : l(0);
        }
      }
    },
    {
      axis: "y",
      filterTaps: !0
    }
  )();
  let C;
  h || u ? C = "none" : m ? C = ci : C = li;
  const le = t ? {
    transform: a !== 0 ? `translateY(${a}px)` : void 0,
    transition: C,
    // When drag-dismissing, disable the CSS exit animation and use our inline transition
    ...m && {
      animation: "none",
      animationName: "none",
      animationDuration: "0s"
    }
  } : {};
  return {
    dragProps: y,
    style: le,
    bodyRef: b
  };
}
var gi = ie({ defaultClassName: "_1omnnrs7 _1omnnrs6", variantClassNames: { placement: { bottom: "_1omnnrs8", center: "_1omnnrs9" } }, defaultVariants: { placement: "bottom" }, compoundVariants: [] }), bi = ie({ defaultClassName: "_1omnnrsb _1omnnrsa", variantClassNames: { placement: { bottom: "_1omnnrsc", center: "_1omnnrsd" }, width: { sm: "_1omnnrse", md: "_1omnnrsf", lg: "_1omnnrsg" }, maxHeight: { full: "_1omnnrsh", sm: "_1omnnrsi", md: "_1omnnrsj", lg: "_1omnnrsk", xl: "_1omnnrsl" } }, defaultVariants: { placement: "bottom", width: "md", maxHeight: "full" }, compoundVariants: [[{ placement: "center", width: "sm" }, "_1omnnrsm"], [{ placement: "center", width: "md" }, "_1omnnrsn"], [{ placement: "center", width: "lg" }, "_1omnnrso"]] }), vi = ie({ defaultClassName: "_1omnnrsp", variantClassNames: { placement: { bottom: "_1omnnrsq", center: "_1omnnrsr" }, draggable: { true: "_1omnnrss", false: "_1omnnrst" } }, defaultVariants: { placement: "bottom", draggable: !1 }, compoundVariants: [[{ placement: "bottom", draggable: !0 }, "_1omnnrsu"]] }), $i = "_1omnnrsv", yi = "_1omnnrsw";
const Gi = ["bottom", "center"], Ei = "bottom", Xi = ["sm", "md", "lg"];
L.components.modal.width.sm, L.components.modal.width.md, L.components.modal.width.lg;
const wi = "md", qi = ["full", "sm", "md", "lg", "xl"];
`${L.spacing[32]}`, L.components.modal.maxHeight.sm, L.components.modal.maxHeight.md, L.components.modal.maxHeight.lg, L.components.modal.maxHeight.xl;
const _i = "full", Zt = A(
  ({
    open: t,
    onOpenChange: e,
    placement: n = Ei,
    width: r = wi,
    maxHeight: i = _i,
    dismissable: s = !0,
    draggable: o = !1,
    children: a,
    className: l,
    ...u
  }, c) => {
    const m = () => {
      e(!1);
    }, f = N(null), b = o && n === "bottom", {
      dragProps: h,
      style: d,
      bodyRef: p
    } = pi({
      enabled: b,
      open: t,
      onDismiss: m,
      modalRef: f
    });
    return /* @__PURE__ */ O(
      Ht,
      {
        isOpen: t,
        onOpenChange: e,
        isDismissable: s,
        className: W(gi({ placement: n }), l),
        ...u,
        children: /* @__PURE__ */ O(
          tr,
          {
            ref: (g) => {
              typeof c == "function" ? c(g) : c && (c.current = g), f.current = g;
            },
            className: bi({ placement: n, width: r, maxHeight: i }),
            style: d,
            children: /* @__PURE__ */ ze(
              Zn,
              {
                className: vi({ placement: n, draggable: b }),
                ...b ? h : {},
                children: [
                  b && /* @__PURE__ */ O("div", { className: $i, "aria-hidden": "true", children: /* @__PURE__ */ O("div", { className: yi }) }),
                  /* @__PURE__ */ O(
                    Ft.Provider,
                    {
                      value: { placement: n, onClose: m, bodyRef: p },
                      children: a
                    }
                  )
                ]
              }
            )
          }
        )
      }
    );
  }
);
Zt.displayName = "Modal";
const Ji = Object.assign(Zt, {
  Header: Ut,
  Body: Vt,
  Footer: jt
});
export {
  _i as DEFAULT_MODAL_MAX_HEIGHT,
  Ei as DEFAULT_MODAL_PLACEMENT,
  wi as DEFAULT_MODAL_WIDTH,
  Ji as Modal,
  qi as ModalMaxHeights,
  Gi as ModalPlacements,
  Xi as ModalWidths
};
