// @angi/one-design-system/dist/components/es/_docs/index.js
import { jsxs as l, Fragment as i, jsx as t } from "react/jsx-runtime";
import { useId as a } from "react";
const c = {
  info: {
    backgroundColor: "#dbeafe",
    borderColor: "#3b82f6",
    textColor: "#1e3a5f",
    // Dark blue-gray for contrast on light blue bg
    linkColor: "#1e40af"
    // Darker blue for links
  },
  warning: {
    backgroundColor: "#fef3c7",
    borderColor: "#f59e0b",
    textColor: "#78350f",
    // Dark brown for contrast on light yellow bg
    linkColor: "#92400e"
    // Dark amber/brown for links
  },
  error: {
    backgroundColor: "#fee2e2",
    borderColor: "#ef4444",
    textColor: "#7f1d1d",
    // Dark red for contrast on light red bg
    linkColor: "#991b1b"
    // Dark red for links
  }
}, s = (o, e) => `
  .${o} a {
    color: ${e};
    font-weight: 600;
    text-decoration: underline;
  }
  .${o} a:hover {
    text-decoration: none;
  }
  .${o} code {
    background-color: rgba(0, 0, 0, 0.06);
    color: rgba(0, 0, 0, 0.85);
    padding: 2px 6px;
    border-radius: 4px;
    font-size: 0.9em;
    font-weight: 600;
  }
  .${o}-content > *:first-child {
    margin-top: 0;
  }
  .${o}-content > *:last-child {
    margin-bottom: 0;
  }
`;
function g({ title: o, children: e, variant: d = "info" }) {
  const r = c[d], n = `docs-callout-${a().replace(/:/g, "")}`;
  return /* @__PURE__ */ l(i, { children: [
    /* @__PURE__ */ t("style", { children: s(n, r.linkColor) }),
    /* @__PURE__ */ l(
      "div",
      {
        className: n,
        style: {
          padding: "16px",
          backgroundColor: r.backgroundColor,
          borderLeft: `4px solid ${r.borderColor}`,
          borderRadius: "4px",
          marginBottom: "24px",
          color: r.textColor
        },
        children: [
          o && /* @__PURE__ */ t("strong", { children: o }),
          /* @__PURE__ */ t(
            "div",
            {
              className: `${n}-content`,
              style: { marginTop: o ? "8px" : 0 },
              children: e
            }
          )
        ]
      }
    )
  ] });
}
export {
  g as Callout
};
