// @angi/one-design-system/dist/components/es/ListItemAction/index.js
import { jsx as t, jsxs as o } from "react/jsx-runtime";
import { forwardRef as c } from "react";
import { $ as I } from "../_chunks/Button-CHUd1jP9.js";
import { c as l } from "../_chunks/clsx-OuTLNxxd.js";
import '../assets/Text.css.ts.css';import '../assets/List.css.ts.css';import '../assets/ListItemAction.css';/* empty css                                      */
/* empty css                                      */
import { c as N } from "../_chunks/createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import { _ as f } from "../_chunks/_BaseIcon-BAJuKzUY.js";
import { _ as A } from "../_chunks/_BaseLink-COWMopm2.js";
var d = N({ defaultClassName: "_16x5nis1 _1sprrkw4 _16x5nis0", variantClassNames: { disabled: { true: "_16x5nis2", false: "_16x5nis3" } }, defaultVariants: { disabled: !1 }, compoundVariants: [] }), p = "_16x5nis4";
const u = c(({ children: s, icon: a, disabled: i = !1, className: e, as: x, ...n }, m) => {
  const r = l(
    d({ disabled: i }),
    e
  );
  return /* @__PURE__ */ o(
    A,
    {
      ref: m,
      className: r,
      isDisabled: i,
      ...n,
      children: [
        /* @__PURE__ */ t("span", { className: p, children: s }),
        /* @__PURE__ */ t(f, { size: 16, children: a })
      ]
    }
  );
});
u.displayName = "ListItemActionLink";
const _ = c(({ children: s, icon: a, disabled: i = !1, className: e, as: x, ...n }, m) => {
  const r = l(
    d({ disabled: i }),
    e
  );
  return /* @__PURE__ */ o(
    I,
    {
      ref: m,
      className: r,
      isDisabled: i,
      ...n,
      children: [
        /* @__PURE__ */ t("span", { className: p, children: s }),
        /* @__PURE__ */ t(f, { size: 16, children: a })
      ]
    }
  );
});
_.displayName = "ListItemActionButton";
function L(s) {
  return s.as === "button" ? /* @__PURE__ */ t(_, { ...s }) : /* @__PURE__ */ t(u, { ...s });
}
L.displayName = "ListItemAction";
export {
  L as ListItemAction
};
