// @angi/one-design-system/dist/components/es/_chunks/_BaseLink-COWMopm2.js
import { jsx as D } from "react/jsx-runtime";
import _, { forwardRef as m, createContext as h } from "react";
import { b as u, $ as F, a as L } from "./utils-CjtkL41C.js";
import { $ as p } from "./filterDOMProps-Cxz9cAAv.js";
import { e as g, f as y, g as E } from "./openLink-EIDcTMUl.js";
import { $ as H } from "./usePress-D9Qp6Amx.js";
import { a as N } from "./useFocusable-CrxUcjB4.js";
import { $ as S } from "./useHover-CfJvhePa.js";
import { $ as T } from "./useFocusRing-CdwnmTWa.js";
import { c as O } from "./clsx-OuTLNxxd.js";
import '../assets/_BaseLink.css';function R(s, e) {
  let { elementType: a = "a", onPress: d, onPressStart: o, onPressEnd: r, onClick: f, isDisabled: i, ...$ } = s, n = {};
  a !== "a" && (n = {
    role: "link",
    tabIndex: i ? void 0 : 0
  });
  let { focusableProps: l } = N(s, e), { pressProps: t, isPressed: c } = H({
    onPress: d,
    onPressStart: o,
    onPressEnd: r,
    onClick: f,
    isDisabled: i,
    ref: e
  }), x = p($, {
    labelable: !0
  }), k = u(l, t), v = g(), C = y(s);
  return {
    isPressed: c,
    linkProps: u(x, C, {
      ...k,
      ...n,
      "aria-disabled": i || void 0,
      "aria-current": s["aria-current"],
      onClick: (P) => {
        var b;
        (b = t.onClick) === null || b === void 0 || b.call(t, P), E(P, v, s.href, s.routerOptions);
      }
    })
  };
}
const V = /* @__PURE__ */ h(null), w = /* @__PURE__ */ m(function(e, a) {
  [e, a] = F(e, a, V);
  let d = e.href && !e.isDisabled ? "a" : "span", { linkProps: o, isPressed: r } = R({
    ...e,
    elementType: d
  }, a), { hoverProps: f, isHovered: i } = S(e), { focusProps: $, isFocused: n, isFocusVisible: l } = T(), t = L({
    ...e,
    defaultClassName: "react-aria-Link",
    values: {
      isCurrent: !!e["aria-current"],
      isDisabled: e.isDisabled || !1,
      isPressed: r,
      isHovered: i,
      isFocused: n,
      isFocusVisible: l
    }
  }), c = p(e, {
    global: !0
  });
  return delete c.onClick, /* @__PURE__ */ _.createElement(d, {
    ref: a,
    slot: e.slot || void 0,
    ...u(c, t, o, f, $),
    "data-focused": n || void 0,
    "data-hovered": i || void 0,
    "data-pressed": r || void 0,
    "data-focus-visible": l || void 0,
    "data-current": !!e["aria-current"] || void 0,
    "data-disabled": e.isDisabled || void 0
  }, t.children);
});
var B = "_7wz7le0";
const j = m(
  ({ className: s, fullPageNavigation: e, routerOptions: a, ...d }, o) => {
    const r = e ? { ...a, fullPageNavigation: !0 } : a;
    return /* @__PURE__ */ D(
      w,
      {
        ref: o,
        className: O(B, s),
        ...r && { routerOptions: r },
        ...d
      }
    );
  }
);
j.displayName = "_BaseLink";
export {
  j as _
};
