// @angi/one-design-system/dist/components/es/_chunks/Button-CG9o4a5a.js
import { jsx as i, jsxs as u, Fragment as B } from "react/jsx-runtime";
import { forwardRef as N, isValidElement as y, cloneElement as T } from "react";
import { c as z } from "./clsx-OuTLNxxd.js";
import { $ as O } from "./Button-CHUd1jP9.js";
import { S as _ } from "./index-BdJ5ogLC.js";
import { _ as j } from "./_BaseIcon-BAJuKzUY.js";
import { S as C } from "./Spinner-B7L4uec2.js";
import { c as S } from "./createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import '../assets/Button.css';var U = S({ defaultClassName: "hofjdi1 hofjdi0", variantClassNames: { variant: { solid: "hofjdi2", outline: "hofjdi3", ghost: "hofjdi4" }, color: { primary: "hofjdi5", neutral: "hofjdi6", danger: "hofjdi7", inverted: "hofjdi8" }, size: { sm: "hofjdi9", md: "hofjdia", lg: "hofjdib" }, iconOnly: { true: "hofjdic", false: "hofjdid" } }, defaultVariants: { variant: "solid", color: "primary", size: "lg" }, compoundVariants: [[{ iconOnly: !0, size: "sm" }, "hofjdie"], [{ iconOnly: !0, size: "md" }, "hofjdif"], [{ iconOnly: !0, size: "lg" }, "hofjdig"], [{ variant: "solid", color: "primary" }, "hofjdih"], [{ variant: "solid", color: "neutral" }, "hofjdii"], [{ variant: "solid", color: "inverted" }, "hofjdij"], [{ variant: "solid", color: "danger" }, "hofjdik"], [{ variant: "outline", color: "primary" }, "hofjdil"], [{ variant: "outline", color: "neutral" }, "hofjdim"], [{ variant: "outline", color: "inverted" }, "hofjdin"], [{ variant: "outline", color: "danger" }, "hofjdio"], [{ variant: "ghost", color: "primary" }, "hofjdip"], [{ variant: "ghost", color: "neutral" }, "hofjdiq"], [{ variant: "ghost", color: "inverted" }, "hofjdir"], [{ variant: "ghost", color: "danger" }, "hofjdis"]] }), x = "hofjdit", A = "hofjdiu";
const Z = ["solid", "outline", "ghost"], G = [
  "primary",
  "neutral",
  "danger",
  "inverted"
], H = ["sm", "md", "lg"], D = "solid", E = "primary", F = "lg", L = ({
  loading: o,
  spinnerSize: a,
  children: r
}) => o ? /* @__PURE__ */ u(B, { children: [
  /* @__PURE__ */ i("span", { className: x, children: r }),
  /* @__PURE__ */ i("span", { className: A, children: /* @__PURE__ */ i(
    C,
    {
      size: a,
      ...typeof o == "object" && o.loading ? { label: o.label } : {}
    }
  ) })
] }) : r, p = ({
  iconStart: o,
  iconEnd: a,
  iconOnly: r,
  loading: n,
  spinnerSize: t,
  children: e
}) => /* @__PURE__ */ u(L, { loading: n, spinnerSize: t, children: [
  o && /* @__PURE__ */ i(j, { size: 16, children: o }),
  !r && e,
  !r && a && /* @__PURE__ */ i(j, { size: 16, children: a })
] }), V = N(
  ({
    variant: o = D,
    color: a = E,
    size: r = F,
    className: n,
    children: t,
    iconStart: e,
    iconEnd: d,
    iconOnly: l = !1,
    loading: s = !1,
    disabled: f = !1,
    asChild: v = !1,
    ...g
  }, c) => {
    const h = z(
      U({
        variant: o,
        color: a,
        size: r,
        iconOnly: l
      }),
      n
    ), m = Object.fromEntries(
      Object.entries(g).filter(([, b]) => b !== void 0)
    );
    return v && y(t) ? /* @__PURE__ */ i(
      _,
      {
        ref: c,
        ...m,
        className: h,
        "aria-disabled": f || !!s,
        children: T(
          t,
          {},
          /* @__PURE__ */ i(
            p,
            {
              iconStart: e,
              iconEnd: d,
              iconOnly: l,
              loading: s,
              spinnerSize: r,
              children: t.props.children
            }
          )
        )
      }
    ) : /* @__PURE__ */ i(
      O,
      {
        ref: c,
        ...m,
        className: h,
        isDisabled: f || !!s,
        children: /* @__PURE__ */ i(
          p,
          {
            iconStart: e,
            iconEnd: d,
            iconOnly: l,
            loading: s,
            spinnerSize: r,
            children: t
          }
        )
      }
    );
  }
);
V.displayName = "Button";
export {
  V as B,
  D,
  Z as a,
  G as b,
  H as c,
  E as d,
  F as e
};
