// @angi/one-design-system/dist/components/es/_chunks/Button-CHUd1jP9.js
import { b as x, $ as _, a as D, d as y } from "./utils-CjtkL41C.js";
import { $ as F } from "./ProgressBar-CJx2i27x.js";
import { $ as I } from "./Label-DLRS1VtT.js";
import { $ as C } from "./filterDOMProps-Cxz9cAAv.js";
import L, { createContext as N, useRef as O, useEffect as V } from "react";
import { $ as w } from "./usePress-D9Qp6Amx.js";
import { a as M } from "./useFocusable-CrxUcjB4.js";
import { $ as R } from "./useFocusRing-CdwnmTWa.js";
import { $ as S } from "./useHover-CfJvhePa.js";
const T = 7e3;
let r = null;
function E(t, e = "assertive", i = T) {
  r ? r.announce(t, e, i) : (r = new k(), (typeof IS_REACT_ACT_ENVIRONMENT == "boolean" ? IS_REACT_ACT_ENVIRONMENT : typeof jest < "u") ? r.announce(t, e, i) : setTimeout(() => {
    r?.isAttached() && r?.announce(t, e, i);
  }, 100));
}
class k {
  isAttached() {
    var e;
    return (e = this.node) === null || e === void 0 ? void 0 : e.isConnected;
  }
  createLog(e) {
    let i = document.createElement("div");
    return i.setAttribute("role", "log"), i.setAttribute("aria-live", e), i.setAttribute("aria-relevant", "additions"), i;
  }
  destroy() {
    this.node && (document.body.removeChild(this.node), this.node = null);
  }
  announce(e, i = "assertive", d = T) {
    var a, n;
    if (!this.node) return;
    let o = document.createElement("div");
    typeof e == "object" ? (o.setAttribute("role", "img"), o.setAttribute("aria-labelledby", e["aria-labelledby"])) : o.textContent = e, i === "assertive" ? (a = this.assertiveLog) === null || a === void 0 || a.appendChild(o) : (n = this.politeLog) === null || n === void 0 || n.appendChild(o), e !== "" && setTimeout(() => {
      o.remove();
    }, d);
  }
  clear(e) {
    this.node && ((!e || e === "assertive") && this.assertiveLog && (this.assertiveLog.innerHTML = ""), (!e || e === "polite") && this.politeLog && (this.politeLog.innerHTML = ""));
  }
  constructor() {
    this.node = null, this.assertiveLog = null, this.politeLog = null, typeof document < "u" && (this.node = document.createElement("div"), this.node.dataset.liveAnnouncer = "true", Object.assign(this.node.style, {
      border: 0,
      clip: "rect(0 0 0 0)",
      clipPath: "inset(50%)",
      height: "1px",
      margin: "-1px",
      overflow: "hidden",
      padding: 0,
      position: "absolute",
      width: "1px",
      whiteSpace: "nowrap"
    }), this.assertiveLog = this.createLog("assertive"), this.node.appendChild(this.assertiveLog), this.politeLog = this.createLog("polite"), this.node.appendChild(this.politeLog), document.body.prepend(this.node));
  }
}
function H(t, e) {
  let { elementType: i = "button", isDisabled: d, onPress: a, onPressStart: n, onPressEnd: o, onPressUp: p, onPressChange: l, preventFocusOnPress: $, allowFocusWhenDisabled: g, onClick: h, href: v, target: m, rel: u, type: c = "button" } = t, s;
  i === "button" ? s = {
    type: c,
    disabled: d,
    form: t.form,
    formAction: t.formAction,
    formEncType: t.formEncType,
    formMethod: t.formMethod,
    formNoValidate: t.formNoValidate,
    formTarget: t.formTarget,
    name: t.name,
    value: t.value
  } : s = {
    role: "button",
    href: i === "a" && !d ? v : void 0,
    target: i === "a" ? m : void 0,
    type: i === "input" ? c : void 0,
    disabled: i === "input" ? d : void 0,
    "aria-disabled": !d || i === "input" ? void 0 : d,
    rel: i === "a" ? u : void 0
  };
  let { pressProps: b, isPressed: P } = w({
    onPressStart: n,
    onPressEnd: o,
    onPressChange: l,
    onPress: a,
    onPressUp: p,
    onClick: h,
    isDisabled: d,
    preventFocusOnPress: $,
    ref: e
  }), { focusableProps: f } = M(t, e);
  g && (f.tabIndex = d ? -1 : f.tabIndex);
  let A = x(f, b, C(t, {
    labelable: !0
  }));
  return {
    isPressed: P,
    buttonProps: x(s, A, {
      "aria-haspopup": t["aria-haspopup"],
      "aria-expanded": t["aria-expanded"],
      "aria-controls": t["aria-controls"],
      "aria-pressed": t["aria-pressed"],
      "aria-current": t["aria-current"],
      "aria-disabled": t["aria-disabled"]
    })
  };
}
const U = /* @__PURE__ */ N({}), X = /* @__PURE__ */ I(function(e, i) {
  [e, i] = _(e, i, U);
  let d = e, { isPending: a } = d, { buttonProps: n, isPressed: o } = H(e, i);
  n = j(n, a);
  let { focusProps: p, isFocused: l, isFocusVisible: $ } = R(e), { hoverProps: g, isHovered: h } = S({
    ...e,
    isDisabled: e.isDisabled || a
  }), v = {
    isHovered: h,
    isPressed: (d.isPressed || o) && !a,
    isFocused: l,
    isFocusVisible: $,
    isDisabled: e.isDisabled || !1,
    isPending: a ?? !1
  }, m = D({
    ...e,
    values: v,
    defaultClassName: "react-aria-Button"
  }), u = y(n.id), c = y(), s = n["aria-labelledby"];
  a && (s ? s = `${s} ${c}` : n["aria-label"] && (s = `${u} ${c}`));
  let b = O(a);
  V(() => {
    let f = {
      "aria-labelledby": s || u
    };
    (!b.current && l && a || b.current && l && !a) && E(f, "assertive"), b.current = a;
  }, [
    a,
    l,
    s,
    u
  ]);
  let P = C(e, {
    global: !0
  });
  return delete P.onClick, /* @__PURE__ */ L.createElement("button", {
    ...x(P, m, n, p, g),
    // When the button is in a pending state, we want to stop implicit form submission (ie. when the user presses enter on a text input).
    // We do this by changing the button's type to button.
    type: n.type === "submit" && a ? "button" : n.type,
    id: u,
    ref: i,
    "aria-labelledby": s,
    slot: e.slot || void 0,
    "aria-disabled": a ? "true" : n["aria-disabled"],
    "data-disabled": e.isDisabled || void 0,
    "data-pressed": v.isPressed || void 0,
    "data-hovered": h || void 0,
    "data-focused": l || void 0,
    "data-pending": a || void 0,
    "data-focus-visible": $ || void 0
  }, /* @__PURE__ */ L.createElement(F.Provider, {
    value: {
      id: c
    }
  }, m.children));
});
function j(t, e) {
  if (e) {
    for (const i in t) i.startsWith("on") && !(i.includes("Focus") || i.includes("Blur")) && (t[i] = void 0);
    t.href = void 0, t.target = void 0;
  }
  return t;
}
export {
  X as $,
  U as a
};
