// @angi/one-design-system/dist/components/es/_chunks/CheckboxItem--3cltbnE.js
import { jsxs as j, jsx as I } from "react/jsx-runtime";
import * as R from "react";
import { useEffect as D, useMemo as F, useState as L, useRef as k, forwardRef as A, useImperativeHandle as H } from "react";
import { a as T, u as G } from "./useCheckboxGroupContext-R2g3qemh.js";
import { c as E } from "./createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import { c as K } from "./clsx-OuTLNxxd.js";
import { b as S } from "./utils-CjtkL41C.js";
import { a as O, g as _, h as X } from "./Form-sh8c7lqF.js";
import { $ as Z, a as J } from "./useFormValidation-CgQ_8wCr.js";
import { $ as V } from "./usePress-D9Qp6Amx.js";
import { $ as Q } from "./filterDOMProps-Cxz9cAAv.js";
import { a as W } from "./useFocusable-CrxUcjB4.js";
import { $ as Y } from "./useControlledState-3TGZUXux.js";
import { $ as ee } from "./useFocusRing-CdwnmTWa.js";
import '../assets/CheckboxItem.css';function ae(e, a, i) {
  let { isDisabled: l = !1, isReadOnly: d = !1, value: c, name: u, form: t, children: n, "aria-label": v, "aria-labelledby": $, validationState: m = "valid", isInvalid: r, onPressStart: b, onPressEnd: P, onPressChange: o, onPress: f, onPressUp: s, onClick: p } = e, g = (h) => {
    h.stopPropagation(), a.setSelected(h.target.checked);
  }, C = n != null, y = v != null || $ != null;
  !C && !y && process.env.NODE_ENV !== "production" && console.warn("If you do not provide children, you must specify an aria-label for accessibility");
  let { pressProps: N, isPressed: B } = V({
    onPressStart: b,
    onPressEnd: P,
    onPressChange: o,
    onPress: f,
    onPressUp: s,
    onClick: p,
    isDisabled: l
  }), { pressProps: q, isPressed: w } = V({
    onPressStart: b,
    onPressEnd: P,
    onPressChange: o,
    onPressUp: s,
    onClick: p,
    onPress(h) {
      var x;
      f?.(h), a.toggle(), (x = i.current) === null || x === void 0 || x.focus();
    },
    isDisabled: l || d
  }), { focusableProps: M } = W(e, i), z = S(N, M), U = Q(e, {
    labelable: !0
  });
  return Z(i, a.defaultSelected, a.setSelected), {
    labelProps: S(q, {
      onClick: (h) => h.preventDefault()
    }),
    inputProps: S(U, {
      "aria-invalid": r || m === "invalid" || void 0,
      "aria-errormessage": e["aria-errormessage"],
      "aria-controls": e["aria-controls"],
      "aria-readonly": d || void 0,
      onChange: g,
      disabled: l,
      ...c == null ? {} : {
        value: c
      },
      name: u,
      form: t,
      type: "checkbox",
      ...z
    }),
    isSelected: a.isSelected,
    isPressed: B || w,
    isDisabled: l,
    isReadOnly: d,
    isInvalid: r || m === "invalid"
  };
}
function ie(e, a, i) {
  let l = O({
    ...e,
    value: a.isSelected
  }), { isInvalid: d, validationErrors: c, validationDetails: u } = l.displayValidation, { labelProps: t, inputProps: n, isSelected: v, isPressed: $, isDisabled: m, isReadOnly: r } = ae({
    ...e,
    isInvalid: d
  }, a, i);
  J(e, l, i);
  let { isIndeterminate: b, isRequired: P, validationBehavior: o = "aria" } = e;
  D(() => {
    i.current && (i.current.indeterminate = !!b);
  });
  let { pressProps: f } = V({
    isDisabled: m || r,
    onPress() {
      let { [_]: s } = e, { commitValidation: p } = s || l;
      p();
    }
  });
  return {
    labelProps: S(t, f, F(() => ({
      // Prevent label from being focused when mouse down on it.
      // Note, this does not prevent the input from being focused in the `click` event.
      onMouseDown: (s) => s.preventDefault()
    }), [])),
    inputProps: {
      ...n,
      checked: v,
      "aria-required": P && o === "aria" || void 0,
      required: P && o === "native"
    },
    isSelected: v,
    isPressed: $,
    isDisabled: m,
    isReadOnly: r,
    isInvalid: d,
    validationErrors: c,
    validationDetails: u
  };
}
function le(e = {}) {
  let { isReadOnly: a } = e, [i, l] = Y(e.isSelected, e.defaultSelected || !1, e.onChange), [d] = L(i);
  function c(n) {
    a || l(n);
  }
  function u() {
    a || l(!i);
  }
  var t;
  return {
    isSelected: i,
    defaultSelected: (t = e.defaultSelected) !== null && t !== void 0 ? t : d,
    setSelected: c,
    toggle: u
  };
}
function se(e, a, i) {
  const l = le({
    isReadOnly: e.isReadOnly || a.isReadOnly,
    isSelected: a.isSelected(e.value),
    defaultSelected: a.defaultValue.includes(e.value),
    onChange(s) {
      s ? a.addValue(e.value) : a.removeValue(e.value), e.onChange && e.onChange(s);
    }
  });
  let { name: d, form: c, descriptionId: u, errorMessageId: t, validationBehavior: n } = T.get(a);
  var v;
  n = (v = e.validationBehavior) !== null && v !== void 0 ? v : n;
  let { realtimeValidation: $ } = O({
    ...e,
    value: l.isSelected,
    // Server validation is handled at the group level.
    name: void 0,
    validationBehavior: "aria"
  }), m = k(X), r = () => {
    a.setInvalid(e.value, $.isInvalid ? $ : m.current);
  };
  D(r);
  let b = a.realtimeValidation.isInvalid ? a.realtimeValidation : $, P = n === "native" ? a.displayValidation : b;
  var o;
  let f = ie({
    ...e,
    isReadOnly: e.isReadOnly || a.isReadOnly,
    isDisabled: e.isDisabled || a.isDisabled,
    name: e.name || d,
    form: e.form || c,
    isRequired: (o = e.isRequired) !== null && o !== void 0 ? o : a.isRequired,
    validationBehavior: n,
    [_]: {
      realtimeValidation: b,
      displayValidation: P,
      resetValidation: a.resetValidation,
      commitValidation: a.commitValidation,
      updateValidation(s) {
        m.current = s, r();
      }
    }
  }, l, i);
  return {
    ...f,
    inputProps: {
      ...f.inputProps,
      "aria-describedby": [
        e["aria-describedby"],
        a.isInvalid ? t : null,
        u
      ].filter(Boolean).join(" ") || void 0
    }
  };
}
var de = "e13vb60", ne = E({ defaultClassName: "e13vb62 e13vb61", variantClassNames: { size: { sm: "e13vb63", md: "e13vb64" } }, defaultVariants: { size: "md" }, compoundVariants: [] }), te = E({ defaultClassName: "e13vb65", variantClassNames: { size: { sm: "e13vb66", md: "e13vb67" } }, defaultVariants: { size: "md" }, compoundVariants: [] });
const xe = ["sm", "md"], re = "md", oe = (e) => /* @__PURE__ */ R.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 448 512", ...e }, /* @__PURE__ */ R.createElement("path", { fill: "currentColor", d: "M434.8 70.1c14.3 10.4 17.5 30.4 7.1 44.7l-256 352c-5.5 7.6-14 12.3-23.4 13.1s-18.5-2.7-25.1-9.3l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l101.5 101.5 234-321.7c10.4-14.3 30.4-17.5 44.7-7.1z" })), ce = A(
  ({
    size: e = re,
    disabled: a = !1,
    value: i,
    className: l,
    id: d,
    name: c,
    "aria-label": u,
    "aria-labelledby": t,
    "aria-describedby": n,
    hideFocusRing: v = !1,
    ...$
  }, m) => {
    const r = k(null);
    H(
      m,
      () => r.current
    );
    const b = G(), { inputProps: P, isSelected: o, isDisabled: f } = se(
      {
        value: i,
        isDisabled: a,
        ...u !== void 0 && { "aria-label": u },
        ...t !== void 0 && {
          "aria-labelledby": t
        },
        ...n !== void 0 && {
          "aria-describedby": n
        },
        ...d !== void 0 && { id: d },
        ...c !== void 0 && { name: c }
      },
      b,
      r
    ), { isFocusVisible: s, focusProps: p } = ee(), g = b.isInvalid, C = K(
      ne({
        size: e
      }),
      l
    ), y = te({
      size: e
    });
    return /* @__PURE__ */ j(
      "span",
      {
        className: C,
        "data-checked": o || void 0,
        "data-disabled": f || void 0,
        "data-invalid": g || void 0,
        "data-focus-visible": !v && s || void 0,
        ...$,
        children: [
          /* @__PURE__ */ I(
            "input",
            {
              ...S(P, p),
              ref: r,
              className: de
            }
          ),
          o && /* @__PURE__ */ I(oe, { className: y, "aria-hidden": "true" })
        ]
      }
    );
  }
);
ce.displayName = "CheckboxItem";
export {
  ce as C,
  re as D,
  xe as a
};
