// @angi/one-design-system/dist/components/es/_chunks/Checkbox-DpPIhntn.js
import { jsxs as h, jsx as r } from "react/jsx-runtime";
import { forwardRef as B, useId as D } from "react";
import { c as N } from "./clsx-OuTLNxxd.js";
import { u as _ } from "./useCheckboxGroupContext-R2g3qemh.js";
import { L as H } from "./Label-BqoOfVfQ.js";
import { H as $ } from "./Hint-CIDmUonJ.js";
import { C as w } from "./CheckboxItem--3cltbnE.js";
import '../assets/Checkbox.css';var G = "_1hxosgt0", L = "_1hxosgt1", q = "_1hxosgt2";
const x = (...o) => {
  const e = o.filter(Boolean);
  return e.length > 0 ? e.join(" ") : void 0;
}, A = B(
  ({
    // Custom props
    label: o,
    hint: e,
    className: f,
    // CheckboxItem props
    id: p,
    value: v,
    size: i,
    name: s,
    hideFocusRing: t,
    "aria-label": d,
    "aria-describedby": C,
    "aria-labelledby": y,
    disabled: l = !1,
    // Rest are label element props (forwarded from Slot)
    ...I
  }, u) => {
    const g = D(), a = p ?? g, n = `${a}-label`, c = e ? `${a}-hint` : void 0, b = x(C, c), m = x(y, n), k = _(), j = l || k.isDisabled;
    return /* @__PURE__ */ h(
      "label",
      {
        ref: u,
        ...I,
        className: N(
          G,
          j && L,
          f
        ),
        children: [
          /* @__PURE__ */ r(
            w,
            {
              id: a,
              value: v,
              disabled: l,
              ...i !== void 0 && { size: i },
              ...s !== void 0 && { name: s },
              ...t !== void 0 && { hideFocusRing: t },
              ...d !== void 0 && { "aria-label": d },
              ...m !== void 0 && { "aria-labelledby": m },
              ...b !== void 0 && {
                "aria-describedby": b
              }
            }
          ),
          /* @__PURE__ */ h("span", { className: q, children: [
            /* @__PURE__ */ r(H, { variant: "normal", ...o, id: n, asChild: !0, children: /* @__PURE__ */ r("span", { children: o.children }) }),
            e && /* @__PURE__ */ r($, { ...e, id: c })
          ] })
        ]
      }
    );
  }
);
A.displayName = "Checkbox";
export {
  A as C
};
