// @angi/one-design-system/dist/components/es/_chunks/context-qNvgpQEb.js
import d, { useState as i, useEffect as $, useContext as u } from "react";
import { e as s } from "./utils-CjtkL41C.js";
const b = /* @__PURE__ */ new Set([
  "Arab",
  "Syrc",
  "Samr",
  "Mand",
  "Thaa",
  "Mend",
  "Nkoo",
  "Adlm",
  "Rohg",
  "Hebr"
]), p = /* @__PURE__ */ new Set([
  "ae",
  "ar",
  "arc",
  "bcc",
  "bqi",
  "ckb",
  "dv",
  "fa",
  "glk",
  "he",
  "ku",
  "mzn",
  "nqo",
  "pnb",
  "ps",
  "sd",
  "ug",
  "ur",
  "yi"
]);
function g(e) {
  if (Intl.Locale) {
    let a = new Intl.Locale(e).maximize(), o = typeof a.getTextInfo == "function" ? a.getTextInfo() : a.textInfo;
    if (o) return o.direction === "rtl";
    if (a.script) return b.has(a.script);
  }
  let n = e.split("-")[0];
  return p.has(n);
}
const c = Symbol.for("react-aria.i18n.locale");
function l() {
  let e = typeof window < "u" && window[c] || typeof navigator < "u" && (navigator.language || navigator.userLanguage) || "en-US";
  try {
    Intl.DateTimeFormat.supportedLocalesOf([
      e
    ]);
  } catch {
    e = "en-US";
  }
  return {
    locale: e,
    direction: g(e) ? "rtl" : "ltr"
  };
}
let r = l(), t = /* @__PURE__ */ new Set();
function f() {
  r = l();
  for (let e of t) e(r);
}
function w() {
  let e = s(), [n, a] = i(r);
  return $(() => (t.size === 0 && window.addEventListener("languagechange", f), t.add(a), () => {
    t.delete(a), t.size === 0 && window.removeEventListener("languagechange", f);
  }), []), e ? {
    locale: typeof window < "u" && window[c] || "en-US",
    direction: "ltr"
  } : n;
}
const x = /* @__PURE__ */ d.createContext(null);
function v() {
  let e = w();
  return u(x) || e;
}
export {
  v as $
};
