// @angi/one-design-system/dist/components/es/_chunks/focusSafely-OgqIqadr.js
import { a as b } from "./useFocusVisible-ROQSYc3m.js";
import { a as $, f as d } from "./useFocusWithin-CAx2NHfT.js";
import { $ as c } from "./focusWithoutScrolling-CDw3JYiq.js";
let r = /* @__PURE__ */ new Map(), i = /* @__PURE__ */ new Set();
function f() {
  if (typeof window > "u") return;
  function t(e) {
    return "propertyName" in e;
  }
  let a = (e) => {
    if (!t(e) || !e.target) return;
    let n = r.get(e.target);
    n || (n = /* @__PURE__ */ new Set(), r.set(e.target, n), e.target.addEventListener("transitioncancel", o, {
      once: !0
    })), n.add(e.propertyName);
  }, o = (e) => {
    if (!t(e) || !e.target) return;
    let n = r.get(e.target);
    if (n && (n.delete(e.propertyName), n.size === 0 && (e.target.removeEventListener("transitioncancel", o), r.delete(e.target)), r.size === 0)) {
      for (let s of i) s();
      i.clear();
    }
  };
  document.body.addEventListener("transitionrun", a), document.body.addEventListener("transitionend", o);
}
typeof document < "u" && (document.readyState !== "loading" ? f() : document.addEventListener("DOMContentLoaded", f));
function l() {
  for (const [t] of r)
    "isConnected" in t && !t.isConnected && r.delete(t);
}
function u(t) {
  requestAnimationFrame(() => {
    l(), r.size === 0 ? t() : i.add(t);
  });
}
function g(t) {
  const a = $(t);
  if (b() === "virtual") {
    let o = d(a);
    u(() => {
      const e = d(a);
      (e === o || e === a.body) && t.isConnected && c(t);
    });
  } else c(t);
}
export {
  g as $,
  u as a
};
