// @angi/one-design-system/dist/components/es/_chunks/FocusScope-DlOb-a2X.js
import { c as J, b as D, d as H, e as B, a as p, f as w, g as _ } from "./useFocusWithin-CAx2NHfT.js";
import { c as E } from "./utils-CjtkL41C.js";
import { a as G, b as V } from "./openLink-EIDcTMUl.js";
import { a as j } from "./useFocusVisible-ROQSYc3m.js";
import { $ as Q } from "./focusSafely-OgqIqadr.js";
import R, { useRef as F, useContext as X, useMemo as L, useEffect as z } from "react";
class Y {
  get currentNode() {
    return this._currentNode;
  }
  set currentNode(e) {
    if (!D(this.root, e)) throw new Error("Cannot set currentNode to a node that is not contained by the root node.");
    const t = [];
    let o = e, n = e;
    for (this._currentNode = e; o && o !== this.root; ) if (o.nodeType === Node.DOCUMENT_FRAGMENT_NODE) {
      const l = o, a = this._doc.createTreeWalker(l, this.whatToShow, {
        acceptNode: this._acceptNode
      });
      t.push(a), a.currentNode = n, this._currentSetFor.add(a), o = n = l.host;
    } else o = o.parentNode;
    const i = this._doc.createTreeWalker(this.root, this.whatToShow, {
      acceptNode: this._acceptNode
    });
    t.push(i), i.currentNode = n, this._currentSetFor.add(i), this._walkerStack = t;
  }
  get doc() {
    return this._doc;
  }
  firstChild() {
    let e = this.currentNode, t = this.nextNode();
    return D(e, t) ? (t && (this.currentNode = t), t) : (this.currentNode = e, null);
  }
  lastChild() {
    let t = this._walkerStack[0].lastChild();
    return t && (this.currentNode = t), t;
  }
  nextNode() {
    const e = this._walkerStack[0].nextNode();
    if (e) {
      if (e.shadowRoot) {
        var t;
        let n;
        if (typeof this.filter == "function" ? n = this.filter(e) : !((t = this.filter) === null || t === void 0) && t.acceptNode && (n = this.filter.acceptNode(e)), n === NodeFilter.FILTER_ACCEPT)
          return this.currentNode = e, e;
        let i = this.nextNode();
        return i && (this.currentNode = i), i;
      }
      return e && (this.currentNode = e), e;
    } else if (this._walkerStack.length > 1) {
      this._walkerStack.shift();
      let o = this.nextNode();
      return o && (this.currentNode = o), o;
    } else return null;
  }
  previousNode() {
    const e = this._walkerStack[0];
    if (e.currentNode === e.root) {
      if (this._currentSetFor.has(e))
        if (this._currentSetFor.delete(e), this._walkerStack.length > 1) {
          this._walkerStack.shift();
          let n = this.previousNode();
          return n && (this.currentNode = n), n;
        } else return null;
      return null;
    }
    const t = e.previousNode();
    if (t) {
      if (t.shadowRoot) {
        var o;
        let i;
        if (typeof this.filter == "function" ? i = this.filter(t) : !((o = this.filter) === null || o === void 0) && o.acceptNode && (i = this.filter.acceptNode(t)), i === NodeFilter.FILTER_ACCEPT)
          return t && (this.currentNode = t), t;
        let l = this.lastChild();
        return l && (this.currentNode = l), l;
      }
      return t && (this.currentNode = t), t;
    } else if (this._walkerStack.length > 1) {
      this._walkerStack.shift();
      let n = this.previousNode();
      return n && (this.currentNode = n), n;
    } else return null;
  }
  /**
   * @deprecated
   */
  nextSibling() {
    return null;
  }
  /**
   * @deprecated
   */
  previousSibling() {
    return null;
  }
  /**
   * @deprecated
   */
  parentNode() {
    return null;
  }
  constructor(e, t, o, n) {
    this._walkerStack = [], this._currentSetFor = /* @__PURE__ */ new Set(), this._acceptNode = (l) => {
      if (l.nodeType === Node.ELEMENT_NODE) {
        const h = l.shadowRoot;
        if (h) {
          const u = this._doc.createTreeWalker(h, this.whatToShow, {
            acceptNode: this._acceptNode
          });
          return this._walkerStack.unshift(u), NodeFilter.FILTER_ACCEPT;
        } else {
          var a;
          if (typeof this.filter == "function") return this.filter(l);
          if (!((a = this.filter) === null || a === void 0) && a.acceptNode) return this.filter.acceptNode(l);
          if (this.filter === null) return NodeFilter.FILTER_ACCEPT;
        }
      }
      return NodeFilter.FILTER_SKIP;
    }, this._doc = e, this.root = t, this.filter = n ?? null, this.whatToShow = o ?? NodeFilter.SHOW_ALL, this._currentNode = t, this._walkerStack.unshift(e.createTreeWalker(t, o, this._acceptNode));
    const i = t.shadowRoot;
    if (i) {
      const l = this._doc.createTreeWalker(i, this.whatToShow, {
        acceptNode: this._acceptNode
      });
      this._walkerStack.unshift(l);
    }
  }
}
function Z(r, e, t, o) {
  return J() ? new Y(r, e, t, o) : r.createTreeWalker(e, t, o);
}
const P = /* @__PURE__ */ R.createContext(null), I = "react-aria-focus-scope-restore";
let f = null;
function fe(r) {
  let { children: e, contain: t, restoreFocus: o, autoFocus: n } = r, i = F(null), l = F(null), a = F([]), { parentNode: h } = X(P) || {}, u = L(() => new A({
    scopeRef: a
  }), [
    a
  ]);
  E(() => {
    let c = h || b.root;
    if (b.getTreeNode(c.scopeRef) && f && !y(f, c.scopeRef)) {
      let s = b.getTreeNode(f);
      s && (c = s);
    }
    c.addChild(u), b.addNode(u);
  }, [
    u,
    h
  ]), E(() => {
    let c = b.getTreeNode(a);
    c && (c.contain = !!t);
  }, [
    t
  ]), E(() => {
    var c;
    let s = (c = i.current) === null || c === void 0 ? void 0 : c.nextSibling, N = [], S = (C) => C.stopPropagation();
    for (; s && s !== l.current; )
      N.push(s), s.addEventListener(I, S), s = s.nextSibling;
    return a.current = N, () => {
      for (let C of N) C.removeEventListener(I, S);
    };
  }, [
    e
  ]), oe(a, o, t), re(a, t), ie(a, o, t), ne(a, n), z(() => {
    const c = w(p(a.current ? a.current[0] : void 0));
    let s = null;
    if ($(c, a.current)) {
      for (let N of b.traverse()) N.scopeRef && $(c, N.scopeRef.current) && (s = N);
      s === b.getTreeNode(a) && (f = s.scopeRef);
    }
  }, [
    a
  ]), E(() => () => {
    var c, s, N;
    let S = (N = (s = b.getTreeNode(a)) === null || s === void 0 || (c = s.parent) === null || c === void 0 ? void 0 : c.scopeRef) !== null && N !== void 0 ? N : null;
    (a === f || y(a, f)) && (!S || b.getTreeNode(S)) && (f = S), b.removeTreeNode(a);
  }, [
    a
  ]);
  let v = L(() => ee(a), []), d = L(() => ({
    focusManager: v,
    parentNode: u
  }), [
    u,
    v
  ]);
  return /* @__PURE__ */ R.createElement(P.Provider, {
    value: d
  }, /* @__PURE__ */ R.createElement("span", {
    "data-focus-scope-start": !0,
    hidden: !0,
    ref: i
  }), e, /* @__PURE__ */ R.createElement("span", {
    "data-focus-scope-end": !0,
    hidden: !0,
    ref: l
  }));
}
function ee(r) {
  return {
    focusNext(e = {}) {
      let t = r.current, { from: o, tabbable: n, wrap: i, accept: l } = e;
      var a;
      let h = o || w(p((a = t[0]) !== null && a !== void 0 ? a : void 0)), u = t[0].previousElementSibling, v = g(t), d = k(v, {
        tabbable: n,
        accept: l
      }, t);
      d.currentNode = $(h, t) ? h : u;
      let c = d.nextNode();
      return !c && i && (d.currentNode = u, c = d.nextNode()), c && T(c, !0), c;
    },
    focusPrevious(e = {}) {
      let t = r.current, { from: o, tabbable: n, wrap: i, accept: l } = e;
      var a;
      let h = o || w(p((a = t[0]) !== null && a !== void 0 ? a : void 0)), u = t[t.length - 1].nextElementSibling, v = g(t), d = k(v, {
        tabbable: n,
        accept: l
      }, t);
      d.currentNode = $(h, t) ? h : u;
      let c = d.previousNode();
      return !c && i && (d.currentNode = u, c = d.previousNode()), c && T(c, !0), c;
    },
    focusFirst(e = {}) {
      let t = r.current, { tabbable: o, accept: n } = e, i = g(t), l = k(i, {
        tabbable: o,
        accept: n
      }, t);
      l.currentNode = t[0].previousElementSibling;
      let a = l.nextNode();
      return a && T(a, !0), a;
    },
    focusLast(e = {}) {
      let t = r.current, { tabbable: o, accept: n } = e, i = g(t), l = k(i, {
        tabbable: o,
        accept: n
      }, t);
      l.currentNode = t[t.length - 1].nextElementSibling;
      let a = l.previousNode();
      return a && T(a, !0), a;
    }
  };
}
function g(r) {
  return r[0].parentElement;
}
function x(r) {
  let e = b.getTreeNode(f);
  for (; e && e.scopeRef !== r; ) {
    if (e.contain) return !1;
    e = e.parent;
  }
  return !0;
}
function te(r) {
  if (r.checked) return !0;
  let e = [];
  if (!r.form) e = [
    ...p(r).querySelectorAll(`input[type="radio"][name="${CSS.escape(r.name)}"]`)
  ].filter((i) => !i.form);
  else {
    var t, o;
    let i = (o = r.form) === null || o === void 0 || (t = o.elements) === null || t === void 0 ? void 0 : t.namedItem(r.name);
    e = [
      ...i ?? []
    ];
  }
  return e ? !e.some((i) => i.checked) : !1;
}
function re(r, e) {
  let t = F(void 0), o = F(void 0);
  E(() => {
    let n = r.current;
    if (!e) {
      o.current && (cancelAnimationFrame(o.current), o.current = void 0);
      return;
    }
    const i = p(n ? n[0] : void 0);
    let l = (u) => {
      if (u.key !== "Tab" || u.altKey || u.ctrlKey || u.metaKey || !x(r) || u.isComposing) return;
      let v = w(i), d = r.current;
      if (!d || !$(v, d)) return;
      let c = g(d), s = k(c, {
        tabbable: !0
      }, d);
      if (!v) return;
      s.currentNode = v;
      let N = u.shiftKey ? s.previousNode() : s.nextNode();
      N || (s.currentNode = u.shiftKey ? d[d.length - 1].nextElementSibling : d[0].previousElementSibling, N = u.shiftKey ? s.previousNode() : s.nextNode()), u.preventDefault(), N && T(N, !0);
    }, a = (u) => {
      (!f || y(f, r)) && $(_(u), r.current) ? (f = r, t.current = _(u)) : x(r) && !m(_(u), r) ? t.current ? t.current.focus() : f && f.current && M(f.current) : x(r) && (t.current = _(u));
    }, h = (u) => {
      o.current && cancelAnimationFrame(o.current), o.current = requestAnimationFrame(() => {
        let v = j(), d = (v === "virtual" || v === null) && G() && V(), c = w(i);
        if (!d && c && x(r) && !m(c, r)) {
          f = r;
          let N = _(u);
          if (N && N.isConnected) {
            var s;
            t.current = N, (s = t.current) === null || s === void 0 || s.focus();
          } else f.current && M(f.current);
        }
      });
    };
    return i.addEventListener("keydown", l, !1), i.addEventListener("focusin", a, !1), n?.forEach((u) => u.addEventListener("focusin", a, !1)), n?.forEach((u) => u.addEventListener("focusout", h, !1)), () => {
      i.removeEventListener("keydown", l, !1), i.removeEventListener("focusin", a, !1), n?.forEach((u) => u.removeEventListener("focusin", a, !1)), n?.forEach((u) => u.removeEventListener("focusout", h, !1));
    };
  }, [
    r,
    e
  ]), E(() => () => {
    o.current && cancelAnimationFrame(o.current);
  }, [
    o
  ]);
}
function U(r) {
  return m(r);
}
function $(r, e) {
  return !r || !e ? !1 : e.some((t) => t.contains(r));
}
function m(r, e = null) {
  if (r instanceof Element && r.closest("[data-react-aria-top-layer]")) return !0;
  for (let { scopeRef: t } of b.traverse(b.getTreeNode(e)))
    if (t && $(r, t.current)) return !0;
  return !1;
}
function he(r) {
  return m(r, f);
}
function y(r, e) {
  var t;
  let o = (t = b.getTreeNode(e)) === null || t === void 0 ? void 0 : t.parent;
  for (; o; ) {
    if (o.scopeRef === r) return !0;
    o = o.parent;
  }
  return !1;
}
function T(r, e = !1) {
  if (r != null && !e) try {
    Q(r);
  } catch {
  }
  else if (r != null) try {
    r.focus();
  } catch {
  }
}
function q(r, e = !0) {
  let t = r[0].previousElementSibling, o = g(r), n = k(o, {
    tabbable: e
  }, r);
  n.currentNode = t;
  let i = n.nextNode();
  return e && !i && (o = g(r), n = k(o, {
    tabbable: !1
  }, r), n.currentNode = t, i = n.nextNode()), i;
}
function M(r, e = !0) {
  T(q(r, e));
}
function ne(r, e) {
  const t = R.useRef(e);
  z(() => {
    if (t.current) {
      f = r;
      const o = p(r.current ? r.current[0] : void 0);
      !$(w(o), f.current) && r.current && M(r.current);
    }
    t.current = !1;
  }, [
    r
  ]);
}
function oe(r, e, t) {
  E(() => {
    if (e || t) return;
    let o = r.current;
    const n = p(o ? o[0] : void 0);
    let i = (l) => {
      let a = _(l);
      $(a, r.current) ? f = r : U(a) || (f = null);
    };
    return n.addEventListener("focusin", i, !1), o?.forEach((l) => l.addEventListener("focusin", i, !1)), () => {
      n.removeEventListener("focusin", i, !1), o?.forEach((l) => l.removeEventListener("focusin", i, !1));
    };
  }, [
    r,
    e,
    t
  ]);
}
function W(r) {
  let e = b.getTreeNode(f);
  for (; e && e.scopeRef !== r; ) {
    if (e.nodeToRestore) return !1;
    e = e.parent;
  }
  return e?.scopeRef === r;
}
function ie(r, e, t) {
  const o = F(typeof document < "u" ? w(p(r.current ? r.current[0] : void 0)) : null);
  E(() => {
    let n = r.current;
    const i = p(n ? n[0] : void 0);
    if (!e || t) return;
    let l = () => {
      (!f || y(f, r)) && $(w(i), r.current) && (f = r);
    };
    return i.addEventListener("focusin", l, !1), n?.forEach((a) => a.addEventListener("focusin", l, !1)), () => {
      i.removeEventListener("focusin", l, !1), n?.forEach((a) => a.removeEventListener("focusin", l, !1));
    };
  }, [
    r,
    t
  ]), E(() => {
    const n = p(r.current ? r.current[0] : void 0);
    if (!e) return;
    let i = (l) => {
      if (l.key !== "Tab" || l.altKey || l.ctrlKey || l.metaKey || !x(r) || l.isComposing) return;
      let a = n.activeElement;
      if (!m(a, r) || !W(r)) return;
      let h = b.getTreeNode(r);
      if (!h) return;
      let u = h.nodeToRestore, v = k(n.body, {
        tabbable: !0
      });
      v.currentNode = a;
      let d = l.shiftKey ? v.previousNode() : v.nextNode();
      if ((!u || !u.isConnected || u === n.body) && (u = void 0, h.nodeToRestore = void 0), (!d || !m(d, r)) && u) {
        v.currentNode = u;
        do
          d = l.shiftKey ? v.previousNode() : v.nextNode();
        while (m(d, r));
        l.preventDefault(), l.stopPropagation(), d ? T(d, !0) : U(u) ? T(u, !0) : a.blur();
      }
    };
    return t || n.addEventListener("keydown", i, !0), () => {
      t || n.removeEventListener("keydown", i, !0);
    };
  }, [
    r,
    e,
    t
  ]), E(() => {
    const n = p(r.current ? r.current[0] : void 0);
    if (!e) return;
    let i = b.getTreeNode(r);
    if (i) {
      var l;
      return i.nodeToRestore = (l = o.current) !== null && l !== void 0 ? l : void 0, () => {
        let a = b.getTreeNode(r);
        if (!a) return;
        let h = a.nodeToRestore, u = w(n);
        if (e && h && (u && m(u, r) || u === n.body && W(r))) {
          let v = b.clone();
          requestAnimationFrame(() => {
            if (n.activeElement === n.body) {
              let d = v.getTreeNode(r);
              for (; d; ) {
                if (d.nodeToRestore && d.nodeToRestore.isConnected) {
                  O(d.nodeToRestore);
                  return;
                }
                d = d.parent;
              }
              for (d = v.getTreeNode(r); d; ) {
                if (d.scopeRef && d.scopeRef.current && b.getTreeNode(d.scopeRef)) {
                  let c = q(d.scopeRef.current, !0);
                  O(c);
                  return;
                }
                d = d.parent;
              }
            }
          });
        }
      };
    }
  }, [
    r,
    e
  ]);
}
function O(r) {
  r.dispatchEvent(new CustomEvent(I, {
    bubbles: !0,
    cancelable: !0
  })) && T(r);
}
function k(r, e, t) {
  let o = e?.tabbable ? H : B, n = r?.nodeType === Node.ELEMENT_NODE ? r : null, i = p(n), l = Z(i, r || i, NodeFilter.SHOW_ELEMENT, {
    acceptNode(a) {
      var h;
      return !(e == null || (h = e.from) === null || h === void 0) && h.contains(a) || e?.tabbable && a.tagName === "INPUT" && a.getAttribute("type") === "radio" && (!te(a) || l.currentNode.tagName === "INPUT" && l.currentNode.type === "radio" && l.currentNode.name === a.name) ? NodeFilter.FILTER_REJECT : o(a) && (!t || $(a, t)) && (!e?.accept || e.accept(a)) ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_SKIP;
    }
  });
  return e?.from && (l.currentNode = e.from), l;
}
class K {
  get size() {
    return this.fastMap.size;
  }
  getTreeNode(e) {
    return this.fastMap.get(e);
  }
  addTreeNode(e, t, o) {
    let n = this.fastMap.get(t ?? null);
    if (!n) return;
    let i = new A({
      scopeRef: e
    });
    n.addChild(i), i.parent = n, this.fastMap.set(e, i), o && (i.nodeToRestore = o);
  }
  addNode(e) {
    this.fastMap.set(e.scopeRef, e);
  }
  removeTreeNode(e) {
    if (e === null) return;
    let t = this.fastMap.get(e);
    if (!t) return;
    let o = t.parent;
    for (let i of this.traverse()) i !== t && t.nodeToRestore && i.nodeToRestore && t.scopeRef && t.scopeRef.current && $(i.nodeToRestore, t.scopeRef.current) && (i.nodeToRestore = t.nodeToRestore);
    let n = t.children;
    o && (o.removeChild(t), n.size > 0 && n.forEach((i) => o && o.addChild(i))), this.fastMap.delete(t.scopeRef);
  }
  // Pre Order Depth First
  *traverse(e = this.root) {
    if (e.scopeRef != null && (yield e), e.children.size > 0) for (let t of e.children) yield* this.traverse(t);
  }
  clone() {
    var e;
    let t = new K();
    var o;
    for (let n of this.traverse()) t.addTreeNode(n.scopeRef, (o = (e = n.parent) === null || e === void 0 ? void 0 : e.scopeRef) !== null && o !== void 0 ? o : null, n.nodeToRestore);
    return t;
  }
  constructor() {
    this.fastMap = /* @__PURE__ */ new Map(), this.root = new A({
      scopeRef: null
    }), this.fastMap.set(null, this.root);
  }
}
class A {
  addChild(e) {
    this.children.add(e), e.parent = this;
  }
  removeChild(e) {
    this.children.delete(e), e.parent = void 0;
  }
  constructor(e) {
    this.children = /* @__PURE__ */ new Set(), this.contain = !1, this.scopeRef = e.scopeRef;
  }
}
let b = new K();
export {
  k as $,
  he as a,
  fe as b
};
