// @angi/one-design-system/dist/components/es/_chunks/Form-sh8c7lqF.js
import { a as O } from "./Label-DLRS1VtT.js";
import { g as P, b as j, $ as Q } from "./utils-CjtkL41C.js";
import U, { useMemo as x, useContext as q, createContext as v, useState as $, useRef as T, useEffect as F, forwardRef as N } from "react";
function Z(e) {
  let { description: a, errorMessage: t, isInvalid: i, validationState: o } = e, { labelProps: l, fieldProps: d } = O(e), n = P([
    !!a,
    !!t,
    i,
    o
  ]), s = P([
    !!a,
    !!t,
    i,
    o
  ]);
  return d = j(d, {
    "aria-describedby": [
      n,
      // Use aria-describedby for error message because aria-errormessage is unsupported using VoiceOver or NVDA. See https://github.com/adobe/react-spectrum/issues/1346#issuecomment-740136268
      s,
      e["aria-describedby"]
    ].filter(Boolean).join(" ") || void 0
  }), {
    labelProps: l,
    fieldProps: d,
    descriptionProps: {
      id: n
    },
    errorMessageProps: {
      id: s
    }
  };
}
const g = {
  badInput: !1,
  customError: !1,
  patternMismatch: !1,
  rangeOverflow: !1,
  rangeUnderflow: !1,
  stepMismatch: !1,
  tooLong: !1,
  tooShort: !1,
  typeMismatch: !1,
  valueMissing: !1,
  valid: !0
}, C = {
  ...g,
  customError: !0,
  valid: !1
}, f = {
  isInvalid: !1,
  validationDetails: g,
  validationErrors: []
}, Y = v({}), A = "__formValidationState" + Date.now();
function ee(e) {
  if (e[A]) {
    let { realtimeValidation: a, displayValidation: t, updateValidation: i, resetValidation: o, commitValidation: l } = e[A];
    return {
      realtimeValidation: a,
      displayValidation: t,
      updateValidation: i,
      resetValidation: o,
      commitValidation: l
    };
  }
  return k(e);
}
function k(e) {
  let { isInvalid: a, validationState: t, name: i, value: o, builtinValidation: l, validate: d, validationBehavior: n = "aria" } = e;
  t && (a || (a = t === "invalid"));
  let s = a !== void 0 ? {
    isInvalid: a,
    validationErrors: [],
    validationDetails: C
  } : null, b = x(() => {
    if (!d || o == null) return null;
    let r = z(d, o);
    return B(r);
  }, [
    d,
    o
  ]);
  l?.validationDetails.valid && (l = void 0);
  let c = q(Y), M = x(() => i ? Array.isArray(i) ? i.flatMap((r) => S(c[r])) : S(c[i]) : [], [
    c,
    i
  ]), [w, R] = $(c), [h, m] = $(!1);
  c !== w && (R(c), m(!1));
  let p = x(() => B(h ? [] : M), [
    h,
    M
  ]), D = T(f), [V, E] = $(f), u = T(f), _ = () => {
    if (!L) return;
    y(!1);
    let r = b || l || D.current;
    I(r, u.current) || (u.current = r, E(r));
  }, [L, y] = $(!1);
  return F(_), {
    realtimeValidation: s || p || b || l || f,
    displayValidation: n === "native" ? s || p || V : s || p || b || l || V,
    updateValidation(r) {
      n === "aria" && !I(V, r) ? E(r) : D.current = r;
    },
    resetValidation() {
      let r = f;
      I(r, u.current) || (u.current = r, E(r)), n === "native" && y(!1), m(!0);
    },
    commitValidation() {
      n === "native" && y(!0), m(!0);
    }
  };
}
function S(e) {
  return e ? Array.isArray(e) ? e : [
    e
  ] : [];
}
function z(e, a) {
  if (typeof e == "function") {
    let t = e(a);
    if (t && typeof t != "boolean") return S(t);
  }
  return [];
}
function B(e) {
  return e.length ? {
    isInvalid: !0,
    validationErrors: e,
    validationDetails: C
  } : null;
}
function I(e, a) {
  return e === a ? !0 : !!e && !!a && e.isInvalid === a.isInvalid && e.validationErrors.length === a.validationErrors.length && e.validationErrors.every((t, i) => t === a.validationErrors[i]) && Object.entries(e.validationDetails).every(([t, i]) => a.validationDetails[t] === i);
}
function ae(...e) {
  let a = /* @__PURE__ */ new Set(), t = !1, i = {
    ...g
  };
  for (let d of e) {
    var o, l;
    for (let n of d.validationErrors) a.add(n);
    t || (t = d.isInvalid);
    for (let n in i) (o = i)[l = n] || (o[l] = d.validationDetails[n]);
  }
  return i.valid = !t, {
    isInvalid: t,
    validationErrors: [
      ...a
    ],
    validationDetails: i
  };
}
const G = /* @__PURE__ */ v({}), te = /* @__PURE__ */ N(function(a, t) {
  [a, t] = Q(a, t, G);
  let { elementType: i = "span", ...o } = a;
  return /* @__PURE__ */ U.createElement(i, {
    className: "react-aria-Text",
    ...o,
    ref: t
  });
}), ie = /* @__PURE__ */ v(null), re = /* @__PURE__ */ v(null);
export {
  Z as $,
  ee as a,
  ae as b,
  re as c,
  G as d,
  ie as e,
  te as f,
  A as g,
  f as h
};
