// @angi/one-design-system/dist/components/es/_chunks/index-BdJ5ogLC.js
import * as l from "react";
import { jsx as y } from "react/jsx-runtime";
function d(n, e) {
  if (typeof n == "function")
    return n(e);
  n != null && (n.current = e);
}
function g(...n) {
  return (e) => {
    let r = !1;
    const i = n.map((t) => {
      const o = d(t, e);
      return !r && typeof o == "function" && (r = !0), o;
    });
    if (r)
      return () => {
        for (let t = 0; t < i.length; t++) {
          const o = i[t];
          typeof o == "function" ? o() : d(n[t], null);
        }
      };
  };
}
var C = Symbol.for("react.lazy"), u = l[" use ".trim().toString()];
function E(n) {
  return typeof n == "object" && n !== null && "then" in n;
}
function m(n) {
  return n != null && typeof n == "object" && "$$typeof" in n && n.$$typeof === C && "_payload" in n && E(n._payload);
}
// @__NO_SIDE_EFFECTS__
function h(n) {
  const e = /* @__PURE__ */ R(n), r = l.forwardRef((i, t) => {
    let { children: o, ...f } = i;
    m(o) && typeof u == "function" && (o = u(o._payload));
    const s = l.Children.toArray(o), a = s.find(b);
    if (a) {
      const c = a.props.children, S = s.map((p) => p === a ? l.Children.count(c) > 1 ? l.Children.only(null) : l.isValidElement(c) ? c.props.children : null : p);
      return /* @__PURE__ */ y(e, { ...f, ref: t, children: l.isValidElement(c) ? l.cloneElement(c, void 0, S) : null });
    }
    return /* @__PURE__ */ y(e, { ...f, ref: t, children: o });
  });
  return r.displayName = `${n}.Slot`, r;
}
var A = /* @__PURE__ */ h("Slot");
// @__NO_SIDE_EFFECTS__
function R(n) {
  const e = l.forwardRef((r, i) => {
    let { children: t, ...o } = r;
    if (m(t) && typeof u == "function" && (t = u(t._payload)), l.isValidElement(t)) {
      const f = P(t), s = $(o, t.props);
      return t.type !== l.Fragment && (s.ref = i ? g(i, f) : f), l.cloneElement(t, s);
    }
    return l.Children.count(t) > 1 ? l.Children.only(null) : null;
  });
  return e.displayName = `${n}.SlotClone`, e;
}
var _ = Symbol("radix.slottable");
function b(n) {
  return l.isValidElement(n) && typeof n.type == "function" && "__radixId" in n.type && n.type.__radixId === _;
}
function $(n, e) {
  const r = { ...e };
  for (const i in e) {
    const t = n[i], o = e[i];
    /^on[A-Z]/.test(i) ? t && o ? r[i] = (...s) => {
      const a = o(...s);
      return t(...s), a;
    } : t && (r[i] = t) : i === "style" ? r[i] = { ...t, ...o } : i === "className" && (r[i] = [t, o].filter(Boolean).join(" "));
  }
  return { ...n, ...r };
}
function P(n) {
  let e = Object.getOwnPropertyDescriptor(n.props, "ref")?.get, r = e && "isReactWarning" in e && e.isReactWarning;
  return r ? n.ref : (e = Object.getOwnPropertyDescriptor(n, "ref")?.get, r = e && "isReactWarning" in e && e.isReactWarning, r ? n.props.ref : n.props.ref || n.ref);
}
export {
  A as S
};
