// @angi/one-design-system/dist/components/es/_chunks/Input-C70jFPqg.js
import { $, a as c, b as u } from "./utils-CjtkL41C.js";
import { $ as l } from "./Label-DLRS1VtT.js";
import b, { createContext as v } from "react";
import { $ as m } from "./useHover-CfJvhePa.js";
import { $ as x } from "./useFocusRing-CdwnmTWa.js";
const H = /* @__PURE__ */ v({});
let F = (o) => {
  let { onHoverStart: e, onHoverChange: a, onHoverEnd: t, ...d } = o;
  return d;
};
const E = /* @__PURE__ */ l(function(e, a) {
  [e, a] = $(e, a, H);
  let { hoverProps: t, isHovered: d } = m({
    ...e,
    isDisabled: e.disabled
  }), { isFocused: i, isFocusVisible: s, focusProps: f } = x({
    isTextInput: !0,
    autoFocus: e.autoFocus
  }), r = !!e["aria-invalid"] && e["aria-invalid"] !== "false", n = c({
    ...e,
    values: {
      isHovered: d,
      isFocused: i,
      isFocusVisible: s,
      isDisabled: e.disabled || !1,
      isInvalid: r
    },
    defaultClassName: "react-aria-Input"
  });
  return /* @__PURE__ */ b.createElement("input", {
    ...u(F(e), f, t),
    ...n,
    ref: a,
    "data-focused": i || void 0,
    "data-disabled": e.disabled || void 0,
    "data-hovered": d || void 0,
    "data-focus-visible": s || void 0,
    "data-invalid": r || void 0
  });
});
export {
  H as $,
  E as a
};
