// @angi/one-design-system/dist/components/es/_chunks/Input-Dyh2Ln22.js
import { jsxs as P, jsx as a } from "react/jsx-runtime";
import * as e from "react";
import { forwardRef as V, useState as $, useRef as k, useImperativeHandle as U } from "react";
import { a as g } from "./Input-C70jFPqg.js";
import { c as S } from "./createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import { c as C } from "./clsx-OuTLNxxd.js";
import { S as F } from "./xmark-CxuwVom6.js";
import { B as _ } from "./Button-CG9o4a5a.js";
import '../assets/Input.css';var j = S({ defaultClassName: "_1c8myl61 _1c8myl60", variantClassNames: { size: { sm: "_1c8myl62", md: "_1c8myl63" }, isDisabled: { true: "_1c8myl64", false: "_1c8myl65" }, isReadOnly: { true: "_1c8myl66", false: "_1c8myl67" } }, defaultVariants: { size: "md", isDisabled: !1, isReadOnly: !1 }, compoundVariants: [] }), O = "_1c8myl68", z = "_1c8myl69", G = "_1c8myl6a", H = S({ defaultClassName: "_1c8myl6c _1c8myl6b", variantClassNames: { size: { sm: "_1c8myl6d", md: "_1c8myl6e" } }, defaultVariants: { size: "md" }, compoundVariants: [] });
const i1 = ["sm", "md"], W = [
  "text",
  "search",
  "tel",
  "url",
  "email",
  "password",
  "number",
  "date",
  "month",
  "week",
  "time",
  "datetime-local"
], m1 = W, d1 = ["invalid", "valid"], X = "md", Y = "text", Z = (s) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 576 512", ...s }, /* @__PURE__ */ e.createElement("path", { fill: "currentColor", d: "M288 80C222.8 80 169.2 109.6 128.1 147.7 89.6 183.5 63 226 49.4 256 63 286 89.6 328.5 128.1 364.3 169.2 402.4 222.8 432 288 432s118.8-29.6 159.9-67.7C486.4 328.5 513 286 526.6 256 513 226 486.4 183.5 447.9 147.7 406.8 109.6 353.2 80 288 80zM95.4 112.6C142.5 68.8 207.2 32 288 32s145.5 36.8 192.6 80.6c46.8 43.5 78.1 95.4 93 131.1 3.3 7.9 3.3 16.7 0 24.6-14.9 35.7-46.2 87.7-93 131.1-47.1 43.7-111.8 80.6-192.6 80.6S142.5 443.2 95.4 399.4c-46.8-43.5-78.1-95.4-93-131.1-3.3-7.9-3.3-16.7 0-24.6 14.9-35.7 46.2-87.7 93-131.1zM288 336c44.2 0 80-35.8 80-80 0-29.6-16.1-55.5-40-69.3-1.4 59.7-49.6 107.9-109.3 109.3 13.8 23.9 39.7 40 69.3 40zm-79.6-88.4c2.5 .3 5 .4 7.6 .4 35.3 0 64-28.7 64-64 0-2.6-.2-5.1-.4-7.6-37.4 3.9-67.2 33.7-71.1 71.1zm45.6-115c10.8-3 22.2-4.5 33.9-4.5 8.8 0 17.5 .9 25.8 2.6 .3 .1 .5 .1 .8 .2 57.9 12.2 101.4 63.7 101.4 125.2 0 70.7-57.3 128-128 128-61.6 0-113-43.5-125.2-101.4-1.8-8.6-2.8-17.5-2.8-26.6 0-11 1.4-21.8 4-32 .2-.7 .3-1.3 .5-1.9 11.9-43.4 46.1-77.6 89.5-89.5z" })), q = (s) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 576 512", ...s }, /* @__PURE__ */ e.createElement("path", { fill: "currentColor", d: "M41-24.9c-9.4-9.4-24.6-9.4-33.9 0S-2.3-.3 7 9.1l528 528c9.4 9.4 24.6 9.4 33.9 0s9.4-24.6 0-33.9l-96.4-96.4c2.7-2.4 5.4-4.8 8-7.2 46.8-43.5 78.1-95.4 93-131.1 3.3-7.9 3.3-16.7 0-24.6-14.9-35.7-46.2-87.7-93-131.1-47.1-43.7-111.8-80.6-192.6-80.6-56.8 0-105.6 18.2-146 44.2L41-24.9zM176.9 111.1c32.1-18.9 69.2-31.1 111.1-31.1 65.2 0 118.8 29.6 159.9 67.7 38.5 35.7 65.1 78.3 78.6 108.3-13.6 30-40.2 72.5-78.6 108.3-3.1 2.8-6.2 5.6-9.4 8.4L393.8 328c14-20.5 22.2-45.3 22.2-72 0-70.7-57.3-128-128-128-26.7 0-51.5 8.2-72 22.2l-39.1-39.1zm182 182l-108-108c11.1-5.8 23.7-9.1 37.1-9.1 44.2 0 80 35.8 80 80 0 13.4-3.3 26-9.1 37.1zM103.4 173.2l-34-34c-32.6 36.8-55 75.8-66.9 104.5-3.3 7.9-3.3 16.7 0 24.6 14.9 35.7 46.2 87.7 93 131.1 47.1 43.7 111.8 80.6 192.6 80.6 37.3 0 71.2-7.9 101.5-20.6L352.2 422c-20 6.4-41.4 10-64.2 10-65.2 0-118.8-29.6-159.9-67.7-38.5-35.7-65.1-78.3-78.6-108.3 10.4-23.1 28.6-53.6 54-82.8z" })), J = (s) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 512 512", ...s }, /* @__PURE__ */ e.createElement("path", { fill: "currentColor", d: "M368 208a160 160 0 1 0 -320 0 160 160 0 1 0 320 0zM337.1 371.1C301.7 399.2 256.8 416 208 416 93.1 416 0 322.9 0 208S93.1 0 208 0 416 93.1 416 208c0 48.8-16.8 93.7-44.9 129.1L505 471c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0L337.1 371.1z" })), K = V(
  ({
    size: s = X,
    validation: i,
    className: u,
    type: m = Y,
    disabled: t,
    readOnly: r,
    showPasswordLabel: I,
    hidePasswordLabel: x,
    clearSearchLabel: E,
    onClear: N,
    value: d,
    defaultValue: h,
    ...p
  }, b) => {
    const [n, B] = $(!1), l = k(null);
    U(b, () => l.current, []);
    const o = m === "password", c = m === "search", M = o || c, v = o && n ? "text" : m, T = o ? /* @__PURE__ */ a(
      _,
      {
        type: "button",
        variant: "ghost",
        color: "neutral",
        size: "sm",
        onClick: () => B(!n),
        disabled: t,
        "aria-disabled": t || void 0,
        "aria-label": n ? x : I,
        "aria-pressed": n,
        iconOnly: !0,
        iconStart: n ? /* @__PURE__ */ a(q, { width: 16, height: 16, "aria-hidden": "true" }) : /* @__PURE__ */ a(Z, { width: 16, height: 16, "aria-hidden": "true" })
      }
    ) : null, R = c ? /* @__PURE__ */ a(
      J,
      {
        width: 20,
        height: 20,
        "aria-hidden": "true",
        className: G
      }
    ) : null, L = c && !!d && !t && !r ? /* @__PURE__ */ a(
      _,
      {
        type: "button",
        variant: "ghost",
        color: "neutral",
        size: "sm",
        onClick: () => {
          N?.(), l.current?.focus();
        },
        "aria-label": E,
        iconOnly: !0,
        iconStart: /* @__PURE__ */ a(F, { width: 16, height: 16, "aria-hidden": "true" })
      }
    ) : null, f = R, w = o ? T : c ? L : null, y = i === "invalid";
    if (M) {
      const D = C(
        j({
          size: s,
          isDisabled: t ?? !1,
          isReadOnly: r ?? !1
        }),
        u
      );
      return /* @__PURE__ */ P(
        "div",
        {
          className: D,
          "data-validation": i ?? void 0,
          children: [
            f && /* @__PURE__ */ a("span", { className: z, children: f }),
            /* @__PURE__ */ a(
              g,
              {
                ref: l,
                className: O,
                type: v,
                disabled: t,
                readOnly: r,
                "aria-invalid": y || void 0,
                ...p,
                value: d,
                defaultValue: h
              }
            ),
            w && /* @__PURE__ */ a("span", { className: z, children: w })
          ]
        }
      );
    }
    const A = C(
      H({
        size: s
      }),
      u
    );
    return /* @__PURE__ */ a(
      g,
      {
        ref: l,
        className: A,
        type: v,
        disabled: t,
        readOnly: r,
        "data-validation": i ?? void 0,
        "aria-invalid": y || void 0,
        ...p,
        value: d,
        defaultValue: h
      }
    );
  }
);
K.displayName = "Input";
export {
  X as D,
  K as I,
  i1 as a,
  m1 as b,
  d1 as c,
  Y as d
};
