// @angi/one-design-system/dist/components/es/_chunks/Label-BqoOfVfQ.js
import { jsx as n } from "react/jsx-runtime";
import { forwardRef as c } from "react";
import { S as i } from "./index-BdJ5ogLC.js";
import { c as p } from "./createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import { c as f } from "./clsx-OuTLNxxd.js";
import '../assets/Label.css';var d = p({ defaultClassName: "_1w8ocpw0", variantClassNames: { variant: { bold: "_1w8ocpw1", normal: "_1w8ocpw2" }, required: { true: "_1w8ocpw3", false: "_1w8ocpw4" } }, defaultVariants: { variant: "bold", required: !1 }, compoundVariants: [] });
const A = ["normal", "bold"], w = "bold", b = c(
  ({
    children: o,
    required: a = !1,
    variant: e = w,
    asChild: r = !1,
    className: s,
    ...l
  }, t) => {
    const m = f(
      d({ variant: e, required: a }),
      s
    );
    return /* @__PURE__ */ n(
      r ? i : "label",
      {
        ref: t,
        className: m,
        ...a && { "data-required": !0 },
        ...l,
        children: o
      }
    );
  }
);
b.displayName = "Label";
export {
  w as D,
  b as L,
  A as a
};
