// @angi/one-design-system/dist/components/es/_chunks/Label-DLRS1VtT.js
import { d as b, $ as f } from "./utils-CjtkL41C.js";
import p, { forwardRef as c, useContext as $, createContext as n } from "react";
function u(r, e) {
  let { id: t, "aria-label": a, "aria-labelledby": l } = r;
  return t = b(t), l && a ? l = [
    .../* @__PURE__ */ new Set([
      t,
      ...l.trim().split(/\s+/)
    ])
  ].join(" ") : l && (l = l.trim().split(/\s+/).join(" ")), !a && !l && e && (a = e), {
    id: t,
    "aria-label": a,
    "aria-labelledby": l
  };
}
function T(r) {
  let { id: e, label: t, "aria-labelledby": a, "aria-label": l, labelElementType: d = "label" } = r;
  e = b(e);
  let i = b(), o = {};
  t ? (a = a ? `${i} ${a}` : i, o = {
    id: i,
    htmlFor: d === "label" ? e : void 0
  }) : !a && !l && process.env.NODE_ENV !== "production" && console.warn("If you do not provide a visible label, you must specify an aria-label or aria-labelledby attribute for accessibility");
  let s = u({
    id: e,
    "aria-label": l,
    "aria-labelledby": a
  });
  return {
    labelProps: o,
    fieldProps: s
  };
}
if (typeof HTMLTemplateElement < "u") {
  const r = Object.getOwnPropertyDescriptor(Node.prototype, "firstChild").get;
  Object.defineProperty(HTMLTemplateElement.prototype, "firstChild", {
    configurable: !0,
    enumerable: !0,
    get: function() {
      return this.dataset.reactAriaHidden ? this.content.firstChild : r.call(this);
    }
  });
}
const m = /* @__PURE__ */ n(!1);
function y(r) {
  let e = (t, a) => $(m) ? null : r(t, a);
  return e.displayName = r.displayName || r.name, c(e);
}
const x = /* @__PURE__ */ n({}), P = /* @__PURE__ */ y(function(e, t) {
  [e, t] = f(e, t, x);
  let { elementType: a = "label", ...l } = e;
  return /* @__PURE__ */ p.createElement(a, {
    className: "react-aria-Label",
    ...l,
    ref: t
  });
});
export {
  y as $,
  T as a,
  x as b,
  P as c,
  u as d
};
