// @angi/one-design-system/dist/components/es/_chunks/openLink-EIDcTMUl.js
import { $ as g } from "./focusWithoutScrolling-CDw3JYiq.js";
import A, { useMemo as x, createContext as K, useContext as _ } from "react";
function b(e) {
  var t;
  if (typeof window > "u" || window.navigator == null) return !1;
  let a = (t = window.navigator.userAgentData) === null || t === void 0 ? void 0 : t.brands;
  return Array.isArray(a) && a.some((n) => e.test(n.brand)) || e.test(window.navigator.userAgent);
}
function s(e) {
  var t;
  return typeof window < "u" && window.navigator != null ? e.test(((t = window.navigator.userAgentData) === null || t === void 0 ? void 0 : t.platform) || window.navigator.platform) : !1;
}
function r(e) {
  if (process.env.NODE_ENV === "test") return e;
  let t = null;
  return () => (t == null && (t = e()), t);
}
const $ = r(function() {
  return s(/^Mac/i);
}), m = r(function() {
  return s(/^iPhone/i);
}), v = r(function() {
  return s(/^iPad/i) || // iPadOS 13 lies and says it's a Mac, but we can distinguish by detecting touch support.
  $() && navigator.maxTouchPoints > 1;
}), E = r(function() {
  return m() || v();
});
r(function() {
  return $() || E();
});
const P = r(function() {
  return b(/AppleWebKit/i) && !D();
}), D = r(function() {
  return b(/Chrome/i);
}), k = r(function() {
  return b(/Android/i);
}), p = r(function() {
  return b(/Firefox/i);
}), y = /* @__PURE__ */ K({
  isNative: !0,
  open: N,
  useHref: (e) => e
});
function M(e) {
  let { children: t, navigate: a, useHref: n } = e, c = x(() => ({
    isNative: !1,
    open: (i, o, u, l) => {
      w(i, (f) => {
        h(f, o) ? a(u, l) : d(f, o);
      });
    },
    useHref: n || ((i) => i)
  }), [
    a,
    n
  ]);
  return /* @__PURE__ */ A.createElement(y.Provider, {
    value: c
  }, t);
}
function H() {
  return _(y);
}
function h(e, t) {
  let a = e.getAttribute("target");
  return (!a || a === "_self") && e.origin === location.origin && !e.hasAttribute("download") && !t.metaKey && // open in new tab (mac)
  !t.ctrlKey && // open in new tab (windows)
  !t.altKey && // download
  !t.shiftKey;
}
function d(e, t, a = !0) {
  var n, c;
  let { metaKey: i, ctrlKey: o, altKey: u, shiftKey: l } = t;
  p() && (!((c = window.event) === null || c === void 0 || (n = c.type) === null || n === void 0) && n.startsWith("key")) && e.target === "_blank" && ($() ? i = !0 : o = !0);
  let f = P() && $() && !v() && process.env.NODE_ENV !== "test" ? new KeyboardEvent("keydown", {
    keyIdentifier: "Enter",
    metaKey: i,
    ctrlKey: o,
    altKey: u,
    shiftKey: l
  }) : new MouseEvent("click", {
    metaKey: i,
    ctrlKey: o,
    altKey: u,
    shiftKey: l,
    detail: 1,
    bubbles: !0,
    cancelable: !0
  });
  d.isOpening = a, g(e), e.dispatchEvent(f), d.isOpening = !1;
}
d.isOpening = !1;
function w(e, t) {
  if (e instanceof HTMLAnchorElement) t(e);
  else if (e.hasAttribute("data-href")) {
    let a = document.createElement("a");
    a.href = e.getAttribute("data-href"), e.hasAttribute("data-target") && (a.target = e.getAttribute("data-target")), e.hasAttribute("data-rel") && (a.rel = e.getAttribute("data-rel")), e.hasAttribute("data-download") && (a.download = e.getAttribute("data-download")), e.hasAttribute("data-ping") && (a.ping = e.getAttribute("data-ping")), e.hasAttribute("data-referrer-policy") && (a.referrerPolicy = e.getAttribute("data-referrer-policy")), e.appendChild(a), t(a), e.removeChild(a);
  }
}
function N(e, t) {
  w(e, (a) => d(a, t));
}
function O(e) {
  let t = H();
  var a;
  const n = t.useHref((a = e?.href) !== null && a !== void 0 ? a : "");
  return {
    href: e?.href ? n : void 0,
    target: e?.target,
    rel: e?.rel,
    download: e?.download,
    ping: e?.ping,
    referrerPolicy: e?.referrerPolicy
  };
}
function L(e, t, a, n) {
  !t.isNative && e.currentTarget instanceof HTMLAnchorElement && e.currentTarget.href && // If props are applied to a router Link component, it may have already prevented default.
  !e.isDefaultPrevented() && h(e.currentTarget, e) && a && (e.preventDefault(), t.open(e.currentTarget, e, a, n));
}
export {
  M as $,
  k as a,
  D as b,
  d as c,
  $ as d,
  H as e,
  O as f,
  L as g,
  E as h
};
