// @angi/one-design-system/dist/components/es/_chunks/ProgressBar-CJx2i27x.js
import { b as h, $ as P, f as w, a as D } from "./utils-CjtkL41C.js";
import { a as F, b as E } from "./Label-DLRS1VtT.js";
import { $ as v } from "./filterDOMProps-Cxz9cAAv.js";
import $, { useMemo as R, forwardRef as O, createContext as V } from "react";
import { $ as T } from "./context-qNvgpQEb.js";
function y(r, e = -1 / 0, t = 1 / 0) {
  return Math.min(Math.max(r, e), t);
}
let b = /* @__PURE__ */ new Map(), p = !1;
try {
  p = new Intl.NumberFormat("de-DE", {
    signDisplay: "exceptZero"
  }).resolvedOptions().signDisplay === "exceptZero";
} catch {
}
let u = !1;
try {
  u = new Intl.NumberFormat("de-DE", {
    style: "unit",
    unit: "degree"
  }).resolvedOptions().style === "unit";
} catch {
}
const g = {
  degree: {
    narrow: {
      default: "°",
      "ja-JP": " 度",
      "zh-TW": "度",
      "sl-SI": " °"
    }
  }
};
class I {
  /** Formats a number value as a string, according to the locale and options provided to the constructor. */
  format(e) {
    let t = "";
    if (!p && this.options.signDisplay != null ? t = j(this.numberFormatter, this.options.signDisplay, e) : t = this.numberFormatter.format(e), this.options.style === "unit" && !u) {
      var a;
      let { unit: i, unitDisplay: l = "short", locale: n } = this.resolvedOptions();
      if (!i) return t;
      let s = (a = g[i]) === null || a === void 0 ? void 0 : a[l];
      t += s[n] || s.default;
    }
    return t;
  }
  /** Formats a number to an array of parts such as separators, digits, punctuation, and more. */
  formatToParts(e) {
    return this.numberFormatter.formatToParts(e);
  }
  /** Formats a number range as a string. */
  formatRange(e, t) {
    if (typeof this.numberFormatter.formatRange == "function") return this.numberFormatter.formatRange(e, t);
    if (t < e) throw new RangeError("End date must be >= start date");
    return `${this.format(e)} – ${this.format(t)}`;
  }
  /** Formats a number range as an array of parts. */
  formatRangeToParts(e, t) {
    if (typeof this.numberFormatter.formatRangeToParts == "function") return this.numberFormatter.formatRangeToParts(e, t);
    if (t < e) throw new RangeError("End date must be >= start date");
    let a = this.numberFormatter.formatToParts(e), i = this.numberFormatter.formatToParts(t);
    return [
      ...a.map((l) => ({
        ...l,
        source: "startRange"
      })),
      {
        type: "literal",
        value: " – ",
        source: "shared"
      },
      ...i.map((l) => ({
        ...l,
        source: "endRange"
      }))
    ];
  }
  /** Returns the resolved formatting options based on the values passed to the constructor. */
  resolvedOptions() {
    let e = this.numberFormatter.resolvedOptions();
    return !p && this.options.signDisplay != null && (e = {
      ...e,
      signDisplay: this.options.signDisplay
    }), !u && this.options.style === "unit" && (e = {
      ...e,
      style: "unit",
      unit: this.options.unit,
      unitDisplay: this.options.unitDisplay
    }), e;
  }
  constructor(e, t = {}) {
    this.numberFormatter = M(e, t), this.options = t;
  }
}
function M(r, e = {}) {
  let { numberingSystem: t } = e;
  if (t && r.includes("-nu-") && (r.includes("-u-") || (r += "-u-"), r += `-nu-${t}`), e.style === "unit" && !u) {
    var a;
    let { unit: n, unitDisplay: s = "short" } = e;
    if (!n) throw new Error('unit option must be provided with style: "unit"');
    if (!(!((a = g[n]) === null || a === void 0) && a[s])) throw new Error(`Unsupported unit ${n} with unitDisplay = ${s}`);
    e = {
      ...e,
      style: "decimal"
    };
  }
  let i = r + (e ? Object.entries(e).sort((n, s) => n[0] < s[0] ? -1 : 1).join() : "");
  if (b.has(i)) return b.get(i);
  let l = new Intl.NumberFormat(r, e);
  return b.set(i, l), l;
}
function j(r, e, t) {
  if (e === "auto") return r.format(t);
  if (e === "never") return r.format(Math.abs(t));
  {
    let a = !1;
    if (e === "always" ? a = t > 0 || Object.is(t, 0) : e === "exceptZero" && (Object.is(t, -0) || Object.is(t, 0) ? t = Math.abs(t) : a = t > 0), a) {
      let i = r.format(-t), l = r.format(t), n = i.replace(l, "").replace(/\u200e|\u061C/, "");
      return [
        ...n
      ].length !== 1 && console.warn("@react-aria/i18n polyfill for NumberFormat signDisplay: Unsupported case"), i.replace(l, "!!!").replace(n, "+").replace("!!!", l);
    } else return r.format(t);
  }
}
function B(r = {}) {
  let { locale: e } = T();
  return R(() => new I(e, r), [
    e,
    r
  ]);
}
function C(r) {
  let { value: e = 0, minValue: t = 0, maxValue: a = 100, valueLabel: i, isIndeterminate: l, formatOptions: n = {
    style: "percent"
  } } = r, s = v(r, {
    labelable: !0
  }), { labelProps: d, fieldProps: o } = F({
    ...r,
    // Progress bar is not an HTML input element so it
    // shouldn't be labeled by a <label> element.
    labelElementType: "span"
  });
  e = y(e, t, a);
  let c = (e - t) / (a - t), m = B(n);
  if (!l && !i) {
    let f = n.style === "percent" ? c : e;
    i = m.format(f);
  }
  return {
    progressBarProps: h(s, {
      ...o,
      "aria-valuenow": l ? void 0 : e,
      "aria-valuemin": t,
      "aria-valuemax": a,
      "aria-valuetext": l ? void 0 : i,
      role: "progressbar"
    }),
    labelProps: d
  };
}
const S = /* @__PURE__ */ V(null), J = /* @__PURE__ */ O(function(e, t) {
  [e, t] = P(e, t, S);
  let { value: a = 0, minValue: i = 0, maxValue: l = 100, isIndeterminate: n = !1 } = e;
  a = y(a, i, l);
  let [s, d] = w(!e["aria-label"] && !e["aria-labelledby"]), { progressBarProps: o, labelProps: c } = C({
    ...e,
    label: d
  }), m = n ? void 0 : (a - i) / (l - i) * 100, f = D({
    ...e,
    defaultClassName: "react-aria-ProgressBar",
    values: {
      percentage: m,
      valueText: o["aria-valuetext"],
      isIndeterminate: n
    }
  }), x = v(e, {
    global: !0
  });
  return /* @__PURE__ */ $.createElement("div", {
    ...h(x, f, o),
    ref: t,
    slot: e.slot || void 0
  }, /* @__PURE__ */ $.createElement(E.Provider, {
    value: {
      ...c,
      ref: s,
      elementType: "span"
    }
  }, f.children));
});
export {
  S as $,
  J as a
};
