// @angi/one-design-system/dist/components/es/_chunks/RadioItem-CkMea6Af.js
import { jsx as y } from "react/jsx-runtime";
import { useMemo as L, forwardRef as w, useRef as z, useImperativeHandle as A } from "react";
import { a as j, u as q } from "./useRadioGroupContext-CuaGc53C.js";
import { c as B } from "./clsx-OuTLNxxd.js";
import { c as O } from "./createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import { b as c } from "./utils-CjtkL41C.js";
import { $ as T } from "./filterDOMProps-Cxz9cAAv.js";
import { $ as G, a as H } from "./useFormValidation-CgQ_8wCr.js";
import { $ as g } from "./usePress-D9Qp6Amx.js";
import { a as Z } from "./useFocusable-CrxUcjB4.js";
import { $ as J } from "./useFocusRing-CdwnmTWa.js";
import '../assets/RadioItem.css';function K(o, e, r) {
  let { value: s, children: u, "aria-label": f, "aria-labelledby": b, onPressStart: n, onPressEnd: t, onPressChange: p, onPress: l, onPressUp: m, onClick: d } = o;
  const a = o.isDisabled || e.isDisabled;
  let $ = u != null, h = f != null || b != null;
  !$ && !h && process.env.NODE_ENV !== "production" && console.warn("If you do not provide children, you must specify an aria-label for accessibility");
  let v = e.selectedValue === s, I = (i) => {
    i.stopPropagation(), e.setSelectedValue(s);
  }, { pressProps: C, isPressed: x } = g({
    onPressStart: n,
    onPressEnd: t,
    onPressChange: p,
    onPress: l,
    onPressUp: m,
    onClick: d,
    isDisabled: a
  }), { pressProps: D, isPressed: R } = g({
    onPressStart: n,
    onPressEnd: t,
    onPressChange: p,
    onPressUp: m,
    onClick: d,
    isDisabled: a,
    onPress(i) {
      var V;
      l?.(i), e.setSelectedValue(s), (V = r.current) === null || V === void 0 || V.focus();
    }
  }), { focusableProps: _ } = Z(c(o, {
    onFocus: () => e.setLastFocusedValue(s)
  }), r), E = c(C, _), k = T(o, {
    labelable: !0
  }), P = -1;
  e.selectedValue != null ? e.selectedValue === s && (P = 0) : (e.lastFocusedValue === s || e.lastFocusedValue == null) && (P = 0), a && (P = void 0);
  let { name: F, form: N, descriptionId: M, errorMessageId: U, validationBehavior: S } = j.get(e);
  return G(r, e.defaultSelectedValue, e.setSelectedValue), H({
    validationBehavior: S
  }, e, r), {
    labelProps: c(D, L(() => ({
      onClick: (i) => i.preventDefault(),
      // Prevent label from being focused when mouse down on it.
      // Note, this does not prevent the input from being focused in the `click` event.
      onMouseDown: (i) => i.preventDefault()
    }), [])),
    inputProps: c(k, {
      ...E,
      type: "radio",
      name: F,
      form: N,
      tabIndex: P,
      disabled: a,
      required: e.isRequired && S === "native",
      checked: v,
      value: s,
      onChange: I,
      "aria-describedby": [
        o["aria-describedby"],
        e.isInvalid ? U : null,
        M
      ].filter(Boolean).join(" ") || void 0
    }),
    isDisabled: a,
    isSelected: v,
    isPressed: x || R
  };
}
var Q = "_12t7avl0", W = O({ defaultClassName: "_12t7avl2 _12t7avl1", variantClassNames: { size: { sm: "_12t7avl3", md: "_12t7avl4" } }, defaultVariants: { size: "md" }, compoundVariants: [] });
const ue = ["sm", "md"], X = "md", Y = w(
  ({
    size: o = X,
    disabled: e = !1,
    value: r,
    className: s,
    id: u,
    name: f,
    "aria-label": b,
    "aria-labelledby": n,
    "aria-describedby": t,
    hideFocusRing: p = !1,
    ...l
  }, m) => {
    const d = z(null);
    A(
      m,
      () => d.current
    );
    const a = q(), { inputProps: $, isSelected: h, isDisabled: v } = K(
      {
        value: r,
        isDisabled: e,
        ...b !== void 0 && { "aria-label": b },
        ...n !== void 0 && {
          "aria-labelledby": n
        },
        ...t !== void 0 && {
          "aria-describedby": t
        },
        ...u !== void 0 && { id: u },
        ...f !== void 0 && { name: f }
      },
      a,
      d
    ), { isFocusVisible: I, focusProps: C } = J(), x = a.isInvalid, D = B(
      W({
        size: o
      }),
      s
    );
    return /* @__PURE__ */ y(
      "span",
      {
        className: D,
        "data-checked": h || void 0,
        "data-disabled": v || void 0,
        "data-invalid": x || void 0,
        "data-focus-visible": !p && I || void 0,
        ...l,
        children: /* @__PURE__ */ y(
          "input",
          {
            ...c($, C),
            ref: d,
            className: Q
          }
        )
      }
    );
  }
);
Y.displayName = "RadioItem";
export {
  X as D,
  Y as R,
  ue as a
};
