// @angi/one-design-system/dist/components/es/_chunks/RadioButton-Cl7_yIuj.js
import { jsxs as f, jsx as o } from "react/jsx-runtime";
import { forwardRef as D, useId as N } from "react";
import { c as g } from "./clsx-OuTLNxxd.js";
import { u as H } from "./useRadioGroupContext-CuaGc53C.js";
import { L as $ } from "./Label-BqoOfVfQ.js";
import { H as w } from "./Hint-CIDmUonJ.js";
import { R as G } from "./RadioItem-CkMea6Af.js";
import '../assets/RadioButton.css';var L = "pcejf60", k = "pcejf61", q = "pcejf62";
const p = (...e) => {
  const a = e.filter(Boolean);
  return a.length > 0 ? a.join(" ") : void 0;
}, A = D(
  ({
    // Custom props
    label: e,
    hint: a,
    className: v,
    // RadioItem props
    id: u,
    value: y,
    size: i,
    name: t,
    hideFocusRing: d,
    "aria-label": s,
    "aria-describedby": I,
    "aria-labelledby": h,
    disabled: l = !1,
    // Rest are label element props (forwarded from Slot)
    ...j
  }, x) => {
    const R = N(), r = u ?? R, n = `${r}-label`, c = a ? `${r}-hint` : void 0, m = p(I, c), b = p(h, n), B = H(), C = l || B.isDisabled;
    return /* @__PURE__ */ f(
      "label",
      {
        ref: x,
        ...j,
        className: g(
          L,
          C && k,
          v
        ),
        children: [
          /* @__PURE__ */ o(
            G,
            {
              id: r,
              value: y,
              disabled: l,
              ...i !== void 0 && { size: i },
              ...t !== void 0 && { name: t },
              ...d !== void 0 && { hideFocusRing: d },
              ...s !== void 0 && { "aria-label": s },
              ...b !== void 0 && { "aria-labelledby": b },
              ...m !== void 0 && {
                "aria-describedby": m
              }
            }
          ),
          /* @__PURE__ */ f("span", { className: q, children: [
            /* @__PURE__ */ o($, { variant: "normal", ...e, id: n, asChild: !0, children: /* @__PURE__ */ o("span", { children: e.children }) }),
            a && /* @__PURE__ */ o(w, { ...a, id: c })
          ] })
        ]
      }
    );
  }
);
A.displayName = "RadioButton";
export {
  A as R
};
