// @angi/one-design-system/dist/components/es/_chunks/Spinner-B7L4uec2.js
import { jsx as r, jsxs as _ } from "react/jsx-runtime";
import { forwardRef as j } from "react";
import { c as x } from "./createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import { c as g } from "./clsx-OuTLNxxd.js";
import { u as S } from "./useTranslations-Da7GR3Ot.js";
import '../assets/Spinner.css';var N = x({ defaultClassName: "_12znjen1", variantClassNames: { size: { sm: "_12znjen2", md: "_12znjen3", lg: "_12znjen4", xl: "_12znjen5", xxl: "_12znjen6" } }, defaultVariants: { size: "md" }, compoundVariants: [] }), f = "_12znjen7", u = { sm: "_12znjen9 _12znjen8", md: "_12znjena _12znjen8" }, v = { sm: "_12znjenc _12znjenb", md: "_12znjend _12znjenb" };
const B = ["sm", "md", "lg", "xl", "xxl"], U = ["sm", "md"], h = "md", E = "sm", $ = {
  sm: 12,
  md: 16,
  lg: 24,
  xl: 32,
  xxl: 40
}, w = {
  sm: 2,
  md: 4
}, C = { loading: "Wird geladen…" }, T = { loading: "Loading…" }, b = { loading: "Chargement…" }, L = { loading: "Caricamento in corso…" }, R = { loading: "Laden…" }, A = {
  de: C,
  en: T,
  fr: b,
  it: L,
  nl: R
}, I = j(
  ({
    size: o = h,
    thickness: a = E,
    label: i,
    className: t,
    ...c
  }, m) => {
    const l = S(A), d = i ?? l.loading, e = $[o], n = e / 2, p = w[a], s = n - p / 2, z = `M ${n} ${n - s} A ${s} ${s} 0 0 1 ${n + s} ${n}`;
    return /* @__PURE__ */ r(
      "span",
      {
        ref: m,
        className: g(N({ size: o }), t),
        role: "status",
        "aria-label": d,
        ...c,
        children: /* @__PURE__ */ _(
          "svg",
          {
            className: f,
            viewBox: `0 0 ${e} ${e}`,
            "aria-hidden": "true",
            xmlns: "http://www.w3.org/2000/svg",
            children: [
              /* @__PURE__ */ r(
                "circle",
                {
                  className: u[a],
                  cx: n,
                  cy: n,
                  r: s
                }
              ),
              /* @__PURE__ */ r("path", { className: v[a], d: z })
            ]
          }
        )
      }
    );
  }
);
I.displayName = "Spinner";
export {
  h as D,
  I as S,
  B as a,
  U as b,
  E as c
};
