// @angi/one-design-system/dist/components/es/_chunks/TextArea-DWRv-YP4.js
import { $ as c, a as $, b as l } from "./utils-CjtkL41C.js";
import u, { forwardRef as v, createContext as b } from "react";
import { $ as x } from "./useHover-CfJvhePa.js";
import { $ as m } from "./useFocusRing-CdwnmTWa.js";
const H = /* @__PURE__ */ b({});
let F = (o) => {
  let { onHoverStart: e, onHoverChange: a, onHoverEnd: t, ...d } = o;
  return d;
};
const I = /* @__PURE__ */ v(function(e, a) {
  [e, a] = c(e, a, H);
  let { hoverProps: t, isHovered: d } = x(e), { isFocused: r, isFocusVisible: i, focusProps: f } = m({
    isTextInput: !0,
    autoFocus: e.autoFocus
  }), s = !!e["aria-invalid"] && e["aria-invalid"] !== "false", n = $({
    ...e,
    values: {
      isHovered: d,
      isFocused: r,
      isFocusVisible: i,
      isDisabled: e.disabled || !1,
      isInvalid: s
    },
    defaultClassName: "react-aria-TextArea"
  });
  return /* @__PURE__ */ u.createElement("textarea", {
    ...l(F(e), f, t),
    ...n,
    ref: a,
    "data-focused": r || void 0,
    "data-disabled": e.disabled || void 0,
    "data-hovered": d || void 0,
    "data-focus-visible": i || void 0,
    "data-invalid": s || void 0
  });
});
export {
  I as $,
  H as a
};
