// @angi/one-design-system/dist/components/es/_chunks/types-BVtnuMQb.js
import { b as T, s as X } from "./tokens-DMvC-b0j.js";
function q(r, e) {
  if (typeof r != "object" || !r) return r;
  var t = r[Symbol.toPrimitive];
  if (t !== void 0) {
    var o = t.call(r, e);
    if (typeof o != "object") return o;
    throw new TypeError("@@toPrimitive must return a primitive value.");
  }
  return (e === "string" ? String : Number)(r);
}
function z(r) {
  var e = q(r, "string");
  return typeof e == "symbol" ? e : String(e);
}
function G(r, e, t) {
  return e = z(e), e in r ? Object.defineProperty(r, e, {
    value: t,
    enumerable: !0,
    configurable: !0,
    writable: !0
  }) : r[e] = t, r;
}
function R(r, e) {
  var t = Object.keys(r);
  if (Object.getOwnPropertySymbols) {
    var o = Object.getOwnPropertySymbols(r);
    e && (o = o.filter(function(u) {
      return Object.getOwnPropertyDescriptor(r, u).enumerable;
    })), t.push.apply(t, o);
  }
  return t;
}
function E(r) {
  for (var e = 1; e < arguments.length; e++) {
    var t = arguments[e] != null ? arguments[e] : {};
    e % 2 ? R(Object(t), !0).forEach(function(o) {
      G(r, o, t[o]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(r, Object.getOwnPropertyDescriptors(t)) : R(Object(t)).forEach(function(o) {
      Object.defineProperty(r, o, Object.getOwnPropertyDescriptor(t, o));
    });
  }
  return r;
}
var H = (r) => function() {
  for (var e = arguments.length, t = new Array(e), o = 0; o < e; o++)
    t[o] = arguments[o];
  var u = Object.assign({}, ...t.map((a) => a.styles)), S = Object.keys(u), $ = S.filter((a) => "mappings" in u[a]), x = (a) => {
    var p = [], P = {}, b = E({}, a), k = !1;
    for (var D of $) {
      var A = a[D];
      if (A != null) {
        var F = u[D];
        k = !0;
        for (var h of F.mappings)
          P[h] = A, b[h] == null && delete b[h];
      }
    }
    var N = k ? E(E({}, P), b) : a, L = function() {
      var s = N[i], n = u[i];
      try {
        if (n.mappings)
          return 1;
        if (typeof s == "string" || typeof s == "number") {
          if (process.env.NODE_ENV !== "production" && !n.values[s].defaultClass)
            throw new Error();
          p.push(n.values[s].defaultClass);
        } else if (Array.isArray(s))
          for (var y = 0; y < s.length; y++) {
            var w = s[y];
            if (w != null) {
              var K = n.responsiveArray[y];
              if (process.env.NODE_ENV !== "production" && !n.values[w].conditions[K])
                throw new Error();
              p.push(n.values[w].conditions[K]);
            }
          }
        else
          for (var g in s) {
            var O = s[g];
            if (O != null) {
              if (process.env.NODE_ENV !== "production" && !n.values[O].conditions[g])
                throw new Error();
              p.push(n.values[O].conditions[g]);
            }
          }
      } catch (U) {
        if (process.env.NODE_ENV !== "production") {
          class c extends Error {
            constructor(j) {
              super(j), this.name = "SprinklesError";
            }
          }
          var l = (f) => typeof f == "string" ? '"'.concat(f, '"') : f, d = (f, j, V) => {
            throw new c('"'.concat(f, '" has no value ').concat(l(j), ". Possible values are ").concat(Object.keys(V).map(l).join(", ")));
          };
          if (!n)
            throw new c('"'.concat(i, '" is not a valid sprinkle'));
          if ((typeof s == "string" || typeof s == "number") && (s in n.values || d(i, s, n.values), !n.values[s].defaultClass))
            throw new c('"'.concat(i, '" has no default condition. You must specify which conditions to target explicitly. Possible options are ').concat(Object.keys(n.values[s].conditions).map(l).join(", ")));
          if (typeof s == "object") {
            if (!("conditions" in n.values[Object.keys(n.values)[0]]))
              throw new c('"'.concat(i, '" is not a conditional property'));
            if (Array.isArray(s)) {
              if (!("responsiveArray" in n))
                throw new c('"'.concat(i, '" does not support responsive arrays'));
              var C = n.responsiveArray.length;
              if (C < s.length)
                throw new c('"'.concat(i, '" only supports up to ').concat(C, " breakpoints. You passed ").concat(s.length));
              for (var B of s)
                n.values[B] || d(i, B, n.values);
            } else
              for (var m in s) {
                var v = s[m];
                if (v != null && (n.values[v] || d(i, v, n.values), !n.values[v].conditions[m]))
                  throw new c('"'.concat(i, '" has no condition named ').concat(l(m), ". Possible values are ").concat(Object.keys(n.values[v].conditions).map(l).join(", ")));
              }
          }
        }
        throw U;
      }
    };
    for (var i in N)
      L();
    return r(p.join(" "));
  };
  return Object.assign(x, {
    properties: new Set(S)
  });
}, J = (r) => r, I = function() {
  return H(J)(...arguments);
};
function _(r, e, t) {
  t !== void 0 && (r[e] = t);
}
const Y = ["2xs"];
function M() {
  const r = {
    base: {}
  };
  for (const [e, t] of Object.entries(T))
    Y.includes(e) || (r[e] = { "@media": `(min-width: ${t})` });
  return r;
}
M();
const rr = [
  "base",
  ...Object.keys(T).filter(
    (r) => !Y.includes(
      r
    )
  )
], Q = Object.keys(X), er = [...Q, "auto"];
export {
  rr as B,
  er as M,
  Q as S,
  _ as a,
  I as c
};
