// @angi/one-design-system/dist/components/es/_chunks/useCheckboxGroupContext-R2g3qemh.js
import x, { useState as y, useRef as C, createContext as m, forwardRef as h, useContext as D } from "react";
import { jsx as I } from "react/jsx-runtime";
import { b as P, $ as V, h as M, f as B, a as O, i as _ } from "./utils-CjtkL41C.js";
import { $ as k, a as q, b as w, c as E, d as G, e as S } from "./Form-sh8c7lqF.js";
import { b as W } from "./Label-DLRS1VtT.js";
import { $ as p } from "./filterDOMProps-Cxz9cAAv.js";
import { $ as F } from "./useControlledState-3TGZUXux.js";
import { i as N } from "./useFocusWithin-CAx2NHfT.js";
const T = /* @__PURE__ */ new WeakMap();
function j(e, a) {
  let { isDisabled: d, name: n, form: o, validationBehavior: r = "aria" } = e, { isInvalid: s, validationErrors: l, validationDetails: c } = a.displayValidation, { labelProps: u, fieldProps: b, descriptionProps: i, errorMessageProps: t } = k({
    ...e,
    // Checkbox group is not an HTML input element so it
    // shouldn't be labeled by a <label> element.
    labelElementType: "span",
    isInvalid: s,
    errorMessage: e.errorMessage || l
  });
  T.set(a, {
    name: n,
    form: o,
    descriptionId: i.id,
    errorMessageId: t.id,
    validationBehavior: r
  });
  let f = p(e, {
    labelable: !0
  }), { focusWithinProps: $ } = N({
    onBlurWithin: e.onBlur,
    onFocusWithin: e.onFocus,
    onFocusWithinChange: e.onFocusChange
  });
  return {
    groupProps: P(f, {
      role: "group",
      "aria-disabled": d || void 0,
      ...b,
      ...$
    }),
    labelProps: u,
    descriptionProps: i,
    errorMessageProps: t,
    isInvalid: s,
    validationErrors: l,
    validationDetails: c
  };
}
function z(e = {}) {
  let [a, d] = F(e.value, e.defaultValue || [], e.onChange), [n] = y(a), o = !!e.isRequired && a.length === 0, r = C(/* @__PURE__ */ new Map()), s = q({
    ...e,
    value: a
  }), l = s.displayValidation.isInvalid;
  var c, u;
  return {
    ...s,
    value: a,
    defaultValue: (c = e.defaultValue) !== null && c !== void 0 ? c : n,
    setValue(i) {
      e.isReadOnly || e.isDisabled || d(i);
    },
    isDisabled: e.isDisabled || !1,
    isReadOnly: e.isReadOnly || !1,
    isSelected(i) {
      return a.includes(i);
    },
    addValue(i) {
      e.isReadOnly || e.isDisabled || d((t) => t.includes(i) ? t : t.concat(i));
    },
    removeValue(i) {
      e.isReadOnly || e.isDisabled || a.includes(i) && d(a.filter((t) => t !== i));
    },
    toggleValue(i) {
      e.isReadOnly || e.isDisabled || (a.includes(i) ? d(a.filter((t) => t !== i)) : d(a.concat(i)));
    },
    setInvalid(i, t) {
      let f = new Map(r.current);
      t.isInvalid ? f.set(i, t) : f.delete(i), r.current = f, s.updateValidation(w(...f.values()));
    },
    validationState: (u = e.validationState) !== null && u !== void 0 ? u : l ? "invalid" : null,
    isInvalid: l,
    isRequired: o
  };
}
const A = /* @__PURE__ */ m(null), R = /* @__PURE__ */ m(null), H = /* @__PURE__ */ h(function(a, d) {
  [a, d] = V(a, d, A);
  let { validationBehavior: n } = M(E) || {};
  var o, r;
  let s = (r = (o = a.validationBehavior) !== null && o !== void 0 ? o : n) !== null && r !== void 0 ? r : "native", l = z({
    ...a,
    validationBehavior: s
  }), [c, u] = B(!a["aria-label"] && !a["aria-labelledby"]), { groupProps: b, labelProps: i, descriptionProps: t, errorMessageProps: f, ...$ } = j({
    ...a,
    label: u,
    validationBehavior: s
  }, l), v = O({
    ...a,
    values: {
      isDisabled: l.isDisabled,
      isReadOnly: l.isReadOnly,
      isRequired: a.isRequired || !1,
      isInvalid: l.isInvalid,
      state: l
    },
    defaultClassName: "react-aria-CheckboxGroup"
  }), g = p(a, {
    global: !0
  });
  return /* @__PURE__ */ x.createElement("div", {
    ...P(g, v, b),
    ref: d,
    slot: a.slot || void 0,
    "data-readonly": l.isReadOnly || void 0,
    "data-required": a.isRequired || void 0,
    "data-invalid": l.isInvalid || void 0,
    "data-disabled": a.isDisabled || void 0
  }, /* @__PURE__ */ x.createElement(_, {
    values: [
      [
        R,
        l
      ],
      [
        W,
        {
          ...i,
          ref: c,
          elementType: "span"
        }
      ],
      [
        G,
        {
          slots: {
            description: t,
            errorMessage: f
          }
        }
      ],
      [
        S,
        $
      ]
    ]
  }, v.children));
}), J = h(
  ({ disabled: e, invalid: a, required: d, readOnly: n, ...o }, r) => /* @__PURE__ */ I(
    H,
    {
      ref: r,
      ...e !== void 0 && { isDisabled: e },
      ...a !== void 0 && { isInvalid: a },
      ...d !== void 0 && { isRequired: d },
      ...n !== void 0 && { isReadOnly: n },
      ...o
    }
  )
);
J.displayName = "CheckboxGroup";
function ae() {
  const e = D(R);
  if (!e)
    throw new Error(
      "useCheckboxGroupContext must be used within a CheckboxGroup"
    );
  return e;
}
export {
  R as $,
  J as C,
  T as a,
  ae as u
};
