// @angi/one-design-system/dist/components/es/_chunks/useFocusVisible-ROQSYc3m.js
import { j as a, a as c, k as E } from "./useFocusWithin-CAx2NHfT.js";
import { a as m, c as y, d as k } from "./openLink-EIDcTMUl.js";
import { useEffect as M } from "react";
function H(e) {
  return e.pointerType === "" && e.isTrusted ? !0 : m() && e.pointerType ? e.type === "click" && e.buttons === 1 : e.detail === 0 && !e.pointerType;
}
function _(e) {
  return !m() && e.width === 0 && e.height === 0 || e.width === 1 && e.height === 1 && e.pressure === 0 && e.detail === 0 && e.pointerType === "mouse";
}
let d = null, l = /* @__PURE__ */ new Set(), f = /* @__PURE__ */ new Map(), i = !1, v = !1;
const O = {
  Tab: !0,
  Escape: !0
};
function s(e, r) {
  for (let t of l) t(e, r);
}
function I(e) {
  return !(e.metaKey || !k() && e.altKey || e.ctrlKey || e.key === "Control" || e.key === "Shift" || e.key === "Meta");
}
function u(e) {
  i = !0, !y.isOpening && I(e) && (d = "keyboard", s("keyboard", e));
}
function o(e) {
  d = "pointer", "pointerType" in e && e.pointerType, (e.type === "mousedown" || e.type === "pointerdown") && (i = !0, s("pointer", e));
}
function L(e) {
  !y.isOpening && H(e) && (i = !0, d = "virtual");
}
function w(e) {
  e.target === window || e.target === document || E || !e.isTrusted || (!i && !v && (d = "virtual", s("virtual", e)), i = !1, v = !1);
}
function T() {
  E || (i = !1, v = !0);
}
function p(e) {
  if (typeof window > "u" || typeof document > "u" || f.get(a(e))) return;
  const r = a(e), t = c(e);
  let n = r.HTMLElement.prototype.focus;
  r.HTMLElement.prototype.focus = function() {
    i = !0, n.apply(this, arguments);
  }, t.addEventListener("keydown", u, !0), t.addEventListener("keyup", u, !0), t.addEventListener("click", L, !0), r.addEventListener("focus", w, !0), r.addEventListener("blur", T, !1), typeof PointerEvent < "u" ? (t.addEventListener("pointerdown", o, !0), t.addEventListener("pointermove", o, !0), t.addEventListener("pointerup", o, !0)) : process.env.NODE_ENV === "test" && (t.addEventListener("mousedown", o, !0), t.addEventListener("mousemove", o, !0), t.addEventListener("mouseup", o, !0)), r.addEventListener("beforeunload", () => {
    g(e);
  }, {
    once: !0
  }), f.set(r, {
    focus: n
  });
}
const g = (e, r) => {
  const t = a(e), n = c(e);
  r && n.removeEventListener("DOMContentLoaded", r), f.has(t) && (t.HTMLElement.prototype.focus = f.get(t).focus, n.removeEventListener("keydown", u, !0), n.removeEventListener("keyup", u, !0), n.removeEventListener("click", L, !0), t.removeEventListener("focus", w, !0), t.removeEventListener("blur", T, !1), typeof PointerEvent < "u" ? (n.removeEventListener("pointerdown", o, !0), n.removeEventListener("pointermove", o, !0), n.removeEventListener("pointerup", o, !0)) : process.env.NODE_ENV === "test" && (n.removeEventListener("mousedown", o, !0), n.removeEventListener("mousemove", o, !0), n.removeEventListener("mouseup", o, !0)), f.delete(t));
};
function K(e) {
  const r = c(e);
  let t;
  return r.readyState !== "loading" ? p(e) : (t = () => {
    p(e);
  }, r.addEventListener("DOMContentLoaded", t)), () => g(e, t);
}
typeof document < "u" && K();
function C() {
  return d !== "pointer";
}
function B() {
  return d;
}
function P(e) {
  d = e, s(e, null);
}
const S = /* @__PURE__ */ new Set([
  "checkbox",
  "radio",
  "range",
  "color",
  "file",
  "image",
  "button",
  "submit",
  "reset"
]);
function j(e, r, t) {
  let n = c(t?.target);
  const $ = typeof window < "u" ? a(t?.target).HTMLInputElement : HTMLInputElement, b = typeof window < "u" ? a(t?.target).HTMLTextAreaElement : HTMLTextAreaElement, h = typeof window < "u" ? a(t?.target).HTMLElement : HTMLElement, x = typeof window < "u" ? a(t?.target).KeyboardEvent : KeyboardEvent;
  return e = e || n.activeElement instanceof $ && !S.has(n.activeElement.type) || n.activeElement instanceof b || n.activeElement instanceof h && n.activeElement.isContentEditable, !(e && r === "keyboard" && t instanceof x && !O[t.key]);
}
function V(e, r, t) {
  p(), M(() => {
    let n = ($, b) => {
      j(!!t?.isTextInput, $, b) && e(C());
    };
    return l.add(n), () => {
      l.delete(n);
    };
  }, r);
}
export {
  P as $,
  B as a,
  C as b,
  V as c,
  _ as d,
  H as e
};
