// @angi/one-design-system/dist/components/es/_chunks/useFocusWithin-CAx2NHfT.js
import { $ as L } from "./focusWithoutScrolling-CDw3JYiq.js";
import { c as S } from "./utils-CjtkL41C.js";
import { useRef as $, useCallback as f, useEffect as P } from "react";
const T = (e) => {
  var t;
  return (t = e?.ownerDocument) !== null && t !== void 0 ? t : document;
}, y = (e) => e && "window" in e && e.window === e ? e : T(e).defaultView || window;
function W(e) {
  return e !== null && typeof e == "object" && "nodeType" in e && typeof e.nodeType == "number";
}
function D(e) {
  return W(e) && e.nodeType === Node.DOCUMENT_FRAGMENT_NODE && "host" in e;
}
let G = !1;
function v() {
  return G;
}
function O(e, t) {
  if (!v()) return t && e ? e.contains(t) : !1;
  if (!e || !t) return !1;
  let n = t;
  for (; n !== null; ) {
    if (n === e) return !0;
    n.tagName === "SLOT" && n.assignedSlot ? n = n.assignedSlot.parentNode : D(n) ? n = n.host : n = n.parentNode;
  }
  return !1;
}
const V = (e = document) => {
  var t;
  if (!v()) return e.activeElement;
  let n = e.activeElement;
  for (; n && "shadowRoot" in n && (!((t = n.shadowRoot) === null || t === void 0) && t.activeElement); ) n = n.shadowRoot.activeElement;
  return n;
};
function N(e) {
  return v() && e.target.shadowRoot && e.composedPath ? e.composedPath()[0] : e.target;
}
function B() {
  let e = $(/* @__PURE__ */ new Map()), t = f((i, o, a, s) => {
    let c = s?.once ? (...l) => {
      e.current.delete(a), a(...l);
    } : a;
    e.current.set(a, {
      type: o,
      eventTarget: i,
      fn: c,
      options: s
    }), i.addEventListener(o, c, s);
  }, []), n = f((i, o, a, s) => {
    var c;
    let l = ((c = e.current.get(a)) === null || c === void 0 ? void 0 : c.fn) || a;
    i.removeEventListener(o, l, s), e.current.delete(a);
  }, []), r = f(() => {
    e.current.forEach((i, o) => {
      n(i.eventTarget, i.type, o, i.options);
    });
  }, [
    n
  ]);
  return P(() => r, [
    r
  ]), {
    addGlobalListener: t,
    removeGlobalListener: n,
    removeAllGlobalListeners: r
  };
}
const C = typeof Element < "u" && "checkVisibility" in Element.prototype;
function R(e) {
  const t = y(e);
  if (!(e instanceof t.HTMLElement) && !(e instanceof t.SVGElement)) return !1;
  let { display: n, visibility: r } = e.style, i = n !== "none" && r !== "hidden" && r !== "collapse";
  if (i) {
    const { getComputedStyle: o } = e.ownerDocument.defaultView;
    let { display: a, visibility: s } = o(e);
    i = a !== "none" && s !== "hidden" && s !== "collapse";
  }
  return i;
}
function j(e, t) {
  return !e.hasAttribute("hidden") && // Ignore HiddenSelect when tree walking.
  !e.hasAttribute("data-react-aria-prevent-focus") && (e.nodeName === "DETAILS" && t && t.nodeName !== "SUMMARY" ? e.hasAttribute("open") : !0);
}
function p(e, t) {
  return C ? e.checkVisibility({
    visibilityProperty: !0
  }) && !e.closest("[data-react-aria-prevent-focus]") : e.nodeName !== "#comment" && R(e) && j(e, t) && (!e.parentElement || p(e.parentElement, e));
}
const m = [
  "input:not([disabled]):not([type=hidden])",
  "select:not([disabled])",
  "textarea:not([disabled])",
  "button:not([disabled])",
  "a[href]",
  "area[href]",
  "summary",
  "iframe",
  "object",
  "embed",
  "audio[controls]",
  "video[controls]",
  '[contenteditable]:not([contenteditable^="false"])',
  "permission"
], I = m.join(":not([hidden]),") + ",[tabindex]:not([disabled]):not([hidden])";
m.push('[tabindex]:not([tabindex="-1"]):not([disabled])');
const _ = m.join(':not([hidden]):not([tabindex="-1"]),');
function H(e) {
  return e.matches(I) && p(e) && !F(e);
}
function J(e) {
  return e.matches(_) && p(e) && !F(e);
}
function F(e) {
  let t = e;
  for (; t != null; ) {
    if (t instanceof t.ownerDocument.defaultView.HTMLElement && t.inert) return !0;
    t = t.parentElement;
  }
  return !1;
}
function x(e) {
  let t = e;
  return t.nativeEvent = e, t.isDefaultPrevented = () => t.defaultPrevented, t.isPropagationStopped = () => t.cancelBubble, t.persist = () => {
  }, t;
}
function k(e, t) {
  Object.defineProperty(e, "target", {
    value: t
  }), Object.defineProperty(e, "currentTarget", {
    value: t
  });
}
function U(e) {
  let t = $({
    isFocused: !1,
    observer: null
  });
  return S(() => {
    const n = t.current;
    return () => {
      n.observer && (n.observer.disconnect(), n.observer = null);
    };
  }, []), f((n) => {
    if (n.target instanceof HTMLButtonElement || n.target instanceof HTMLInputElement || n.target instanceof HTMLTextAreaElement || n.target instanceof HTMLSelectElement) {
      t.current.isFocused = !0;
      let r = n.target, i = (o) => {
        if (t.current.isFocused = !1, r.disabled) {
          let a = x(o);
          e?.(a);
        }
        t.current.observer && (t.current.observer.disconnect(), t.current.observer = null);
      };
      r.addEventListener("focusout", i, {
        once: !0
      }), t.current.observer = new MutationObserver(() => {
        if (t.current.isFocused && r.disabled) {
          var o;
          (o = t.current.observer) === null || o === void 0 || o.disconnect();
          let a = r === document.activeElement ? null : document.activeElement;
          r.dispatchEvent(new FocusEvent("blur", {
            relatedTarget: a
          })), r.dispatchEvent(new FocusEvent("focusout", {
            bubbles: !0,
            relatedTarget: a
          }));
        }
      }), t.current.observer.observe(r, {
        attributes: !0,
        attributeFilter: [
          "disabled"
        ]
      });
    }
  }, [
    e
  ]);
}
let w = !1;
function K(e) {
  for (; e && !H(e); ) e = e.parentElement;
  let t = y(e), n = t.document.activeElement;
  if (!n || n === e) return;
  w = !0;
  let r = !1, i = (u) => {
    (u.target === n || r) && u.stopImmediatePropagation();
  }, o = (u) => {
    (u.target === n || r) && (u.stopImmediatePropagation(), !e && !r && (r = !0, L(n), c()));
  }, a = (u) => {
    (u.target === e || r) && u.stopImmediatePropagation();
  }, s = (u) => {
    (u.target === e || r) && (u.stopImmediatePropagation(), r || (r = !0, L(n), c()));
  };
  t.addEventListener("blur", i, !0), t.addEventListener("focusout", o, !0), t.addEventListener("focusin", s, !0), t.addEventListener("focus", a, !0);
  let c = () => {
    cancelAnimationFrame(l), t.removeEventListener("blur", i, !0), t.removeEventListener("focusout", o, !0), t.removeEventListener("focusin", s, !0), t.removeEventListener("focus", a, !0), w = !1, r = !1;
  }, l = requestAnimationFrame(c);
  return c;
}
function Q(e) {
  let { isDisabled: t, onBlurWithin: n, onFocusWithin: r, onFocusWithinChange: i } = e, o = $({
    isFocusWithin: !1
  }), { addGlobalListener: a, removeAllGlobalListeners: s } = B(), c = f((d) => {
    d.currentTarget.contains(d.target) && o.current.isFocusWithin && !d.currentTarget.contains(d.relatedTarget) && (o.current.isFocusWithin = !1, s(), n && n(d), i && i(!1));
  }, [
    n,
    i,
    o,
    s
  ]), l = U(c), u = f((d) => {
    if (!d.currentTarget.contains(d.target)) return;
    const b = T(d.target), A = V(b);
    if (!o.current.isFocusWithin && A === N(d.nativeEvent)) {
      r && r(d), i && i(!0), o.current.isFocusWithin = !0, l(d);
      let E = d.currentTarget;
      a(b, "focus", (h) => {
        if (o.current.isFocusWithin && !O(E, h.target)) {
          let g = new b.defaultView.FocusEvent("blur", {
            relatedTarget: h.target
          });
          k(g, E);
          let M = x(g);
          c(M);
        }
      }, {
        capture: !0
      });
    }
  }, [
    r,
    i,
    l,
    a,
    c
  ]);
  return t ? {
    focusWithinProps: {
      // These cannot be null, that would conflict in mergeProps
      onFocus: void 0,
      onBlur: void 0
    }
  } : {
    focusWithinProps: {
      onFocus: u,
      onBlur: c
    }
  };
}
export {
  B as $,
  T as a,
  O as b,
  v as c,
  J as d,
  H as e,
  V as f,
  N as g,
  U as h,
  Q as i,
  y as j,
  w as k,
  k as l,
  x as m,
  K as n
};
