// @angi/one-design-system/dist/components/es/_chunks/useFormValidation-CgQ_8wCr.js
import { $ as v } from "./useFocusable-CrxUcjB4.js";
import { useEffect as f, useRef as $ } from "react";
import { $ as g } from "./useFocusVisible-ROQSYc3m.js";
import { c as p } from "./utils-CjtkL41C.js";
function w(n, e, i) {
  let d = v(() => {
    i && i(e);
  });
  f(() => {
    var l;
    let o = n == null || (l = n.current) === null || l === void 0 ? void 0 : l.form;
    return o?.addEventListener("reset", d), () => {
      o?.removeEventListener("reset", d);
    };
  }, [
    n
  ]);
}
function L(n, e, i) {
  let { validationBehavior: d, focus: l } = n;
  p(() => {
    if (d === "native" && i?.current && !i.current.disabled) {
      let a = e.realtimeValidation.isInvalid ? e.realtimeValidation.validationErrors.join(" ") || "Invalid value." : "";
      i.current.setCustomValidity(a), i.current.hasAttribute("title") || (i.current.title = ""), e.realtimeValidation.isInvalid || e.updateValidation(h(i.current));
    }
  });
  let o = $(!1), u = v(() => {
    o.current || e.resetValidation();
  }), c = v((a) => {
    var t;
    e.displayValidation.isInvalid || e.commitValidation();
    let r = i == null || (t = i.current) === null || t === void 0 ? void 0 : t.form;
    if (!a.defaultPrevented && i && r && E(r) === i.current) {
      var s;
      l ? l() : (s = i.current) === null || s === void 0 || s.focus(), g("keyboard");
    }
    a.preventDefault();
  }), m = v(() => {
    e.commitValidation();
  });
  f(() => {
    let a = i?.current;
    if (!a) return;
    let t = a.form, r = t?.reset;
    return t && (t.reset = () => {
      o.current = !window.event || window.event.type === "message" && window.event.target instanceof MessagePort, r?.call(t), o.current = !1;
    }), a.addEventListener("invalid", c), a.addEventListener("change", m), t?.addEventListener("reset", u), () => {
      a.removeEventListener("invalid", c), a.removeEventListener("change", m), t?.removeEventListener("reset", u), t && (t.reset = r);
    };
  }, [
    i,
    d
  ]);
}
function b(n) {
  let e = n.validity;
  return {
    badInput: e.badInput,
    customError: e.customError,
    patternMismatch: e.patternMismatch,
    rangeOverflow: e.rangeOverflow,
    rangeUnderflow: e.rangeUnderflow,
    stepMismatch: e.stepMismatch,
    tooLong: e.tooLong,
    tooShort: e.tooShort,
    typeMismatch: e.typeMismatch,
    valueMissing: e.valueMissing,
    valid: e.valid
  };
}
function h(n) {
  return {
    isInvalid: !n.validity.valid,
    validationDetails: b(n),
    validationErrors: n.validationMessage ? [
      n.validationMessage
    ] : []
  };
}
function E(n) {
  for (let e = 0; e < n.elements.length; e++) {
    let i = n.elements[e];
    if (!i.validity.valid) return i;
  }
  return null;
}
export {
  w as $,
  L as a
};
