// @angi/one-design-system/dist/components/es/_chunks/useFocusRing-CdwnmTWa.js
import { h as x, a as h, f as l, g, i as m } from "./useFocusWithin-CAx2NHfT.js";
import { useCallback as f, useRef as P, useState as $ } from "react";
import { b as V, c as C } from "./useFocusVisible-ROQSYc3m.js";
function D(d) {
  let { isDisabled: i, onFocus: o, onBlur: t, onFocusChange: e } = d;
  const u = f((s) => {
    if (s.target === s.currentTarget)
      return t && t(s), e && e(!1), !0;
  }, [
    t,
    e
  ]), r = x(u), b = f((s) => {
    const c = h(s.target), n = c ? l(c) : l();
    s.target === s.currentTarget && n === g(s.nativeEvent) && (o && o(s), e && e(!0), r(s));
  }, [
    e,
    o,
    r
  ]);
  return {
    focusProps: {
      onFocus: !i && (o || e || t) ? b : void 0,
      onBlur: !i && (t || e) ? u : void 0
    }
  };
}
function v(d = {}) {
  let { autoFocus: i = !1, isTextInput: o, within: t } = d, e = P({
    isFocused: !1,
    isFocusVisible: i || V()
  }), [u, r] = $(!1), [b, s] = $(() => e.current.isFocused && e.current.isFocusVisible), c = f(() => s(e.current.isFocused && e.current.isFocusVisible), []), n = f((a) => {
    e.current.isFocused = a, r(a), c();
  }, [
    c
  ]);
  C((a) => {
    e.current.isFocusVisible = a, c();
  }, [], {
    isTextInput: o
  });
  let { focusProps: p } = D({
    isDisabled: t,
    onFocusChange: n
  }), { focusWithinProps: F } = m({
    isDisabled: !t,
    onFocusWithinChange: n
  });
  return {
    isFocused: u,
    isFocusVisible: b,
    focusProps: t ? F : p
  };
}
export {
  v as $,
  D as a
};
