// @angi/one-design-system/dist/components/es/_chunks/useFocusable-CrxUcjB4.js
import { c as i, b as c } from "./utils-CjtkL41C.js";
import s, { useRef as d, useCallback as p, useEffect as P, useContext as v } from "react";
import { $ as x } from "./focusSafely-OgqIqadr.js";
import { a as D } from "./useFocusRing-CdwnmTWa.js";
var a;
const m = (a = s.useInsertionEffect) !== null && a !== void 0 ? a : i;
function I(e) {
  const t = d(null);
  return m(() => {
    t.current = e;
  }, [
    e
  ]), p((...r) => {
    const o = t.current;
    return o?.(...r);
  }, []);
}
function y(e, t) {
  i(() => {
    if (e && e.ref && t)
      return e.ref.current = t.current, () => {
        e.ref && (e.ref.current = null);
      };
  });
}
function u(e) {
  if (!e) return;
  let t = !0;
  return (r) => {
    let o = {
      ...r,
      preventDefault() {
        r.preventDefault();
      },
      isDefaultPrevented() {
        return r.isDefaultPrevented();
      },
      stopPropagation() {
        t && process.env.NODE_ENV !== "production" ? console.error("stopPropagation is now the default behavior for events in React Spectrum. You can use continuePropagation() to revert this behavior.") : t = !0;
      },
      continuePropagation() {
        t = !1;
      },
      isPropagationStopped() {
        return t;
      }
    };
    e(o), t && r.stopPropagation();
  };
}
function g(e) {
  return {
    keyboardProps: e.isDisabled ? {} : {
      onKeyDown: u(e.onKeyDown),
      onKeyUp: u(e.onKeyUp)
    }
  };
}
let E = /* @__PURE__ */ s.createContext(null);
function _(e) {
  let t = v(E) || {};
  y(t, e);
  let { ref: r, ...o } = t;
  return o;
}
function K(e, t) {
  let { focusProps: r } = D(e), { keyboardProps: o } = g(e), $ = c(r, o), l = _(t), b = e.isDisabled ? {} : l, n = d(e.autoFocus);
  P(() => {
    n.current && t.current && x(t.current), n.current = !1;
  }, [
    t
  ]);
  let f = e.excludeFromTabOrder ? -1 : 0;
  return e.isDisabled && (f = void 0), {
    focusableProps: c({
      ...$,
      tabIndex: f
    }, b)
  };
}
export {
  I as $,
  K as a,
  y as b
};
