// @angi/one-design-system/dist/components/es/_chunks/useRadioGroupContext-CuaGc53C.js
import h, { useMemo as G, useState as y, useRef as S, createContext as V, forwardRef as D, useContext as q } from "react";
import { jsx as F } from "react/jsx-runtime";
import { d as O, b as I, $ as k, h as N, f as T, a as W, i as L } from "./utils-CjtkL41C.js";
import { $ as A, a as j, c as K, d as H, e as U } from "./Form-sh8c7lqF.js";
import { b as z } from "./Label-DLRS1VtT.js";
import "react-dom";
import { $ as M } from "./filterDOMProps-Cxz9cAAv.js";
import { $ as J } from "./useControlledState-3TGZUXux.js";
import { i as Q, j as X } from "./useFocusWithin-CAx2NHfT.js";
import { $ as Y } from "./FocusScope-DlOb-a2X.js";
import { $ as Z } from "./context-qNvgpQEb.js";
const ee = /* @__PURE__ */ new WeakMap();
function ae(e, a) {
  let { name: r, form: l, isReadOnly: d, isRequired: o, isDisabled: u, orientation: i = "vertical", validationBehavior: c = "aria" } = e, { direction: $ } = Z(), { isInvalid: b, validationErrors: s, validationDetails: m } = a.displayValidation, { labelProps: R, fieldProps: P, descriptionProps: p, errorMessageProps: x } = A({
    ...e,
    // Radio group is not an HTML input element so it
    // shouldn't be labeled by a <label> element.
    labelElementType: "span",
    isInvalid: a.isInvalid,
    errorMessage: e.errorMessage || s
  }), B = M(e, {
    labelable: !0
  }), { focusWithinProps: E } = Q({
    onBlurWithin(f) {
      var t;
      (t = e.onBlur) === null || t === void 0 || t.call(e, f), a.selectedValue || a.setLastFocusedValue(null);
    },
    onFocusWithin: e.onFocus,
    onFocusWithinChange: e.onFocusChange
  }), C = (f) => {
    let t;
    switch (f.key) {
      case "ArrowRight":
        $ === "rtl" && i !== "vertical" ? t = "prev" : t = "next";
        break;
      case "ArrowLeft":
        $ === "rtl" && i !== "vertical" ? t = "next" : t = "prev";
        break;
      case "ArrowDown":
        t = "next";
        break;
      case "ArrowUp":
        t = "prev";
        break;
      default:
        return;
    }
    f.preventDefault();
    let v = Y(f.currentTarget, {
      from: f.target,
      accept: (g) => g instanceof X(g).HTMLInputElement && g.type === "radio"
    }), n;
    t === "next" ? (n = v.nextNode(), n || (v.currentNode = f.currentTarget, n = v.firstChild())) : (n = v.previousNode(), n || (v.currentNode = f.currentTarget, n = v.lastChild())), n && (n.focus(), a.setSelectedValue(n.value));
  }, _ = O(r);
  return ee.set(a, {
    name: _,
    form: l,
    descriptionId: p.id,
    errorMessageId: x.id,
    validationBehavior: c
  }), {
    radioGroupProps: I(B, {
      // https://www.w3.org/TR/wai-aria-1.2/#radiogroup
      role: "radiogroup",
      onKeyDown: C,
      "aria-invalid": a.isInvalid || void 0,
      "aria-errormessage": e["aria-errormessage"],
      "aria-readonly": d || void 0,
      "aria-required": o || void 0,
      "aria-disabled": u || void 0,
      "aria-orientation": i,
      ...P,
      ...E
    }),
    labelProps: R,
    descriptionProps: p,
    errorMessageProps: x,
    isInvalid: b,
    validationErrors: s,
    validationDetails: m
  };
}
let ie = Math.round(Math.random() * 1e10), re = 0;
function te(e) {
  let a = G(() => e.name || `radio-group-${ie}-${++re}`, [
    e.name
  ]);
  var r;
  let [l, d] = J(e.value, (r = e.defaultValue) !== null && r !== void 0 ? r : null, e.onChange), [o] = y(l), [u, i] = y(null), c = j({
    ...e,
    value: l
  }), $ = (m) => {
    !e.isReadOnly && !e.isDisabled && (d(m), c.commitValidation());
  }, b = c.displayValidation.isInvalid;
  var s;
  return {
    ...c,
    name: a,
    selectedValue: l,
    defaultSelectedValue: e.value !== void 0 ? o : (s = e.defaultValue) !== null && s !== void 0 ? s : null,
    setSelectedValue: $,
    lastFocusedValue: u,
    setLastFocusedValue: i,
    isDisabled: e.isDisabled || !1,
    isReadOnly: e.isReadOnly || !1,
    isRequired: e.isRequired || !1,
    validationState: e.validationState || (b ? "invalid" : null),
    isInvalid: b
  };
}
const le = /* @__PURE__ */ V(null);
function de(e) {
  let a = S({});
  return /* @__PURE__ */ h.createElement(le.Provider, {
    value: a
  }, e.children);
}
const oe = /* @__PURE__ */ V(null), w = /* @__PURE__ */ V(null), ne = /* @__PURE__ */ D(function(a, r) {
  [a, r] = k(a, r, oe);
  let { validationBehavior: l } = N(K) || {};
  var d, o;
  let u = (o = (d = a.validationBehavior) !== null && d !== void 0 ? d : l) !== null && o !== void 0 ? o : "native", i = te({
    ...a,
    validationBehavior: u
  }), [c, $] = T(!a["aria-label"] && !a["aria-labelledby"]), { radioGroupProps: b, labelProps: s, descriptionProps: m, errorMessageProps: R, ...P } = ae({
    ...a,
    label: $,
    validationBehavior: u
  }, i), p = W({
    ...a,
    values: {
      orientation: a.orientation || "vertical",
      isDisabled: i.isDisabled,
      isReadOnly: i.isReadOnly,
      isRequired: i.isRequired,
      isInvalid: i.isInvalid,
      state: i
    },
    defaultClassName: "react-aria-RadioGroup"
  }), x = M(a, {
    global: !0
  });
  return /* @__PURE__ */ h.createElement("div", {
    ...I(x, p, b),
    ref: r,
    slot: a.slot || void 0,
    "data-orientation": a.orientation || "vertical",
    "data-invalid": i.isInvalid || void 0,
    "data-disabled": i.isDisabled || void 0,
    "data-readonly": i.isReadOnly || void 0,
    "data-required": i.isRequired || void 0
  }, /* @__PURE__ */ h.createElement(L, {
    values: [
      [
        w,
        i
      ],
      [
        z,
        {
          ...s,
          ref: c,
          elementType: "span"
        }
      ],
      [
        H,
        {
          slots: {
            description: m,
            errorMessage: R
          }
        }
      ],
      [
        U,
        P
      ]
    ]
  }, /* @__PURE__ */ h.createElement(de, null, p.children)));
}), se = D(
  ({ disabled: e, invalid: a, required: r, readOnly: l, ...d }, o) => /* @__PURE__ */ F(
    ne,
    {
      ref: o,
      ...e !== void 0 && { isDisabled: e },
      ...a !== void 0 && { isInvalid: a },
      ...r !== void 0 && { isRequired: r },
      ...l !== void 0 && { isReadOnly: l },
      ...d
    }
  )
);
se.displayName = "RadioGroup";
function Pe() {
  const e = q(w);
  if (!e)
    throw new Error("useRadioGroupContext must be used within a RadioGroup");
  return e;
}
export {
  w as $,
  se as R,
  ee as a,
  Pe as u
};
