// @angi/one-design-system/dist/components/es/_chunks/useTranslations-Da7GR3Ot.js
import { useMemo as l } from "react";
import { u as p } from "./ConfigProvider-BEJKo3yW.js";
function d(e) {
  return e == null || typeof e != "object" && typeof e != "function";
}
function O(e) {
  return ArrayBuffer.isView(e) && !(e instanceof DataView);
}
function u(e) {
  if (d(e))
    return e;
  if (Array.isArray(e) || O(e) || e instanceof ArrayBuffer || typeof SharedArrayBuffer < "u" && e instanceof SharedArrayBuffer)
    return e.slice(0);
  const n = Object.getPrototypeOf(e);
  if (n == null)
    return Object.assign(Object.create(n), e);
  const r = n.constructor;
  if (e instanceof Date || e instanceof Map || e instanceof Set)
    return new r(e);
  if (e instanceof RegExp) {
    const t = new r(e);
    return t.lastIndex = e.lastIndex, t;
  }
  if (e instanceof DataView)
    return new r(e.buffer.slice(0));
  if (e instanceof Error) {
    let t;
    return e instanceof AggregateError ? t = new r(e.errors, e.message, { cause: e.cause }) : t = new r(e.message, { cause: e.cause }), t.stack = e.stack, Object.assign(t, e), t;
  }
  if (typeof File < "u" && e instanceof File)
    return new r([e], e.name, { type: e.type, lastModified: e.lastModified });
  if (typeof e == "object") {
    const t = Object.create(n);
    return Object.assign(t, e);
  }
  return e;
}
function a(e) {
  if (!e || typeof e != "object")
    return !1;
  const n = Object.getPrototypeOf(e);
  return n === null || n === Object.prototype || Object.getPrototypeOf(n) === null ? Object.prototype.toString.call(e) === "[object Object]" : !1;
}
function A(e) {
  return e === "__proto__";
}
function c(e, n, r) {
  const t = Object.keys(n);
  for (let i = 0; i < t.length; i++) {
    const s = t[i];
    if (A(s))
      continue;
    const f = n[s], o = e[s], y = r(o, f, s, e, n);
    y !== void 0 ? e[s] = y : Array.isArray(f) ? Array.isArray(o) ? e[s] = c(o, f, r) : e[s] = c([], f, r) : a(f) ? a(o) ? e[s] = c(o, f, r) : e[s] = c({}, f, r) : (o === void 0 || f !== void 0) && (e[s] = f);
  }
  return e;
}
function w(e, n) {
  return c(u(e), n, function r(t, i) {
    if (Array.isArray(i))
      return Array.isArray(t) ? c(u(t), i, r) : c([], i, r);
    if (a(i))
      return a(t) ? c(u(t), i, r) : c({}, i, r);
  });
}
function P(e, n) {
  return e.replace(
    /\{(\w+)\}/g,
    (r, t) => String(n[t] ?? "")
  );
}
function h(e) {
  const { locale: n } = p();
  return l(() => {
    const r = n.includes("-") ? n.split("-")[0] : n, t = e[r] ?? {}, i = n.includes("-") && n !== r ? e[n] ?? {} : {};
    return w(
      t,
      i
    );
  }, [e, n]);
}
export {
  P as i,
  h as u
};
