// @angi/one-design-system/dist/components/es/_chunks/utils-CjtkL41C.js
import i, { useState as b, useRef as s, useCallback as x, useContext as m, useEffect as M, useMemo as h } from "react";
import { c as V } from "./clsx-OuTLNxxd.js";
const v = typeof document < "u" ? i.useLayoutEffect : () => {
};
function k(e) {
  let [t, r] = b(e), l = s(t), n = s(null), u = s(() => {
    if (!n.current) return;
    let a = n.current.next();
    if (a.done) {
      n.current = null;
      return;
    }
    l.current === a.value ? u.current() : r(a.value);
  });
  v(() => {
    l.current = t, n.current && u.current();
  });
  let f = x((a) => {
    n.current = a(l.current), u.current();
  }, [
    u
  ]);
  return [
    t,
    f
  ];
}
const p = {
  prefix: String(Math.round(Math.random() * 1e10)),
  current: 0
}, w = /* @__PURE__ */ i.createContext(p), I = /* @__PURE__ */ i.createContext(!1);
let N = !!(typeof window < "u" && window.document && window.document.createElement), S = /* @__PURE__ */ new WeakMap();
function j(e = !1) {
  let t = m(w), r = s(null);
  if (r.current === null && !e) {
    var l, n;
    let u = (n = i.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED) === null || n === void 0 || (l = n.ReactCurrentOwner) === null || l === void 0 ? void 0 : l.current;
    if (u) {
      let f = S.get(u);
      f == null ? S.set(u, {
        id: t.current,
        state: u.memoizedState
      }) : u.memoizedState !== f.state && (t.current = f.id, S.delete(u));
    }
    r.current = ++t.current;
  }
  return r.current;
}
function z(e) {
  let t = m(w);
  t === p && !N && process.env.NODE_ENV !== "production" && console.warn("When server rendering, you must wrap your application in an <SSRProvider> to ensure consistent ids are generated between the client and server.");
  let r = j(!!e), l = t === p && process.env.NODE_ENV === "test" ? "react-aria" : `react-aria${t.prefix}`;
  return e || `${l}-${r}`;
}
function P(e) {
  let t = i.useId(), [r] = b(H()), l = r || process.env.NODE_ENV === "test" ? "react-aria" : `react-aria${p.prefix}`;
  return e || `${l}-${t}`;
}
const R = typeof i.useId == "function" ? P : z;
function q() {
  return !1;
}
function E() {
  return !0;
}
function F(e) {
  return () => {
  };
}
function H() {
  return typeof i.useSyncExternalStore == "function" ? i.useSyncExternalStore(F, q, E) : m(I);
}
let K = !!(typeof window < "u" && window.document && window.document.createElement), $ = /* @__PURE__ */ new Map(), y;
typeof FinalizationRegistry < "u" && (y = new FinalizationRegistry((e) => {
  $.delete(e);
}));
function G(e) {
  let [t, r] = b(e), l = s(null), n = R(t), u = s(null);
  if (y && y.register(u, n), K) {
    const f = $.get(n);
    f && !f.includes(l) ? f.push(l) : $.set(n, [
      l
    ]);
  }
  return v(() => {
    let f = n;
    return () => {
      y && y.unregister(u), $.delete(f);
    };
  }, [
    n
  ]), M(() => {
    let f = l.current;
    return f && r(f), () => {
      f && (l.current = null);
    };
  }), n;
}
function J(e, t) {
  if (e === t) return e;
  let r = $.get(e);
  if (r)
    return r.forEach((n) => n.current = t), t;
  let l = $.get(t);
  return l ? (l.forEach((n) => n.current = e), e) : t;
}
function Y(e = []) {
  let t = G(), [r, l] = k(t), n = x(() => {
    l(function* () {
      yield t, yield document.getElementById(t) ? t : void 0;
    });
  }, [
    t,
    l
  ]);
  return v(n, [
    t,
    n,
    ...e
  ]), r;
}
function Q(...e) {
  return (...t) => {
    for (let r of e) typeof r == "function" && r(...t);
  };
}
function X(...e) {
  let t = {
    ...e[0]
  };
  for (let r = 1; r < e.length; r++) {
    let l = e[r];
    for (let n in l) {
      let u = t[n], f = l[n];
      typeof u == "function" && typeof f == "function" && // This is a lot faster than a regex.
      n[0] === "o" && n[1] === "n" && n.charCodeAt(2) >= /* 'A' */
      65 && n.charCodeAt(2) <= /* 'Z' */
      90 ? t[n] = Q(u, f) : (n === "className" || n === "UNSAFE_className") && typeof u == "string" && typeof f == "string" ? t[n] = V(u, f) : n === "id" && u && f ? t.id = J(u, f) : t[n] = f !== void 0 ? f : u;
    }
  }
  return t;
}
function Z(...e) {
  return e.length === 1 && e[0] ? e[0] : (t) => {
    let r = !1;
    const l = e.map((n) => {
      const u = g(n, t);
      return r || (r = typeof u == "function"), u;
    });
    if (r) return () => {
      l.forEach((n, u) => {
        typeof n == "function" ? n() : g(e[u], null);
      });
    };
  };
}
function g(e, t) {
  if (typeof e == "function") return e(t);
  e != null && (e.current = t);
}
function D(e) {
  const t = s(null), r = s(void 0), l = x((n) => {
    if (typeof e == "function") {
      const u = e, f = u(n);
      return () => {
        typeof f == "function" ? f() : u(null);
      };
    } else if (e)
      return e.current = n, () => {
        e.current = null;
      };
  }, [
    e
  ]);
  return h(() => ({
    get current() {
      return t.current;
    },
    set current(n) {
      t.current = n, r.current && (r.current(), r.current = void 0), n != null && (r.current = l(n));
    }
  }), [
    l
  ]);
}
const W = Symbol("default");
function L({ values: e, children: t }) {
  for (let [r, l] of e)
    t = /* @__PURE__ */ i.createElement(r.Provider, {
      value: l
    }, t);
  return t;
}
function O(e) {
  let { className: t, style: r, children: l, defaultClassName: n, defaultChildren: u, defaultStyle: f, values: a } = e;
  return h(() => {
    let c, d, o;
    return typeof t == "function" ? c = t({
      ...a,
      defaultClassName: n
    }) : c = t, typeof r == "function" ? d = r({
      ...a,
      defaultStyle: f || {}
    }) : d = r, typeof l == "function" ? o = l({
      ...a,
      defaultChildren: u
    }) : l == null ? o = u : o = l, {
      className: c ?? n,
      style: d || f ? {
        ...f,
        ...d
      } : void 0,
      children: o ?? u,
      "data-rac": ""
    };
  }, [
    t,
    r,
    l,
    n,
    u,
    f,
    a
  ]);
}
function A(e, t) {
  let r = m(e);
  if (t === null)
    return null;
  if (r && typeof r == "object" && "slots" in r && r.slots) {
    let l = t || W;
    if (!r.slots[l]) {
      let n = new Intl.ListFormat().format(Object.keys(r.slots).map((f) => `"${f}"`)), u = t ? `Invalid slot "${t}".` : "A slot prop is required.";
      throw new Error(`${u} Valid slot names are ${n}.`);
    }
    return r.slots[l];
  }
  return r;
}
function T(e, t, r) {
  let l = A(r, e.slot) || {}, { ref: n, ...u } = l, f = D(h(() => Z(t, n), [
    t,
    n
  ])), a = X(u, e);
  return "style" in u && u.style && "style" in e && e.style && (typeof u.style == "function" || typeof e.style == "function" ? a.style = (c) => {
    let d = typeof u.style == "function" ? u.style(c) : u.style, o = {
      ...c.defaultStyle,
      ...d
    }, C = typeof e.style == "function" ? e.style({
      ...c,
      defaultStyle: o
    }) : e.style;
    return {
      ...o,
      ...C
    };
  } : a.style = {
    ...u.style,
    ...e.style
  }), [
    a,
    f
  ];
}
function _(e = !0) {
  let [t, r] = b(e), l = s(!1), n = x((u) => {
    l.current = !0, r(!!u);
  }, []);
  return v(() => {
    l.current || r(!1);
  }, []), [
    n,
    t
  ];
}
function ee(e) {
  const t = /^(data-.*)$/;
  let r = {};
  for (const l in e) t.test(l) || (r[l] = e[l]);
  return r;
}
export {
  T as $,
  O as a,
  X as b,
  v as c,
  G as d,
  H as e,
  _ as f,
  Y as g,
  A as h,
  L as i,
  ee as j,
  Q as k,
  W as l,
  D as m,
  Z as n
};
