// @angi/one-design-system/dist/components/es/_chunks/ValidationMessage-Dc42Hn25.js
import { jsxs as d, jsx as r } from "react/jsx-runtime";
import * as e from "react";
import { forwardRef as g } from "react";
import { c as s } from "./clsx-OuTLNxxd.js";
import { I as p } from "./Icon-DITD-yoX.js";
import '../assets/ValidationMessage.css';const x = (a) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 512 512", ...a }, /* @__PURE__ */ e.createElement("path", { fill: "currentColor", d: "M256 512a256 256 0 1 1 0-512 256 256 0 1 1 0 512zm0-464a208 208 0 1 0 0 416 208 208 0 1 0 0-416zm70.7 121.9c7.8-10.7 22.8-13.1 33.5-5.3 10.7 7.8 13.1 22.8 5.3 33.5L243.4 366.1c-4.1 5.7-10.5 9.3-17.5 9.8-7 .5-13.9-2-18.8-6.9l-55.9-55.9c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l36 36 105.6-145.2z" })), f = (a) => /* @__PURE__ */ e.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 512 512", ...a }, /* @__PURE__ */ e.createElement("path", { fill: "currentColor", d: "M256 512a256 256 0 1 1 0-512 256 256 0 1 1 0 512zm0-464a208 208 0 1 0 0 416 208 208 0 1 0 0-416zm0 336a32 32 0 1 1 0-64 32 32 0 1 1 0 64zm0-256c18.2 0 32.7 15.5 31.4 33.7l-7.4 104c-.9 12.6-11.4 22.3-23.9 22.3-12.6 0-23-9.7-23.9-22.3l-7.4-104C223.3 143.5 237.8 128 256 128z" }));
var w = "_4squmg0", u = "_4squmg1", z = { invalid: "_4squmg2", valid: "_4squmg3" }, h = { invalid: "_4squmg4", valid: "_4squmg5" };
const C = g(
  ({
    children: a,
    className: l,
    variant: i,
    errorMessagePrefix: o,
    successMessagePrefix: t,
    ...n
  }, c) => {
    const { icon: m, prefix: v } = i === "invalid" ? { icon: f, prefix: o } : { icon: x, prefix: t };
    return /* @__PURE__ */ d(
      "p",
      {
        ref: c,
        role: "alert",
        className: s(w, l),
        ...n,
        children: [
          /* @__PURE__ */ r(
            p,
            {
              size: 20,
              "aria-label": v,
              className: h[i],
              children: /* @__PURE__ */ r(m, {})
            }
          ),
          /* @__PURE__ */ r(
            "span",
            {
              className: s(
                u,
                z[i]
              ),
              children: a
            }
          )
        ]
      }
    );
  }
);
C.displayName = "ValidationMessage";
export {
  C as V
};
