// @angi/one-design-system/dist/tokens/js/contract.js
/**
 * Do not edit directly, this file was auto-generated.
 */


export const borderRadius = {
"2xl": "border-radius-2xl",
"3xl": "border-radius-3xl",
default: "border-radius-default",
full: "border-radius-full",
lg: "border-radius-lg",
md: "border-radius-md",
none: "border-radius-none",
sm: "border-radius-sm",
xl: "border-radius-xl"
};

export const borderWidth = {
"1": "border-width-1",
"2": "border-width-2",
"4": "border-width-4",
"8": "border-width-8",
none: "border-width-none"
};

export const breakpoints = {
"2xs": "breakpoints-2xs",
"2xl": "breakpoints-2xl",
lg: "breakpoints-lg",
md: "breakpoints-md",
sm: "breakpoints-sm",
xl: "breakpoints-xl",
xs: "breakpoints-xs"
};

export const color = {
bg: {
  base: {
    active: "color-bg-base-active",
    default: "color-bg-base-default",
    disabled: "color-bg-base-disabled",
    hover: "color-bg-base-hover",
    readonly: "color-bg-base-readonly"
  },
  checked: {
    default: "color-bg-checked-default",
    hover: "color-bg-checked-hover"
  },
  danger: {
    solid: {
      default: "color-bg-danger-solid-default",
      hover: "color-bg-danger-solid-hover"
    },
    subtle: {
      default: "color-bg-danger-subtle-default",
      hover: "color-bg-danger-subtle-hover"
    }
  },
  info: {
    solid: {
      default: "color-bg-info-solid-default",
      hover: "color-bg-info-solid-hover"
    },
    subtle: {
      default: "color-bg-info-subtle-default",
      hover: "color-bg-info-subtle-hover"
    }
  },
  neutral: {
    solid: {
      default: "color-bg-neutral-solid-default",
      hover: "color-bg-neutral-solid-hover"
    },
    subtle: {
      active: "color-bg-neutral-subtle-active",
      default: "color-bg-neutral-subtle-default",
      hover: "color-bg-neutral-subtle-hover"
    }
  },
  overlay: {
    default: "color-bg-overlay-default"
  },
  page: {
    default: "color-bg-page-default"
  },
  primary: {
    solid: {
      default: "color-bg-primary-solid-default",
      disabled: "color-bg-primary-solid-disabled",
      hover: "color-bg-primary-solid-hover"
    },
    subtle: {
      default: "color-bg-primary-subtle-default",
      disabled: "color-bg-primary-subtle-disabled",
      hover: "color-bg-primary-subtle-hover"
    }
  },
  secondary: {
    solid: {
      default: "color-bg-secondary-solid-default",
      hover: "color-bg-secondary-solid-hover"
    },
    subtle: {
      default: "color-bg-secondary-subtle-default",
      hover: "color-bg-secondary-subtle-hover"
    }
  },
  section: {
    neutral: {
      subtle: {
        default: "color-bg-section-neutral-subtle-default"
      }
    },
    primary: {
      subtle: {
        default: "color-bg-section-primary-subtle-default"
      }
    },
    white: {
      default: "color-bg-section-white-default"
    }
  },
  success: {
    solid: {
      default: "color-bg-success-solid-default",
      hover: "color-bg-success-solid-hover"
    },
    subtle: {
      default: "color-bg-success-subtle-default",
      hover: "color-bg-success-subtle-hover"
    }
  },
  warning: {
    solid: {
      default: "color-bg-warning-solid-default",
      hover: "color-bg-warning-solid-hover"
    },
    subtle: {
      default: "color-bg-warning-subtle-default",
      hover: "color-bg-warning-subtle-hover"
    }
  }
},
black: "color-black",
border: {
  checked: {
    default: "color-border-checked-default"
  },
  danger: {
    solid: {
      default: "color-border-danger-solid-default"
    }
  },
  info: {
    solid: {
      default: "color-border-info-solid-default"
    }
  },
  neutral: {
    solid: {
      default: "color-border-neutral-solid-default",
      disabled: "color-border-neutral-solid-disabled"
    },
    subtle: {
      default: "color-border-neutral-subtle-default"
    }
  },
  primary: {
    solid: {
      default: "color-border-primary-solid-default",
      disabled: "color-border-primary-solid-disabled",
      hover: "color-border-primary-solid-hover"
    }
  },
  success: {
    solid: {
      default: "color-border-success-solid-default"
    }
  },
  warning: {
    solid: {
      default: "color-border-warning-solid-default"
    }
  }
},
focus: {
  default: "color-focus-default"
},
icon: {
  danger: {
    default: "color-icon-danger-default",
    disabled: "color-icon-danger-disabled",
    hover: "color-icon-danger-hover"
  },
  info: {
    default: "color-icon-info-default",
    disabled: "color-icon-info-disabled",
    hover: "color-icon-info-hover"
  },
  neutral: {
    default: "color-icon-neutral-default",
    disabled: "color-icon-neutral-disabled",
    hover: "color-icon-neutral-hover"
  },
  primary: {
    default: "color-icon-primary-default",
    disabled: "color-icon-primary-disabled",
    hover: "color-icon-primary-hover"
  },
  secondary: {
    default: "color-icon-secondary-default",
    disabled: "color-icon-secondary-disabled",
    hover: "color-icon-secondary-hover"
  },
  success: {
    default: "color-icon-success-default",
    disabled: "color-icon-success-disabled",
    hover: "color-icon-success-hover"
  },
  warning: {
    default: "color-icon-warning-default",
    disabled: "color-icon-warning-disabled",
    hover: "color-icon-warning-hover"
  }
},
text: {
  danger: {
    default: "color-text-danger-default",
    disabled: "color-text-danger-disabled",
    hover: "color-text-danger-hover"
  },
  info: {
    default: "color-text-info-default",
    disabled: "color-text-info-disabled",
    hover: "color-text-info-hover"
  },
  link: {
    inverted: {
      active: "color-text-link-inverted-active",
      default: "color-text-link-inverted-default",
      hover: "color-text-link-inverted-hover",
      visited: "color-text-link-inverted-visited"
    },
    neutral: {
      active: "color-text-link-neutral-active",
      default: "color-text-link-neutral-default",
      hover: "color-text-link-neutral-hover",
      visited: "color-text-link-neutral-visited"
    }
  },
  neutral: {
    default: "color-text-neutral-default",
    disabled: "color-text-neutral-disabled",
    placeholder: "color-text-neutral-placeholder",
    subtle: "color-text-neutral-subtle"
  },
  primary: {
    default: "color-text-primary-default",
    disabled: "color-text-primary-disabled",
    hover: "color-text-primary-hover"
  },
  secondary: {
    default: "color-text-secondary-default",
    disabled: "color-text-secondary-disabled",
    hover: "color-text-secondary-hover"
  },
  success: {
    default: "color-text-success-default",
    disabled: "color-text-success-disabled",
    hover: "color-text-success-hover"
  },
  warning: {
    default: "color-text-warning-default",
    disabled: "color-text-warning-disabled",
    hover: "color-text-warning-hover"
  }
},
transparent: "color-transparent",
white: "color-white"
};

export const componentColor = {
bg: {
  danger: {
    ghost: {
      active: "component-color-bg-danger-ghost-active",
      default: "component-color-bg-danger-ghost-default",
      disabled: "component-color-bg-danger-ghost-disabled",
      hover: "component-color-bg-danger-ghost-hover"
    },
    outline: {
      active: "component-color-bg-danger-outline-active",
      default: "component-color-bg-danger-outline-default",
      disabled: "component-color-bg-danger-outline-disabled",
      hover: "component-color-bg-danger-outline-hover"
    },
    solid: {
      active: "component-color-bg-danger-solid-active",
      default: "component-color-bg-danger-solid-default",
      disabled: "component-color-bg-danger-solid-disabled",
      hover: "component-color-bg-danger-solid-hover"
    },
    subtle: {
      active: "component-color-bg-danger-subtle-active",
      default: "component-color-bg-danger-subtle-default",
      disabled: "component-color-bg-danger-subtle-disabled",
      hover: "component-color-bg-danger-subtle-hover"
    }
  },
  info: {
    ghost: {
      active: "component-color-bg-info-ghost-active",
      default: "component-color-bg-info-ghost-default",
      disabled: "component-color-bg-info-ghost-disabled",
      hover: "component-color-bg-info-ghost-hover"
    },
    outline: {
      active: "component-color-bg-info-outline-active",
      default: "component-color-bg-info-outline-default",
      disabled: "component-color-bg-info-outline-disabled",
      hover: "component-color-bg-info-outline-hover"
    },
    solid: {
      active: "component-color-bg-info-solid-active",
      default: "component-color-bg-info-solid-default",
      disabled: "component-color-bg-info-solid-disabled",
      hover: "component-color-bg-info-solid-hover"
    },
    subtle: {
      active: "component-color-bg-info-subtle-active",
      default: "component-color-bg-info-subtle-default",
      disabled: "component-color-bg-info-subtle-disabled",
      hover: "component-color-bg-info-subtle-hover"
    }
  },
  inverted: {
    ghost: {
      active: "component-color-bg-inverted-ghost-active",
      default: "component-color-bg-inverted-ghost-default",
      disabled: "component-color-bg-inverted-ghost-disabled",
      hover: "component-color-bg-inverted-ghost-hover"
    },
    outline: {
      active: "component-color-bg-inverted-outline-active",
      default: "component-color-bg-inverted-outline-default",
      disabled: "component-color-bg-inverted-outline-disabled",
      hover: "component-color-bg-inverted-outline-hover"
    },
    solid: {
      active: "component-color-bg-inverted-solid-active",
      default: "component-color-bg-inverted-solid-default",
      disabled: "component-color-bg-inverted-solid-disabled",
      hover: "component-color-bg-inverted-solid-hover"
    },
    subtle: {
      active: "component-color-bg-inverted-subtle-active",
      default: "component-color-bg-inverted-subtle-default",
      disabled: "component-color-bg-inverted-subtle-disabled",
      hover: "component-color-bg-inverted-subtle-hover"
    }
  },
  neutral: {
    ghost: {
      active: "component-color-bg-neutral-ghost-active",
      default: "component-color-bg-neutral-ghost-default",
      disabled: "component-color-bg-neutral-ghost-disabled",
      hover: "component-color-bg-neutral-ghost-hover"
    },
    outline: {
      active: "component-color-bg-neutral-outline-active",
      default: "component-color-bg-neutral-outline-default",
      disabled: "component-color-bg-neutral-outline-disabled",
      hover: "component-color-bg-neutral-outline-hover"
    },
    solid: {
      active: "component-color-bg-neutral-solid-active",
      default: "component-color-bg-neutral-solid-default",
      disabled: "component-color-bg-neutral-solid-disabled",
      hover: "component-color-bg-neutral-solid-hover"
    },
    subtle: {
      active: "component-color-bg-neutral-subtle-active",
      default: "component-color-bg-neutral-subtle-default",
      disabled: "component-color-bg-neutral-subtle-disabled",
      hover: "component-color-bg-neutral-subtle-hover"
    }
  },
  primary: {
    ghost: {
      active: "component-color-bg-primary-ghost-active",
      default: "component-color-bg-primary-ghost-default",
      disabled: "component-color-bg-primary-ghost-disabled",
      hover: "component-color-bg-primary-ghost-hover"
    },
    outline: {
      active: "component-color-bg-primary-outline-active",
      default: "component-color-bg-primary-outline-default",
      disabled: "component-color-bg-primary-outline-disabled",
      hover: "component-color-bg-primary-outline-hover"
    },
    solid: {
      active: "component-color-bg-primary-solid-active",
      default: "component-color-bg-primary-solid-default",
      disabled: "component-color-bg-primary-solid-disabled",
      hover: "component-color-bg-primary-solid-hover"
    },
    subtle: {
      active: "component-color-bg-primary-subtle-active",
      default: "component-color-bg-primary-subtle-default",
      disabled: "component-color-bg-primary-subtle-disabled",
      hover: "component-color-bg-primary-subtle-hover"
    }
  },
  success: {
    ghost: {
      active: "component-color-bg-success-ghost-active",
      default: "component-color-bg-success-ghost-default",
      disabled: "component-color-bg-success-ghost-disabled",
      hover: "component-color-bg-success-ghost-hover"
    },
    outline: {
      active: "component-color-bg-success-outline-active",
      default: "component-color-bg-success-outline-default",
      disabled: "component-color-bg-success-outline-disabled",
      hover: "component-color-bg-success-outline-hover"
    },
    solid: {
      active: "component-color-bg-success-solid-active",
      default: "component-color-bg-success-solid-default",
      disabled: "component-color-bg-success-solid-disabled",
      hover: "component-color-bg-success-solid-hover"
    },
    subtle: {
      active: "component-color-bg-success-subtle-active",
      default: "component-color-bg-success-subtle-default",
      disabled: "component-color-bg-success-subtle-disabled",
      hover: "component-color-bg-success-subtle-hover"
    }
  },
  warning: {
    ghost: {
      active: "component-color-bg-warning-ghost-active",
      default: "component-color-bg-warning-ghost-default",
      disabled: "component-color-bg-warning-ghost-disabled",
      hover: "component-color-bg-warning-ghost-hover"
    },
    outline: {
      active: "component-color-bg-warning-outline-active",
      default: "component-color-bg-warning-outline-default",
      disabled: "component-color-bg-warning-outline-disabled",
      hover: "component-color-bg-warning-outline-hover"
    },
    solid: {
      active: "component-color-bg-warning-solid-active",
      default: "component-color-bg-warning-solid-default",
      disabled: "component-color-bg-warning-solid-disabled",
      hover: "component-color-bg-warning-solid-hover"
    },
    subtle: {
      active: "component-color-bg-warning-subtle-active",
      default: "component-color-bg-warning-subtle-default",
      disabled: "component-color-bg-warning-subtle-disabled",
      hover: "component-color-bg-warning-subtle-hover"
    }
  }
},
border: {
  danger: {
    outline: {
      default: "component-color-border-danger-outline-default",
      disabled: "component-color-border-danger-outline-disabled"
    }
  },
  info: {
    outline: {
      default: "component-color-border-info-outline-default",
      disabled: "component-color-border-info-outline-disabled"
    }
  },
  inverted: {
    outline: {
      default: "component-color-border-inverted-outline-default",
      disabled: "component-color-border-inverted-outline-disabled"
    }
  },
  neutral: {
    outline: {
      default: "component-color-border-neutral-outline-default",
      disabled: "component-color-border-neutral-outline-disabled"
    }
  },
  primary: {
    outline: {
      default: "component-color-border-primary-outline-default",
      disabled: "component-color-border-primary-outline-disabled"
    }
  },
  success: {
    outline: {
      default: "component-color-border-success-outline-default",
      disabled: "component-color-border-success-outline-disabled"
    }
  },
  warning: {
    outline: {
      default: "component-color-border-warning-outline-default",
      disabled: "component-color-border-warning-outline-disabled"
    }
  }
},
icon: {
  danger: {
    ghost: {
      default: "component-color-icon-danger-ghost-default",
      disabled: "component-color-icon-danger-ghost-disabled"
    },
    outline: {
      default: "component-color-icon-danger-outline-default",
      disabled: "component-color-icon-danger-outline-disabled"
    },
    solid: {
      default: "component-color-icon-danger-solid-default",
      disabled: "component-color-icon-danger-solid-disabled"
    },
    subtle: {
      default: "component-color-icon-danger-subtle-default",
      disabled: "component-color-icon-danger-subtle-disabled"
    }
  },
  info: {
    ghost: {
      default: "component-color-icon-info-ghost-default",
      disabled: "component-color-icon-info-ghost-disabled"
    },
    outline: {
      default: "component-color-icon-info-outline-default",
      disabled: "component-color-icon-info-outline-disabled"
    },
    solid: {
      default: "component-color-icon-info-solid-default",
      disabled: "component-color-icon-info-solid-disabled"
    },
    subtle: {
      default: "component-color-icon-info-subtle-default",
      disabled: "component-color-icon-info-subtle-disabled"
    }
  },
  inverted: {
    ghost: {
      default: "component-color-icon-inverted-ghost-default",
      disabled: "component-color-icon-inverted-ghost-disabled"
    },
    outline: {
      default: "component-color-icon-inverted-outline-default",
      disabled: "component-color-icon-inverted-outline-disabled"
    },
    solid: {
      default: "component-color-icon-inverted-solid-default",
      disabled: "component-color-icon-inverted-solid-disabled"
    },
    subtle: {
      default: "component-color-icon-inverted-subtle-default",
      disabled: "component-color-icon-inverted-subtle-disabled"
    }
  },
  neutral: {
    ghost: {
      default: "component-color-icon-neutral-ghost-default",
      disabled: "component-color-icon-neutral-ghost-disabled"
    },
    outline: {
      default: "component-color-icon-neutral-outline-default",
      disabled: "component-color-icon-neutral-outline-disabled"
    },
    solid: {
      default: "component-color-icon-neutral-solid-default",
      disabled: "component-color-icon-neutral-solid-disabled"
    },
    subtle: {
      default: "component-color-icon-neutral-subtle-default",
      disabled: "component-color-icon-neutral-subtle-disabled"
    }
  },
  primary: {
    ghost: {
      default: "component-color-icon-primary-ghost-default",
      disabled: "component-color-icon-primary-ghost-disabled"
    },
    outline: {
      default: "component-color-icon-primary-outline-default",
      disabled: "component-color-icon-primary-outline-disabled"
    },
    solid: {
      default: "component-color-icon-primary-solid-default",
      disabled: "component-color-icon-primary-solid-disabled"
    },
    subtle: {
      default: "component-color-icon-primary-subtle-default",
      disabled: "component-color-icon-primary-subtle-disabled"
    }
  },
  success: {
    ghost: {
      default: "component-color-icon-success-ghost-default",
      disabled: "component-color-icon-success-ghost-disabled"
    },
    outline: {
      default: "component-color-icon-success-outline-default",
      disabled: "component-color-icon-success-outline-disabled"
    },
    solid: {
      default: "component-color-icon-success-solid-default",
      disabled: "component-color-icon-success-solid-disabled"
    },
    subtle: {
      default: "component-color-icon-success-subtle-default",
      disabled: "component-color-icon-success-subtle-disabled"
    }
  },
  warning: {
    ghost: {
      default: "component-color-icon-warning-ghost-default",
      disabled: "component-color-icon-warning-ghost-disabled"
    },
    outline: {
      default: "component-color-icon-warning-outline-default",
      disabled: "component-color-icon-warning-outline-disabled"
    },
    solid: {
      default: "component-color-icon-warning-solid-default",
      disabled: "component-color-icon-warning-solid-disabled"
    },
    subtle: {
      default: "component-color-icon-warning-subtle-default",
      disabled: "component-color-icon-warning-subtle-disabled"
    }
  }
},
text: {
  danger: {
    ghost: {
      default: "component-color-text-danger-ghost-default",
      disabled: "component-color-text-danger-ghost-disabled"
    },
    outline: {
      default: "component-color-text-danger-outline-default",
      disabled: "component-color-text-danger-outline-disabled"
    },
    solid: {
      default: "component-color-text-danger-solid-default",
      disabled: "component-color-text-danger-solid-disabled"
    },
    subtle: {
      default: "component-color-text-danger-subtle-default",
      disabled: "component-color-text-danger-subtle-disabled"
    }
  },
  info: {
    ghost: {
      default: "component-color-text-info-ghost-default",
      disabled: "component-color-text-info-ghost-disabled"
    },
    outline: {
      default: "component-color-text-info-outline-default",
      disabled: "component-color-text-info-outline-disabled"
    },
    solid: {
      default: "component-color-text-info-solid-default",
      disabled: "component-color-text-info-solid-disabled"
    },
    subtle: {
      default: "component-color-text-info-subtle-default",
      disabled: "component-color-text-info-subtle-disabled"
    }
  },
  inverted: {
    ghost: {
      default: "component-color-text-inverted-ghost-default",
      disabled: "component-color-text-inverted-ghost-disabled"
    },
    outline: {
      default: "component-color-text-inverted-outline-default",
      disabled: "component-color-text-inverted-outline-disabled"
    },
    solid: {
      default: "component-color-text-inverted-solid-default",
      disabled: "component-color-text-inverted-solid-disabled"
    },
    subtle: {
      default: "component-color-text-inverted-subtle-default",
      disabled: "component-color-text-inverted-subtle-disabled"
    }
  },
  neutral: {
    ghost: {
      default: "component-color-text-neutral-ghost-default",
      disabled: "component-color-text-neutral-ghost-disabled"
    },
    outline: {
      default: "component-color-text-neutral-outline-default",
      disabled: "component-color-text-neutral-outline-disabled"
    },
    solid: {
      default: "component-color-text-neutral-solid-default",
      disabled: "component-color-text-neutral-solid-disabled"
    },
    subtle: {
      default: "component-color-text-neutral-subtle-default",
      disabled: "component-color-text-neutral-subtle-disabled"
    }
  },
  primary: {
    ghost: {
      default: "component-color-text-primary-ghost-default",
      disabled: "component-color-text-primary-ghost-disabled"
    },
    outline: {
      default: "component-color-text-primary-outline-default",
      disabled: "component-color-text-primary-outline-disabled"
    },
    solid: {
      default: "component-color-text-primary-solid-default",
      disabled: "component-color-text-primary-solid-disabled"
    },
    subtle: {
      default: "component-color-text-primary-subtle-default",
      disabled: "component-color-text-primary-subtle-disabled"
    }
  },
  success: {
    ghost: {
      default: "component-color-text-success-ghost-default",
      disabled: "component-color-text-success-ghost-disabled"
    },
    outline: {
      default: "component-color-text-success-outline-default",
      disabled: "component-color-text-success-outline-disabled"
    },
    solid: {
      default: "component-color-text-success-solid-default",
      disabled: "component-color-text-success-solid-disabled"
    },
    subtle: {
      default: "component-color-text-success-subtle-default",
      disabled: "component-color-text-success-subtle-disabled"
    }
  },
  warning: {
    ghost: {
      default: "component-color-text-warning-ghost-default",
      disabled: "component-color-text-warning-ghost-disabled"
    },
    outline: {
      default: "component-color-text-warning-outline-default",
      disabled: "component-color-text-warning-outline-disabled"
    },
    solid: {
      default: "component-color-text-warning-solid-default",
      disabled: "component-color-text-warning-solid-disabled"
    },
    subtle: {
      default: "component-color-text-warning-subtle-default",
      disabled: "component-color-text-warning-subtle-disabled"
    }
  }
}
};

export const components = {
button: {
  fontWeight: "components-button-font-weight",
  fontWeightSmall: "components-button-font-weight-small",
  minWidth: "components-button-min-width"
},
divider: {
  color: {
    default: "components-divider-color-default"
  }
},
input: {
  height: {
    large: "components-input-height-large",
    medium: "components-input-height-medium",
    small: "components-input-height-small"
  },
  width: {
    large: "components-input-width-large",
    medium: "components-input-width-medium",
    small: "components-input-width-small"
  }
},
modal: {
  maxHeight: {
    lg: "components-modal-max-height-lg",
    md: "components-modal-max-height-md",
    sm: "components-modal-max-height-sm",
    xl: "components-modal-max-height-xl"
  },
  minWidth: "components-modal-min-width",
  width: {
    lg: "components-modal-width-lg",
    md: "components-modal-width-md",
    sm: "components-modal-width-sm"
  }
},
textarea: {
  minHeight: "components-textarea-min-height"
}
};

export const font = {
family: {
  body: "font-family-body",
  heading: "font-family-heading"
},
letterSpacing: {
  loose: "font-letter-spacing-loose",
  looser: "font-letter-spacing-looser",
  normal: "font-letter-spacing-normal",
  tight: "font-letter-spacing-tight",
  tighter: "font-letter-spacing-tighter"
},
size: {
  "2xs": "font-size-2xs",
  "2xl": "font-size-2xl",
  "3xl": "font-size-3xl",
  "4xl": "font-size-4xl",
  "5xl": "font-size-5xl",
  "6xl": "font-size-6xl",
  "7xl": "font-size-7xl",
  base: "font-size-base",
  lg: "font-size-lg",
  sm: "font-size-sm",
  xl: "font-size-xl",
  xs: "font-size-xs"
},
weight: {
  body: "font-weight-body",
  bodyHeavy: "font-weight-body-heavy",
  heading: "font-weight-heading",
  headingSlim: "font-weight-heading-slim"
}
};

export const spacing = {
"0": "spacing-0",
"2": "spacing-2",
"4": "spacing-4",
"6": "spacing-6",
"8": "spacing-8",
"12": "spacing-12",
"16": "spacing-16",
"20": "spacing-20",
"24": "spacing-24",
"28": "spacing-28",
"32": "spacing-32",
"36": "spacing-36",
"40": "spacing-40",
"44": "spacing-44",
"48": "spacing-48",
"56": "spacing-56",
"60": "spacing-60",
"64": "spacing-64",
"72": "spacing-72",
"80": "spacing-80",
"96": "spacing-96",
"128": "spacing-128",
"160": "spacing-160"
};

export const text = {
body: {
  "2xs": {
    heavy: {
      fontFamily: "text-body-2xs-heavy-font-family",
      fontSize: "text-body-2xs-heavy-font-size",
      fontWeight: "text-body-2xs-heavy-font-weight",
      letterSpacing: "text-body-2xs-heavy-letter-spacing",
      lineHeight: "text-body-2xs-heavy-line-height",
      textDecoration: "text-body-2xs-heavy-text-decoration",
      textTransform: "text-body-2xs-heavy-text-transform"
    },
    normal: {
      fontFamily: "text-body-2xs-normal-font-family",
      fontSize: "text-body-2xs-normal-font-size",
      fontWeight: "text-body-2xs-normal-font-weight",
      letterSpacing: "text-body-2xs-normal-letter-spacing",
      lineHeight: "text-body-2xs-normal-line-height",
      textDecoration: "text-body-2xs-normal-text-decoration",
      textTransform: "text-body-2xs-normal-text-transform"
    }
  },
  base: {
    heavy: {
      fontFamily: "text-body-base-heavy-font-family",
      fontSize: "text-body-base-heavy-font-size",
      fontWeight: "text-body-base-heavy-font-weight",
      letterSpacing: "text-body-base-heavy-letter-spacing",
      lineHeight: "text-body-base-heavy-line-height",
      textDecoration: "text-body-base-heavy-text-decoration",
      textTransform: "text-body-base-heavy-text-transform"
    },
    normal: {
      fontFamily: "text-body-base-normal-font-family",
      fontSize: "text-body-base-normal-font-size",
      fontWeight: "text-body-base-normal-font-weight",
      letterSpacing: "text-body-base-normal-letter-spacing",
      lineHeight: "text-body-base-normal-line-height",
      textDecoration: "text-body-base-normal-text-decoration",
      textTransform: "text-body-base-normal-text-transform"
    }
  },
  lg: {
    heavy: {
      fontFamily: "text-body-lg-heavy-font-family",
      fontSize: "text-body-lg-heavy-font-size",
      fontWeight: "text-body-lg-heavy-font-weight",
      letterSpacing: "text-body-lg-heavy-letter-spacing",
      lineHeight: "text-body-lg-heavy-line-height",
      textDecoration: "text-body-lg-heavy-text-decoration",
      textTransform: "text-body-lg-heavy-text-transform"
    },
    normal: {
      fontFamily: "text-body-lg-normal-font-family",
      fontSize: "text-body-lg-normal-font-size",
      fontWeight: "text-body-lg-normal-font-weight",
      letterSpacing: "text-body-lg-normal-letter-spacing",
      lineHeight: "text-body-lg-normal-line-height",
      textDecoration: "text-body-lg-normal-text-decoration",
      textTransform: "text-body-lg-normal-text-transform"
    }
  },
  sm: {
    heavy: {
      fontFamily: "text-body-sm-heavy-font-family",
      fontSize: "text-body-sm-heavy-font-size",
      fontWeight: "text-body-sm-heavy-font-weight",
      letterSpacing: "text-body-sm-heavy-letter-spacing",
      lineHeight: "text-body-sm-heavy-line-height",
      textDecoration: "text-body-sm-heavy-text-decoration",
      textTransform: "text-body-sm-heavy-text-transform"
    },
    normal: {
      fontFamily: "text-body-sm-normal-font-family",
      fontSize: "text-body-sm-normal-font-size",
      fontWeight: "text-body-sm-normal-font-weight",
      letterSpacing: "text-body-sm-normal-letter-spacing",
      lineHeight: "text-body-sm-normal-line-height",
      textDecoration: "text-body-sm-normal-text-decoration",
      textTransform: "text-body-sm-normal-text-transform"
    }
  },
  xs: {
    heavy: {
      fontFamily: "text-body-xs-heavy-font-family",
      fontSize: "text-body-xs-heavy-font-size",
      fontWeight: "text-body-xs-heavy-font-weight",
      letterSpacing: "text-body-xs-heavy-letter-spacing",
      lineHeight: "text-body-xs-heavy-line-height",
      textDecoration: "text-body-xs-heavy-text-decoration",
      textTransform: "text-body-xs-heavy-text-transform"
    },
    normal: {
      fontFamily: "text-body-xs-normal-font-family",
      fontSize: "text-body-xs-normal-font-size",
      fontWeight: "text-body-xs-normal-font-weight",
      letterSpacing: "text-body-xs-normal-letter-spacing",
      lineHeight: "text-body-xs-normal-line-height",
      textDecoration: "text-body-xs-normal-text-decoration",
      textTransform: "text-body-xs-normal-text-transform"
    }
  }
},
heading: {
  "2xl": {
    fontFamily: "text-heading-2xl-font-family",
    fontSize: "text-heading-2xl-font-size",
    fontWeight: "text-heading-2xl-font-weight",
    letterSpacing: "text-heading-2xl-letter-spacing",
    lineHeight: "text-heading-2xl-line-height",
    textDecoration: "text-heading-2xl-text-decoration",
    textTransform: "text-heading-2xl-text-transform"
  },
  "3xl": {
    fontFamily: "text-heading-3xl-font-family",
    fontSize: "text-heading-3xl-font-size",
    fontWeight: "text-heading-3xl-font-weight",
    letterSpacing: "text-heading-3xl-letter-spacing",
    lineHeight: "text-heading-3xl-line-height",
    textDecoration: "text-heading-3xl-text-decoration",
    textTransform: "text-heading-3xl-text-transform"
  },
  "4xl": {
    fontFamily: "text-heading-4xl-font-family",
    fontSize: "text-heading-4xl-font-size",
    fontWeight: "text-heading-4xl-font-weight",
    letterSpacing: "text-heading-4xl-letter-spacing",
    lineHeight: "text-heading-4xl-line-height",
    textDecoration: "text-heading-4xl-text-decoration",
    textTransform: "text-heading-4xl-text-transform"
  },
  "5xl": {
    fontFamily: "text-heading-5xl-font-family",
    fontSize: "text-heading-5xl-font-size",
    fontWeight: "text-heading-5xl-font-weight",
    letterSpacing: "text-heading-5xl-letter-spacing",
    lineHeight: "text-heading-5xl-line-height",
    textDecoration: "text-heading-5xl-text-decoration",
    textTransform: "text-heading-5xl-text-transform"
  },
  "6xl": {
    fontFamily: "text-heading-6xl-font-family",
    fontSize: "text-heading-6xl-font-size",
    fontWeight: "text-heading-6xl-font-weight",
    letterSpacing: "text-heading-6xl-letter-spacing",
    lineHeight: "text-heading-6xl-line-height",
    textDecoration: "text-heading-6xl-text-decoration",
    textTransform: "text-heading-6xl-text-transform"
  },
  "7xl": {
    fontFamily: "text-heading-7xl-font-family",
    fontSize: "text-heading-7xl-font-size",
    fontWeight: "text-heading-7xl-font-weight",
    letterSpacing: "text-heading-7xl-letter-spacing",
    lineHeight: "text-heading-7xl-line-height",
    textDecoration: "text-heading-7xl-text-decoration",
    textTransform: "text-heading-7xl-text-transform"
  },
  base: {
    fontFamily: "text-heading-base-font-family",
    fontSize: "text-heading-base-font-size",
    fontWeight: "text-heading-base-font-weight",
    letterSpacing: "text-heading-base-letter-spacing",
    lineHeight: "text-heading-base-line-height",
    textDecoration: "text-heading-base-text-decoration",
    textTransform: "text-heading-base-text-transform"
  },
  lg: {
    fontFamily: "text-heading-lg-font-family",
    fontSize: "text-heading-lg-font-size",
    fontWeight: "text-heading-lg-font-weight",
    letterSpacing: "text-heading-lg-letter-spacing",
    lineHeight: "text-heading-lg-line-height",
    textDecoration: "text-heading-lg-text-decoration",
    textTransform: "text-heading-lg-text-transform"
  },
  xl: {
    fontFamily: "text-heading-xl-font-family",
    fontSize: "text-heading-xl-font-size",
    fontWeight: "text-heading-xl-font-weight",
    letterSpacing: "text-heading-xl-letter-spacing",
    lineHeight: "text-heading-xl-line-height",
    textDecoration: "text-heading-xl-text-decoration",
    textTransform: "text-heading-xl-text-transform"
  }
},
link: {
  base: {
    default: {
      fontFamily: "text-link-base-default-font-family",
      fontSize: "text-link-base-default-font-size",
      fontWeight: "text-link-base-default-font-weight",
      letterSpacing: "text-link-base-default-letter-spacing",
      lineHeight: "text-link-base-default-line-height",
      textDecoration: "text-link-base-default-text-decoration",
      textTransform: "text-link-base-default-text-transform"
    },
    hover: {
      fontFamily: "text-link-base-hover-font-family",
      fontSize: "text-link-base-hover-font-size",
      fontWeight: "text-link-base-hover-font-weight",
      letterSpacing: "text-link-base-hover-letter-spacing",
      lineHeight: "text-link-base-hover-line-height",
      textDecoration: "text-link-base-hover-text-decoration",
      textTransform: "text-link-base-hover-text-transform"
    }
  },
  sm: {
    default: {
      fontFamily: "text-link-sm-default-font-family",
      fontSize: "text-link-sm-default-font-size",
      fontWeight: "text-link-sm-default-font-weight",
      letterSpacing: "text-link-sm-default-letter-spacing",
      lineHeight: "text-link-sm-default-line-height",
      textDecoration: "text-link-sm-default-text-decoration",
      textTransform: "text-link-sm-default-text-transform"
    },
    hover: {
      fontFamily: "text-link-sm-hover-font-family",
      fontSize: "text-link-sm-hover-font-size",
      fontWeight: "text-link-sm-hover-font-weight",
      letterSpacing: "text-link-sm-hover-letter-spacing",
      lineHeight: "text-link-sm-hover-line-height",
      textDecoration: "text-link-sm-hover-text-decoration",
      textTransform: "text-link-sm-hover-text-transform"
    }
  }
},
list: {
  base: {
    normal: {
      fontFamily: "text-list-base-normal-font-family",
      fontSize: "text-list-base-normal-font-size",
      fontWeight: "text-list-base-normal-font-weight",
      letterSpacing: "text-list-base-normal-letter-spacing",
      lineHeight: "text-list-base-normal-line-height",
      textDecoration: "text-list-base-normal-text-decoration",
      textTransform: "text-list-base-normal-text-transform"
    }
  },
  sm: {
    normal: {
      fontFamily: "text-list-sm-normal-font-family",
      fontSize: "text-list-sm-normal-font-size",
      fontWeight: "text-list-sm-normal-font-weight",
      letterSpacing: "text-list-sm-normal-letter-spacing",
      lineHeight: "text-list-sm-normal-line-height",
      textDecoration: "text-list-sm-normal-text-decoration",
      textTransform: "text-list-sm-normal-text-transform"
    }
  }
}
};

export default {
  borderRadius,
  borderWidth,
  breakpoints,
  color,
  componentColor,
  components,
  font,
  spacing,
  text
};
