// @angi/one-design-system/dist/tokens/js/tokens.js
/**
 * Do not edit directly, this file was auto-generated.
 */


export const borderRadius = {
"2xl": "var(--border-radius-2xl)",
"3xl": "var(--border-radius-3xl)",
default: "var(--border-radius-default)",
full: "var(--border-radius-full)",
lg: "var(--border-radius-lg)",
md: "var(--border-radius-md)",
none: "var(--border-radius-none)",
sm: "var(--border-radius-sm)",
xl: "var(--border-radius-xl)"
};

export const borderWidth = {
"1": "var(--border-width-1)",
"2": "var(--border-width-2)",
"4": "var(--border-width-4)",
"8": "var(--border-width-8)",
none: "var(--border-width-none)"
};

export const breakpoints = {
"2xs": "20rem",
"2xl": "96rem",
lg: "64rem",
md: "48rem",
sm: "40rem",
xl: "80rem",
xs: "23.5rem"
};

export const color = {
bg: {
  base: {
    active: "var(--color-bg-base-active)",
    default: "var(--color-bg-base-default)",
    disabled: "var(--color-bg-base-disabled)",
    hover: "var(--color-bg-base-hover)",
    readonly: "var(--color-bg-base-readonly)"
  },
  checked: {
    default: "var(--color-bg-checked-default)",
    hover: "var(--color-bg-checked-hover)"
  },
  danger: {
    solid: {
      default: "var(--color-bg-danger-solid-default)",
      hover: "var(--color-bg-danger-solid-hover)"
    },
    subtle: {
      default: "var(--color-bg-danger-subtle-default)",
      hover: "var(--color-bg-danger-subtle-hover)"
    }
  },
  info: {
    solid: {
      default: "var(--color-bg-info-solid-default)",
      hover: "var(--color-bg-info-solid-hover)"
    },
    subtle: {
      default: "var(--color-bg-info-subtle-default)",
      hover: "var(--color-bg-info-subtle-hover)"
    }
  },
  neutral: {
    solid: {
      default: "var(--color-bg-neutral-solid-default)",
      hover: "var(--color-bg-neutral-solid-hover)"
    },
    subtle: {
      active: "var(--color-bg-neutral-subtle-active)",
      default: "var(--color-bg-neutral-subtle-default)",
      hover: "var(--color-bg-neutral-subtle-hover)"
    }
  },
  overlay: {
    default: "var(--color-bg-overlay-default)"
  },
  page: {
    default: "var(--color-bg-page-default)"
  },
  primary: {
    solid: {
      default: "var(--color-bg-primary-solid-default)",
      disabled: "var(--color-bg-primary-solid-disabled)",
      hover: "var(--color-bg-primary-solid-hover)"
    },
    subtle: {
      default: "var(--color-bg-primary-subtle-default)",
      disabled: "var(--color-bg-primary-subtle-disabled)",
      hover: "var(--color-bg-primary-subtle-hover)"
    }
  },
  secondary: {
    solid: {
      default: "var(--color-bg-secondary-solid-default)",
      hover: "var(--color-bg-secondary-solid-hover)"
    },
    subtle: {
      default: "var(--color-bg-secondary-subtle-default)",
      hover: "var(--color-bg-secondary-subtle-hover)"
    }
  },
  section: {
    neutral: {
      subtle: {
        default: "var(--color-bg-section-neutral-subtle-default)"
      }
    },
    primary: {
      subtle: {
        default: "var(--color-bg-section-primary-subtle-default)"
      }
    },
    white: {
      default: "var(--color-bg-section-white-default)"
    }
  },
  success: {
    solid: {
      default: "var(--color-bg-success-solid-default)",
      hover: "var(--color-bg-success-solid-hover)"
    },
    subtle: {
      default: "var(--color-bg-success-subtle-default)",
      hover: "var(--color-bg-success-subtle-hover)"
    }
  },
  warning: {
    solid: {
      default: "var(--color-bg-warning-solid-default)",
      hover: "var(--color-bg-warning-solid-hover)"
    },
    subtle: {
      default: "var(--color-bg-warning-subtle-default)",
      hover: "var(--color-bg-warning-subtle-hover)"
    }
  }
},
black: "var(--color-black)",
border: {
  checked: {
    default: "var(--color-border-checked-default)"
  },
  danger: {
    solid: {
      default: "var(--color-border-danger-solid-default)"
    }
  },
  info: {
    solid: {
      default: "var(--color-border-info-solid-default)"
    }
  },
  neutral: {
    solid: {
      default: "var(--color-border-neutral-solid-default)",
      disabled: "var(--color-border-neutral-solid-disabled)"
    },
    subtle: {
      default: "var(--color-border-neutral-subtle-default)"
    }
  },
  primary: {
    solid: {
      default: "var(--color-border-primary-solid-default)",
      disabled: "var(--color-border-primary-solid-disabled)",
      hover: "var(--color-border-primary-solid-hover)"
    }
  },
  success: {
    solid: {
      default: "var(--color-border-success-solid-default)"
    }
  },
  warning: {
    solid: {
      default: "var(--color-border-warning-solid-default)"
    }
  }
},
focus: {
  default: "var(--color-focus-default)"
},
icon: {
  danger: {
    default: "var(--color-icon-danger-default)",
    disabled: "var(--color-icon-danger-disabled)",
    hover: "var(--color-icon-danger-hover)"
  },
  info: {
    default: "var(--color-icon-info-default)",
    disabled: "var(--color-icon-info-disabled)",
    hover: "var(--color-icon-info-hover)"
  },
  neutral: {
    default: "var(--color-icon-neutral-default)",
    disabled: "var(--color-icon-neutral-disabled)",
    hover: "var(--color-icon-neutral-hover)"
  },
  primary: {
    default: "var(--color-icon-primary-default)",
    disabled: "var(--color-icon-primary-disabled)",
    hover: "var(--color-icon-primary-hover)"
  },
  secondary: {
    default: "var(--color-icon-secondary-default)",
    disabled: "var(--color-icon-secondary-disabled)",
    hover: "var(--color-icon-secondary-hover)"
  },
  success: {
    default: "var(--color-icon-success-default)",
    disabled: "var(--color-icon-success-disabled)",
    hover: "var(--color-icon-success-hover)"
  },
  warning: {
    default: "var(--color-icon-warning-default)",
    disabled: "var(--color-icon-warning-disabled)",
    hover: "var(--color-icon-warning-hover)"
  }
},
text: {
  danger: {
    default: "var(--color-text-danger-default)",
    disabled: "var(--color-text-danger-disabled)",
    hover: "var(--color-text-danger-hover)"
  },
  info: {
    default: "var(--color-text-info-default)",
    disabled: "var(--color-text-info-disabled)",
    hover: "var(--color-text-info-hover)"
  },
  link: {
    inverted: {
      active: "var(--color-text-link-inverted-active)",
      default: "var(--color-text-link-inverted-default)",
      hover: "var(--color-text-link-inverted-hover)",
      visited: "var(--color-text-link-inverted-visited)"
    },
    neutral: {
      active: "var(--color-text-link-neutral-active)",
      default: "var(--color-text-link-neutral-default)",
      hover: "var(--color-text-link-neutral-hover)",
      visited: "var(--color-text-link-neutral-visited)"
    }
  },
  neutral: {
    default: "var(--color-text-neutral-default)",
    disabled: "var(--color-text-neutral-disabled)",
    placeholder: "var(--color-text-neutral-placeholder)",
    subtle: "var(--color-text-neutral-subtle)"
  },
  primary: {
    default: "var(--color-text-primary-default)",
    disabled: "var(--color-text-primary-disabled)",
    hover: "var(--color-text-primary-hover)"
  },
  secondary: {
    default: "var(--color-text-secondary-default)",
    disabled: "var(--color-text-secondary-disabled)",
    hover: "var(--color-text-secondary-hover)"
  },
  success: {
    default: "var(--color-text-success-default)",
    disabled: "var(--color-text-success-disabled)",
    hover: "var(--color-text-success-hover)"
  },
  warning: {
    default: "var(--color-text-warning-default)",
    disabled: "var(--color-text-warning-disabled)",
    hover: "var(--color-text-warning-hover)"
  }
},
transparent: "var(--color-transparent)",
white: "var(--color-white)"
};

export const componentColor = {
bg: {
  danger: {
    ghost: {
      active: "var(--component-color-bg-danger-ghost-active)",
      default: "var(--component-color-bg-danger-ghost-default)",
      disabled: "var(--component-color-bg-danger-ghost-disabled)",
      hover: "var(--component-color-bg-danger-ghost-hover)"
    },
    outline: {
      active: "var(--component-color-bg-danger-outline-active)",
      default: "var(--component-color-bg-danger-outline-default)",
      disabled: "var(--component-color-bg-danger-outline-disabled)",
      hover: "var(--component-color-bg-danger-outline-hover)"
    },
    solid: {
      active: "var(--component-color-bg-danger-solid-active)",
      default: "var(--component-color-bg-danger-solid-default)",
      disabled: "var(--component-color-bg-danger-solid-disabled)",
      hover: "var(--component-color-bg-danger-solid-hover)"
    },
    subtle: {
      active: "var(--component-color-bg-danger-subtle-active)",
      default: "var(--component-color-bg-danger-subtle-default)",
      disabled: "var(--component-color-bg-danger-subtle-disabled)",
      hover: "var(--component-color-bg-danger-subtle-hover)"
    }
  },
  info: {
    ghost: {
      active: "var(--component-color-bg-info-ghost-active)",
      default: "var(--component-color-bg-info-ghost-default)",
      disabled: "var(--component-color-bg-info-ghost-disabled)",
      hover: "var(--component-color-bg-info-ghost-hover)"
    },
    outline: {
      active: "var(--component-color-bg-info-outline-active)",
      default: "var(--component-color-bg-info-outline-default)",
      disabled: "var(--component-color-bg-info-outline-disabled)",
      hover: "var(--component-color-bg-info-outline-hover)"
    },
    solid: {
      active: "var(--component-color-bg-info-solid-active)",
      default: "var(--component-color-bg-info-solid-default)",
      disabled: "var(--component-color-bg-info-solid-disabled)",
      hover: "var(--component-color-bg-info-solid-hover)"
    },
    subtle: {
      active: "var(--component-color-bg-info-subtle-active)",
      default: "var(--component-color-bg-info-subtle-default)",
      disabled: "var(--component-color-bg-info-subtle-disabled)",
      hover: "var(--component-color-bg-info-subtle-hover)"
    }
  },
  inverted: {
    ghost: {
      active: "var(--component-color-bg-inverted-ghost-active)",
      default: "var(--component-color-bg-inverted-ghost-default)",
      disabled: "var(--component-color-bg-inverted-ghost-disabled)",
      hover: "var(--component-color-bg-inverted-ghost-hover)"
    },
    outline: {
      active: "var(--component-color-bg-inverted-outline-active)",
      default: "var(--component-color-bg-inverted-outline-default)",
      disabled: "var(--component-color-bg-inverted-outline-disabled)",
      hover: "var(--component-color-bg-inverted-outline-hover)"
    },
    solid: {
      active: "var(--component-color-bg-inverted-solid-active)",
      default: "var(--component-color-bg-inverted-solid-default)",
      disabled: "var(--component-color-bg-inverted-solid-disabled)",
      hover: "var(--component-color-bg-inverted-solid-hover)"
    },
    subtle: {
      active: "var(--component-color-bg-inverted-subtle-active)",
      default: "var(--component-color-bg-inverted-subtle-default)",
      disabled: "var(--component-color-bg-inverted-subtle-disabled)",
      hover: "var(--component-color-bg-inverted-subtle-hover)"
    }
  },
  neutral: {
    ghost: {
      active: "var(--component-color-bg-neutral-ghost-active)",
      default: "var(--component-color-bg-neutral-ghost-default)",
      disabled: "var(--component-color-bg-neutral-ghost-disabled)",
      hover: "var(--component-color-bg-neutral-ghost-hover)"
    },
    outline: {
      active: "var(--component-color-bg-neutral-outline-active)",
      default: "var(--component-color-bg-neutral-outline-default)",
      disabled: "var(--component-color-bg-neutral-outline-disabled)",
      hover: "var(--component-color-bg-neutral-outline-hover)"
    },
    solid: {
      active: "var(--component-color-bg-neutral-solid-active)",
      default: "var(--component-color-bg-neutral-solid-default)",
      disabled: "var(--component-color-bg-neutral-solid-disabled)",
      hover: "var(--component-color-bg-neutral-solid-hover)"
    },
    subtle: {
      active: "var(--component-color-bg-neutral-subtle-active)",
      default: "var(--component-color-bg-neutral-subtle-default)",
      disabled: "var(--component-color-bg-neutral-subtle-disabled)",
      hover: "var(--component-color-bg-neutral-subtle-hover)"
    }
  },
  primary: {
    ghost: {
      active: "var(--component-color-bg-primary-ghost-active)",
      default: "var(--component-color-bg-primary-ghost-default)",
      disabled: "var(--component-color-bg-primary-ghost-disabled)",
      hover: "var(--component-color-bg-primary-ghost-hover)"
    },
    outline: {
      active: "var(--component-color-bg-primary-outline-active)",
      default: "var(--component-color-bg-primary-outline-default)",
      disabled: "var(--component-color-bg-primary-outline-disabled)",
      hover: "var(--component-color-bg-primary-outline-hover)"
    },
    solid: {
      active: "var(--component-color-bg-primary-solid-active)",
      default: "var(--component-color-bg-primary-solid-default)",
      disabled: "var(--component-color-bg-primary-solid-disabled)",
      hover: "var(--component-color-bg-primary-solid-hover)"
    },
    subtle: {
      active: "var(--component-color-bg-primary-subtle-active)",
      default: "var(--component-color-bg-primary-subtle-default)",
      disabled: "var(--component-color-bg-primary-subtle-disabled)",
      hover: "var(--component-color-bg-primary-subtle-hover)"
    }
  },
  success: {
    ghost: {
      active: "var(--component-color-bg-success-ghost-active)",
      default: "var(--component-color-bg-success-ghost-default)",
      disabled: "var(--component-color-bg-success-ghost-disabled)",
      hover: "var(--component-color-bg-success-ghost-hover)"
    },
    outline: {
      active: "var(--component-color-bg-success-outline-active)",
      default: "var(--component-color-bg-success-outline-default)",
      disabled: "var(--component-color-bg-success-outline-disabled)",
      hover: "var(--component-color-bg-success-outline-hover)"
    },
    solid: {
      active: "var(--component-color-bg-success-solid-active)",
      default: "var(--component-color-bg-success-solid-default)",
      disabled: "var(--component-color-bg-success-solid-disabled)",
      hover: "var(--component-color-bg-success-solid-hover)"
    },
    subtle: {
      active: "var(--component-color-bg-success-subtle-active)",
      default: "var(--component-color-bg-success-subtle-default)",
      disabled: "var(--component-color-bg-success-subtle-disabled)",
      hover: "var(--component-color-bg-success-subtle-hover)"
    }
  },
  warning: {
    ghost: {
      active: "var(--component-color-bg-warning-ghost-active)",
      default: "var(--component-color-bg-warning-ghost-default)",
      disabled: "var(--component-color-bg-warning-ghost-disabled)",
      hover: "var(--component-color-bg-warning-ghost-hover)"
    },
    outline: {
      active: "var(--component-color-bg-warning-outline-active)",
      default: "var(--component-color-bg-warning-outline-default)",
      disabled: "var(--component-color-bg-warning-outline-disabled)",
      hover: "var(--component-color-bg-warning-outline-hover)"
    },
    solid: {
      active: "var(--component-color-bg-warning-solid-active)",
      default: "var(--component-color-bg-warning-solid-default)",
      disabled: "var(--component-color-bg-warning-solid-disabled)",
      hover: "var(--component-color-bg-warning-solid-hover)"
    },
    subtle: {
      active: "var(--component-color-bg-warning-subtle-active)",
      default: "var(--component-color-bg-warning-subtle-default)",
      disabled: "var(--component-color-bg-warning-subtle-disabled)",
      hover: "var(--component-color-bg-warning-subtle-hover)"
    }
  }
},
border: {
  danger: {
    outline: {
      default: "var(--component-color-border-danger-outline-default)",
      disabled: "var(--component-color-border-danger-outline-disabled)"
    }
  },
  info: {
    outline: {
      default: "var(--component-color-border-info-outline-default)",
      disabled: "var(--component-color-border-info-outline-disabled)"
    }
  },
  inverted: {
    outline: {
      default: "var(--component-color-border-inverted-outline-default)",
      disabled: "var(--component-color-border-inverted-outline-disabled)"
    }
  },
  neutral: {
    outline: {
      default: "var(--component-color-border-neutral-outline-default)",
      disabled: "var(--component-color-border-neutral-outline-disabled)"
    }
  },
  primary: {
    outline: {
      default: "var(--component-color-border-primary-outline-default)",
      disabled: "var(--component-color-border-primary-outline-disabled)"
    }
  },
  success: {
    outline: {
      default: "var(--component-color-border-success-outline-default)",
      disabled: "var(--component-color-border-success-outline-disabled)"
    }
  },
  warning: {
    outline: {
      default: "var(--component-color-border-warning-outline-default)",
      disabled: "var(--component-color-border-warning-outline-disabled)"
    }
  }
},
icon: {
  danger: {
    ghost: {
      default: "var(--component-color-icon-danger-ghost-default)",
      disabled: "var(--component-color-icon-danger-ghost-disabled)"
    },
    outline: {
      default: "var(--component-color-icon-danger-outline-default)",
      disabled: "var(--component-color-icon-danger-outline-disabled)"
    },
    solid: {
      default: "var(--component-color-icon-danger-solid-default)",
      disabled: "var(--component-color-icon-danger-solid-disabled)"
    },
    subtle: {
      default: "var(--component-color-icon-danger-subtle-default)",
      disabled: "var(--component-color-icon-danger-subtle-disabled)"
    }
  },
  info: {
    ghost: {
      default: "var(--component-color-icon-info-ghost-default)",
      disabled: "var(--component-color-icon-info-ghost-disabled)"
    },
    outline: {
      default: "var(--component-color-icon-info-outline-default)",
      disabled: "var(--component-color-icon-info-outline-disabled)"
    },
    solid: {
      default: "var(--component-color-icon-info-solid-default)",
      disabled: "var(--component-color-icon-info-solid-disabled)"
    },
    subtle: {
      default: "var(--component-color-icon-info-subtle-default)",
      disabled: "var(--component-color-icon-info-subtle-disabled)"
    }
  },
  inverted: {
    ghost: {
      default: "var(--component-color-icon-inverted-ghost-default)",
      disabled: "var(--component-color-icon-inverted-ghost-disabled)"
    },
    outline: {
      default: "var(--component-color-icon-inverted-outline-default)",
      disabled: "var(--component-color-icon-inverted-outline-disabled)"
    },
    solid: {
      default: "var(--component-color-icon-inverted-solid-default)",
      disabled: "var(--component-color-icon-inverted-solid-disabled)"
    },
    subtle: {
      default: "var(--component-color-icon-inverted-subtle-default)",
      disabled: "var(--component-color-icon-inverted-subtle-disabled)"
    }
  },
  neutral: {
    ghost: {
      default: "var(--component-color-icon-neutral-ghost-default)",
      disabled: "var(--component-color-icon-neutral-ghost-disabled)"
    },
    outline: {
      default: "var(--component-color-icon-neutral-outline-default)",
      disabled: "var(--component-color-icon-neutral-outline-disabled)"
    },
    solid: {
      default: "var(--component-color-icon-neutral-solid-default)",
      disabled: "var(--component-color-icon-neutral-solid-disabled)"
    },
    subtle: {
      default: "var(--component-color-icon-neutral-subtle-default)",
      disabled: "var(--component-color-icon-neutral-subtle-disabled)"
    }
  },
  primary: {
    ghost: {
      default: "var(--component-color-icon-primary-ghost-default)",
      disabled: "var(--component-color-icon-primary-ghost-disabled)"
    },
    outline: {
      default: "var(--component-color-icon-primary-outline-default)",
      disabled: "var(--component-color-icon-primary-outline-disabled)"
    },
    solid: {
      default: "var(--component-color-icon-primary-solid-default)",
      disabled: "var(--component-color-icon-primary-solid-disabled)"
    },
    subtle: {
      default: "var(--component-color-icon-primary-subtle-default)",
      disabled: "var(--component-color-icon-primary-subtle-disabled)"
    }
  },
  success: {
    ghost: {
      default: "var(--component-color-icon-success-ghost-default)",
      disabled: "var(--component-color-icon-success-ghost-disabled)"
    },
    outline: {
      default: "var(--component-color-icon-success-outline-default)",
      disabled: "var(--component-color-icon-success-outline-disabled)"
    },
    solid: {
      default: "var(--component-color-icon-success-solid-default)",
      disabled: "var(--component-color-icon-success-solid-disabled)"
    },
    subtle: {
      default: "var(--component-color-icon-success-subtle-default)",
      disabled: "var(--component-color-icon-success-subtle-disabled)"
    }
  },
  warning: {
    ghost: {
      default: "var(--component-color-icon-warning-ghost-default)",
      disabled: "var(--component-color-icon-warning-ghost-disabled)"
    },
    outline: {
      default: "var(--component-color-icon-warning-outline-default)",
      disabled: "var(--component-color-icon-warning-outline-disabled)"
    },
    solid: {
      default: "var(--component-color-icon-warning-solid-default)",
      disabled: "var(--component-color-icon-warning-solid-disabled)"
    },
    subtle: {
      default: "var(--component-color-icon-warning-subtle-default)",
      disabled: "var(--component-color-icon-warning-subtle-disabled)"
    }
  }
},
text: {
  danger: {
    ghost: {
      default: "var(--component-color-text-danger-ghost-default)",
      disabled: "var(--component-color-text-danger-ghost-disabled)"
    },
    outline: {
      default: "var(--component-color-text-danger-outline-default)",
      disabled: "var(--component-color-text-danger-outline-disabled)"
    },
    solid: {
      default: "var(--component-color-text-danger-solid-default)",
      disabled: "var(--component-color-text-danger-solid-disabled)"
    },
    subtle: {
      default: "var(--component-color-text-danger-subtle-default)",
      disabled: "var(--component-color-text-danger-subtle-disabled)"
    }
  },
  info: {
    ghost: {
      default: "var(--component-color-text-info-ghost-default)",
      disabled: "var(--component-color-text-info-ghost-disabled)"
    },
    outline: {
      default: "var(--component-color-text-info-outline-default)",
      disabled: "var(--component-color-text-info-outline-disabled)"
    },
    solid: {
      default: "var(--component-color-text-info-solid-default)",
      disabled: "var(--component-color-text-info-solid-disabled)"
    },
    subtle: {
      default: "var(--component-color-text-info-subtle-default)",
      disabled: "var(--component-color-text-info-subtle-disabled)"
    }
  },
  inverted: {
    ghost: {
      default: "var(--component-color-text-inverted-ghost-default)",
      disabled: "var(--component-color-text-inverted-ghost-disabled)"
    },
    outline: {
      default: "var(--component-color-text-inverted-outline-default)",
      disabled: "var(--component-color-text-inverted-outline-disabled)"
    },
    solid: {
      default: "var(--component-color-text-inverted-solid-default)",
      disabled: "var(--component-color-text-inverted-solid-disabled)"
    },
    subtle: {
      default: "var(--component-color-text-inverted-subtle-default)",
      disabled: "var(--component-color-text-inverted-subtle-disabled)"
    }
  },
  neutral: {
    ghost: {
      default: "var(--component-color-text-neutral-ghost-default)",
      disabled: "var(--component-color-text-neutral-ghost-disabled)"
    },
    outline: {
      default: "var(--component-color-text-neutral-outline-default)",
      disabled: "var(--component-color-text-neutral-outline-disabled)"
    },
    solid: {
      default: "var(--component-color-text-neutral-solid-default)",
      disabled: "var(--component-color-text-neutral-solid-disabled)"
    },
    subtle: {
      default: "var(--component-color-text-neutral-subtle-default)",
      disabled: "var(--component-color-text-neutral-subtle-disabled)"
    }
  },
  primary: {
    ghost: {
      default: "var(--component-color-text-primary-ghost-default)",
      disabled: "var(--component-color-text-primary-ghost-disabled)"
    },
    outline: {
      default: "var(--component-color-text-primary-outline-default)",
      disabled: "var(--component-color-text-primary-outline-disabled)"
    },
    solid: {
      default: "var(--component-color-text-primary-solid-default)",
      disabled: "var(--component-color-text-primary-solid-disabled)"
    },
    subtle: {
      default: "var(--component-color-text-primary-subtle-default)",
      disabled: "var(--component-color-text-primary-subtle-disabled)"
    }
  },
  success: {
    ghost: {
      default: "var(--component-color-text-success-ghost-default)",
      disabled: "var(--component-color-text-success-ghost-disabled)"
    },
    outline: {
      default: "var(--component-color-text-success-outline-default)",
      disabled: "var(--component-color-text-success-outline-disabled)"
    },
    solid: {
      default: "var(--component-color-text-success-solid-default)",
      disabled: "var(--component-color-text-success-solid-disabled)"
    },
    subtle: {
      default: "var(--component-color-text-success-subtle-default)",
      disabled: "var(--component-color-text-success-subtle-disabled)"
    }
  },
  warning: {
    ghost: {
      default: "var(--component-color-text-warning-ghost-default)",
      disabled: "var(--component-color-text-warning-ghost-disabled)"
    },
    outline: {
      default: "var(--component-color-text-warning-outline-default)",
      disabled: "var(--component-color-text-warning-outline-disabled)"
    },
    solid: {
      default: "var(--component-color-text-warning-solid-default)",
      disabled: "var(--component-color-text-warning-solid-disabled)"
    },
    subtle: {
      default: "var(--component-color-text-warning-subtle-default)",
      disabled: "var(--component-color-text-warning-subtle-disabled)"
    }
  }
}
};

export const components = {
button: {
  fontWeight: "var(--components-button-font-weight)",
  fontWeightSmall: "var(--components-button-font-weight-small)",
  minWidth: "var(--components-button-min-width)"
},
divider: {
  color: {
    default: "var(--components-divider-color-default)"
  }
},
input: {
  height: {
    large: "var(--components-input-height-large)",
    medium: "var(--components-input-height-medium)",
    small: "var(--components-input-height-small)"
  },
  width: {
    large: "var(--components-input-width-large)",
    medium: "var(--components-input-width-medium)",
    small: "var(--components-input-width-small)"
  }
},
modal: {
  maxHeight: {
    lg: "var(--components-modal-max-height-lg)",
    md: "var(--components-modal-max-height-md)",
    sm: "var(--components-modal-max-height-sm)",
    xl: "var(--components-modal-max-height-xl)"
  },
  minWidth: "var(--components-modal-min-width)",
  width: {
    lg: "var(--components-modal-width-lg)",
    md: "var(--components-modal-width-md)",
    sm: "var(--components-modal-width-sm)"
  }
},
textarea: {
  minHeight: "var(--components-textarea-min-height)"
}
};

export const font = {
family: {
  body: "var(--font-family-body)",
  heading: "var(--font-family-heading)"
},
letterSpacing: {
  loose: "var(--font-letter-spacing-loose)",
  looser: "var(--font-letter-spacing-looser)",
  normal: "var(--font-letter-spacing-normal)",
  tight: "var(--font-letter-spacing-tight)",
  tighter: "var(--font-letter-spacing-tighter)"
},
size: {
  "2xs": "var(--font-size-2xs)",
  "2xl": "var(--font-size-2xl)",
  "3xl": "var(--font-size-3xl)",
  "4xl": "var(--font-size-4xl)",
  "5xl": "var(--font-size-5xl)",
  "6xl": "var(--font-size-6xl)",
  "7xl": "var(--font-size-7xl)",
  base: "var(--font-size-base)",
  lg: "var(--font-size-lg)",
  sm: "var(--font-size-sm)",
  xl: "var(--font-size-xl)",
  xs: "var(--font-size-xs)"
},
weight: {
  body: "var(--font-weight-body)",
  bodyHeavy: "var(--font-weight-body-heavy)",
  heading: "var(--font-weight-heading)",
  headingSlim: "var(--font-weight-heading-slim)"
}
};

export const spacing = {
"0": "var(--spacing-0)",
"2": "var(--spacing-2)",
"4": "var(--spacing-4)",
"6": "var(--spacing-6)",
"8": "var(--spacing-8)",
"12": "var(--spacing-12)",
"16": "var(--spacing-16)",
"20": "var(--spacing-20)",
"24": "var(--spacing-24)",
"28": "var(--spacing-28)",
"32": "var(--spacing-32)",
"36": "var(--spacing-36)",
"40": "var(--spacing-40)",
"44": "var(--spacing-44)",
"48": "var(--spacing-48)",
"56": "var(--spacing-56)",
"60": "var(--spacing-60)",
"64": "var(--spacing-64)",
"72": "var(--spacing-72)",
"80": "var(--spacing-80)",
"96": "var(--spacing-96)",
"128": "var(--spacing-128)",
"160": "var(--spacing-160)"
};

export const text = {
body: {
  "2xs": {
    heavy: {
      fontFamily: "var(--text-body-2xs-heavy-font-family)",
      fontSize: "var(--text-body-2xs-heavy-font-size)",
      fontWeight: "var(--text-body-2xs-heavy-font-weight)",
      letterSpacing: "var(--text-body-2xs-heavy-letter-spacing)",
      lineHeight: "var(--text-body-2xs-heavy-line-height)",
      textDecoration: "var(--text-body-2xs-heavy-text-decoration)",
      textTransform: "var(--text-body-2xs-heavy-text-transform)"
    },
    normal: {
      fontFamily: "var(--text-body-2xs-normal-font-family)",
      fontSize: "var(--text-body-2xs-normal-font-size)",
      fontWeight: "var(--text-body-2xs-normal-font-weight)",
      letterSpacing: "var(--text-body-2xs-normal-letter-spacing)",
      lineHeight: "var(--text-body-2xs-normal-line-height)",
      textDecoration: "var(--text-body-2xs-normal-text-decoration)",
      textTransform: "var(--text-body-2xs-normal-text-transform)"
    }
  },
  base: {
    heavy: {
      fontFamily: "var(--text-body-base-heavy-font-family)",
      fontSize: "var(--text-body-base-heavy-font-size)",
      fontWeight: "var(--text-body-base-heavy-font-weight)",
      letterSpacing: "var(--text-body-base-heavy-letter-spacing)",
      lineHeight: "var(--text-body-base-heavy-line-height)",
      textDecoration: "var(--text-body-base-heavy-text-decoration)",
      textTransform: "var(--text-body-base-heavy-text-transform)"
    },
    normal: {
      fontFamily: "var(--text-body-base-normal-font-family)",
      fontSize: "var(--text-body-base-normal-font-size)",
      fontWeight: "var(--text-body-base-normal-font-weight)",
      letterSpacing: "var(--text-body-base-normal-letter-spacing)",
      lineHeight: "var(--text-body-base-normal-line-height)",
      textDecoration: "var(--text-body-base-normal-text-decoration)",
      textTransform: "var(--text-body-base-normal-text-transform)"
    }
  },
  lg: {
    heavy: {
      fontFamily: "var(--text-body-lg-heavy-font-family)",
      fontSize: "var(--text-body-lg-heavy-font-size)",
      fontWeight: "var(--text-body-lg-heavy-font-weight)",
      letterSpacing: "var(--text-body-lg-heavy-letter-spacing)",
      lineHeight: "var(--text-body-lg-heavy-line-height)",
      textDecoration: "var(--text-body-lg-heavy-text-decoration)",
      textTransform: "var(--text-body-lg-heavy-text-transform)"
    },
    normal: {
      fontFamily: "var(--text-body-lg-normal-font-family)",
      fontSize: "var(--text-body-lg-normal-font-size)",
      fontWeight: "var(--text-body-lg-normal-font-weight)",
      letterSpacing: "var(--text-body-lg-normal-letter-spacing)",
      lineHeight: "var(--text-body-lg-normal-line-height)",
      textDecoration: "var(--text-body-lg-normal-text-decoration)",
      textTransform: "var(--text-body-lg-normal-text-transform)"
    }
  },
  sm: {
    heavy: {
      fontFamily: "var(--text-body-sm-heavy-font-family)",
      fontSize: "var(--text-body-sm-heavy-font-size)",
      fontWeight: "var(--text-body-sm-heavy-font-weight)",
      letterSpacing: "var(--text-body-sm-heavy-letter-spacing)",
      lineHeight: "var(--text-body-sm-heavy-line-height)",
      textDecoration: "var(--text-body-sm-heavy-text-decoration)",
      textTransform: "var(--text-body-sm-heavy-text-transform)"
    },
    normal: {
      fontFamily: "var(--text-body-sm-normal-font-family)",
      fontSize: "var(--text-body-sm-normal-font-size)",
      fontWeight: "var(--text-body-sm-normal-font-weight)",
      letterSpacing: "var(--text-body-sm-normal-letter-spacing)",
      lineHeight: "var(--text-body-sm-normal-line-height)",
      textDecoration: "var(--text-body-sm-normal-text-decoration)",
      textTransform: "var(--text-body-sm-normal-text-transform)"
    }
  },
  xs: {
    heavy: {
      fontFamily: "var(--text-body-xs-heavy-font-family)",
      fontSize: "var(--text-body-xs-heavy-font-size)",
      fontWeight: "var(--text-body-xs-heavy-font-weight)",
      letterSpacing: "var(--text-body-xs-heavy-letter-spacing)",
      lineHeight: "var(--text-body-xs-heavy-line-height)",
      textDecoration: "var(--text-body-xs-heavy-text-decoration)",
      textTransform: "var(--text-body-xs-heavy-text-transform)"
    },
    normal: {
      fontFamily: "var(--text-body-xs-normal-font-family)",
      fontSize: "var(--text-body-xs-normal-font-size)",
      fontWeight: "var(--text-body-xs-normal-font-weight)",
      letterSpacing: "var(--text-body-xs-normal-letter-spacing)",
      lineHeight: "var(--text-body-xs-normal-line-height)",
      textDecoration: "var(--text-body-xs-normal-text-decoration)",
      textTransform: "var(--text-body-xs-normal-text-transform)"
    }
  }
},
heading: {
  "2xl": {
    fontFamily: "var(--text-heading-2xl-font-family)",
    fontSize: "var(--text-heading-2xl-font-size)",
    fontWeight: "var(--text-heading-2xl-font-weight)",
    letterSpacing: "var(--text-heading-2xl-letter-spacing)",
    lineHeight: "var(--text-heading-2xl-line-height)",
    textDecoration: "var(--text-heading-2xl-text-decoration)",
    textTransform: "var(--text-heading-2xl-text-transform)"
  },
  "3xl": {
    fontFamily: "var(--text-heading-3xl-font-family)",
    fontSize: "var(--text-heading-3xl-font-size)",
    fontWeight: "var(--text-heading-3xl-font-weight)",
    letterSpacing: "var(--text-heading-3xl-letter-spacing)",
    lineHeight: "var(--text-heading-3xl-line-height)",
    textDecoration: "var(--text-heading-3xl-text-decoration)",
    textTransform: "var(--text-heading-3xl-text-transform)"
  },
  "4xl": {
    fontFamily: "var(--text-heading-4xl-font-family)",
    fontSize: "var(--text-heading-4xl-font-size)",
    fontWeight: "var(--text-heading-4xl-font-weight)",
    letterSpacing: "var(--text-heading-4xl-letter-spacing)",
    lineHeight: "var(--text-heading-4xl-line-height)",
    textDecoration: "var(--text-heading-4xl-text-decoration)",
    textTransform: "var(--text-heading-4xl-text-transform)"
  },
  "5xl": {
    fontFamily: "var(--text-heading-5xl-font-family)",
    fontSize: "var(--text-heading-5xl-font-size)",
    fontWeight: "var(--text-heading-5xl-font-weight)",
    letterSpacing: "var(--text-heading-5xl-letter-spacing)",
    lineHeight: "var(--text-heading-5xl-line-height)",
    textDecoration: "var(--text-heading-5xl-text-decoration)",
    textTransform: "var(--text-heading-5xl-text-transform)"
  },
  "6xl": {
    fontFamily: "var(--text-heading-6xl-font-family)",
    fontSize: "var(--text-heading-6xl-font-size)",
    fontWeight: "var(--text-heading-6xl-font-weight)",
    letterSpacing: "var(--text-heading-6xl-letter-spacing)",
    lineHeight: "var(--text-heading-6xl-line-height)",
    textDecoration: "var(--text-heading-6xl-text-decoration)",
    textTransform: "var(--text-heading-6xl-text-transform)"
  },
  "7xl": {
    fontFamily: "var(--text-heading-7xl-font-family)",
    fontSize: "var(--text-heading-7xl-font-size)",
    fontWeight: "var(--text-heading-7xl-font-weight)",
    letterSpacing: "var(--text-heading-7xl-letter-spacing)",
    lineHeight: "var(--text-heading-7xl-line-height)",
    textDecoration: "var(--text-heading-7xl-text-decoration)",
    textTransform: "var(--text-heading-7xl-text-transform)"
  },
  base: {
    fontFamily: "var(--text-heading-base-font-family)",
    fontSize: "var(--text-heading-base-font-size)",
    fontWeight: "var(--text-heading-base-font-weight)",
    letterSpacing: "var(--text-heading-base-letter-spacing)",
    lineHeight: "var(--text-heading-base-line-height)",
    textDecoration: "var(--text-heading-base-text-decoration)",
    textTransform: "var(--text-heading-base-text-transform)"
  },
  lg: {
    fontFamily: "var(--text-heading-lg-font-family)",
    fontSize: "var(--text-heading-lg-font-size)",
    fontWeight: "var(--text-heading-lg-font-weight)",
    letterSpacing: "var(--text-heading-lg-letter-spacing)",
    lineHeight: "var(--text-heading-lg-line-height)",
    textDecoration: "var(--text-heading-lg-text-decoration)",
    textTransform: "var(--text-heading-lg-text-transform)"
  },
  xl: {
    fontFamily: "var(--text-heading-xl-font-family)",
    fontSize: "var(--text-heading-xl-font-size)",
    fontWeight: "var(--text-heading-xl-font-weight)",
    letterSpacing: "var(--text-heading-xl-letter-spacing)",
    lineHeight: "var(--text-heading-xl-line-height)",
    textDecoration: "var(--text-heading-xl-text-decoration)",
    textTransform: "var(--text-heading-xl-text-transform)"
  }
},
link: {
  base: {
    default: {
      fontFamily: "var(--text-link-base-default-font-family)",
      fontSize: "var(--text-link-base-default-font-size)",
      fontWeight: "var(--text-link-base-default-font-weight)",
      letterSpacing: "var(--text-link-base-default-letter-spacing)",
      lineHeight: "var(--text-link-base-default-line-height)",
      textDecoration: "var(--text-link-base-default-text-decoration)",
      textTransform: "var(--text-link-base-default-text-transform)"
    },
    hover: {
      fontFamily: "var(--text-link-base-hover-font-family)",
      fontSize: "var(--text-link-base-hover-font-size)",
      fontWeight: "var(--text-link-base-hover-font-weight)",
      letterSpacing: "var(--text-link-base-hover-letter-spacing)",
      lineHeight: "var(--text-link-base-hover-line-height)",
      textDecoration: "var(--text-link-base-hover-text-decoration)",
      textTransform: "var(--text-link-base-hover-text-transform)"
    }
  },
  sm: {
    default: {
      fontFamily: "var(--text-link-sm-default-font-family)",
      fontSize: "var(--text-link-sm-default-font-size)",
      fontWeight: "var(--text-link-sm-default-font-weight)",
      letterSpacing: "var(--text-link-sm-default-letter-spacing)",
      lineHeight: "var(--text-link-sm-default-line-height)",
      textDecoration: "var(--text-link-sm-default-text-decoration)",
      textTransform: "var(--text-link-sm-default-text-transform)"
    },
    hover: {
      fontFamily: "var(--text-link-sm-hover-font-family)",
      fontSize: "var(--text-link-sm-hover-font-size)",
      fontWeight: "var(--text-link-sm-hover-font-weight)",
      letterSpacing: "var(--text-link-sm-hover-letter-spacing)",
      lineHeight: "var(--text-link-sm-hover-line-height)",
      textDecoration: "var(--text-link-sm-hover-text-decoration)",
      textTransform: "var(--text-link-sm-hover-text-transform)"
    }
  }
},
list: {
  base: {
    normal: {
      fontFamily: "var(--text-list-base-normal-font-family)",
      fontSize: "var(--text-list-base-normal-font-size)",
      fontWeight: "var(--text-list-base-normal-font-weight)",
      letterSpacing: "var(--text-list-base-normal-letter-spacing)",
      lineHeight: "var(--text-list-base-normal-line-height)",
      textDecoration: "var(--text-list-base-normal-text-decoration)",
      textTransform: "var(--text-list-base-normal-text-transform)"
    }
  },
  sm: {
    normal: {
      fontFamily: "var(--text-list-sm-normal-font-family)",
      fontSize: "var(--text-list-sm-normal-font-size)",
      fontWeight: "var(--text-list-sm-normal-font-weight)",
      letterSpacing: "var(--text-list-sm-normal-letter-spacing)",
      lineHeight: "var(--text-list-sm-normal-line-height)",
      textDecoration: "var(--text-list-sm-normal-text-decoration)",
      textTransform: "var(--text-list-sm-normal-text-transform)"
    }
  }
}
};

export default {
  borderRadius,
  borderWidth,
  breakpoints,
  color,
  componentColor,
  components,
  font,
  spacing,
  text
};
