// @angi/one-design-system/dist/components/es/Link/index.js
import { jsxs as d, jsx as r } from "react/jsx-runtime";
import { forwardRef as c } from "react";
import { c as f } from "../_chunks/clsx-OuTLNxxd.js";
import { _ as u } from "../_chunks/_BaseLink-COWMopm2.js";
import { _ as e } from "../_chunks/_BaseIcon-BAJuKzUY.js";
import { T as p } from "../_chunks/Text-B3tFn66X.js";
import { c as L } from "../_chunks/createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import '../assets/Link.css';var _ = L({ defaultClassName: "csd5h20", variantClassNames: { color: { neutral: "csd5h21", inverted: "csd5h22" } }, defaultVariants: { color: "neutral" }, compoundVariants: [] });
const E = ["sm", "md"], h = "md", F = ["neutral", "inverted"], k = "neutral", N = {
  sm: "link.sm.default",
  md: "link.base.default"
}, T = c(
  ({
    size: n = h,
    color: i = k,
    iconStart: a,
    iconEnd: s,
    className: t,
    children: o,
    ...m
  }, l) => /* @__PURE__ */ d(
    u,
    {
      ref: l,
      className: f(_({ color: i }), t),
      ...m,
      children: [
        a && /* @__PURE__ */ r(e, { size: 16, "aria-hidden": "true", children: a }),
        /* @__PURE__ */ r(p, { variant: N[n], children: /* @__PURE__ */ r("span", { children: o }) }),
        s && /* @__PURE__ */ r(e, { size: 16, "aria-hidden": "true", children: s })
      ]
    }
  )
);
T.displayName = "Link";
export {
  k as DEFAULT_LINK_COLOR,
  h as DEFAULT_LINK_SIZE,
  T as Link,
  F as LinkColors,
  E as LinkSizes
};
