// @angi/one-design-system/dist/components/es/FormField/index.js
import { jsxs as R, jsx as l } from "react/jsx-runtime";
import g, { useState as T, createContext as q, useRef as O, useCallback as V, forwardRef as _ } from "react";
import { $ as S, b as F, c as L } from "../_chunks/Label-DLRS1VtT.js";
import { a as H, $ as k, c as N, d as j, e as z, f as I } from "../_chunks/Form-sh8c7lqF.js";
import { b as A, $ as E, h as K, f as U, j as w, a as G, i as J } from "../_chunks/utils-CjtkL41C.js";
import { $ as Q } from "../_chunks/RSPContexts-Dm3BezC-.js";
import { $ as W } from "../_chunks/Input-C70jFPqg.js";
import { a as X } from "../_chunks/TextArea-DWRv-YP4.js";
import { $ as D } from "../_chunks/filterDOMProps-Cxz9cAAv.js";
import { $ as Y, a as Z } from "../_chunks/useFormValidation-CgQ_8wCr.js";
import { $ as ee } from "../_chunks/useControlledState-3TGZUXux.js";
import { a as ae } from "../_chunks/useFocusable-CrxUcjB4.js";
import { L as ie } from "../_chunks/Label-BqoOfVfQ.js";
import { H as te } from "../_chunks/Hint-CIDmUonJ.js";
import { V as re } from "../_chunks/ValidationMessage-Dc42Hn25.js";
import { c as de } from "../_chunks/clsx-OuTLNxxd.js";
import '../assets/FormField.css';function ne(e, a) {
  let { inputElementType: i = "input", isDisabled: n = !1, isRequired: r = !1, isReadOnly: d = !1, type: s = "text", validationBehavior: t = "aria" } = e, [f, b] = ee(e.value, e.defaultValue || "", e.onChange), { focusableProps: h } = ae(e, a), m = H({
    ...e,
    value: f
  }), { isInvalid: c, validationErrors: $, validationDetails: P } = m.displayValidation, { labelProps: y, fieldProps: o, descriptionProps: v, errorMessageProps: x } = k({
    ...e,
    isInvalid: c,
    errorMessage: e.errorMessage || $
  }), p = D(e, {
    labelable: !0
  });
  const u = {
    type: s,
    pattern: e.pattern
  };
  let [M] = T(f);
  var C;
  return Y(a, (C = e.defaultValue) !== null && C !== void 0 ? C : M, b), Z(e, m, a), {
    labelProps: y,
    inputProps: A(p, i === "input" ? u : void 0, {
      disabled: n,
      readOnly: d,
      required: r && t === "native",
      "aria-required": r && t === "aria" || void 0,
      "aria-invalid": c || void 0,
      "aria-errormessage": e["aria-errormessage"],
      "aria-activedescendant": e["aria-activedescendant"],
      "aria-autocomplete": e["aria-autocomplete"],
      "aria-haspopup": e["aria-haspopup"],
      "aria-controls": e["aria-controls"],
      value: f,
      onChange: (B) => b(B.target.value),
      autoComplete: e.autoComplete,
      autoCapitalize: e.autoCapitalize,
      maxLength: e.maxLength,
      minLength: e.minLength,
      name: e.name,
      form: e.form,
      placeholder: e.placeholder,
      inputMode: e.inputMode,
      autoCorrect: e.autoCorrect,
      spellCheck: e.spellCheck,
      [parseInt(g.version, 10) >= 17 ? "enterKeyHint" : "enterkeyhint"]: e.enterKeyHint,
      // Clipboard events
      onCopy: e.onCopy,
      onCut: e.onCut,
      onPaste: e.onPaste,
      // Composition events
      onCompositionEnd: e.onCompositionEnd,
      onCompositionStart: e.onCompositionStart,
      onCompositionUpdate: e.onCompositionUpdate,
      // Selection events
      onSelect: e.onSelect,
      // Input events
      onBeforeInput: e.onBeforeInput,
      onInput: e.onInput,
      ...h,
      ...o
    }),
    descriptionProps: v,
    errorMessageProps: x,
    isInvalid: c,
    validationErrors: $,
    validationDetails: P
  };
}
const oe = /* @__PURE__ */ q({}), le = /* @__PURE__ */ q(null), se = /* @__PURE__ */ S(function(a, i) {
  [a, i] = E(a, i, le);
  let { validationBehavior: n } = K(N) || {};
  var r, d;
  let s = (d = (r = a.validationBehavior) !== null && r !== void 0 ? r : n) !== null && d !== void 0 ? d : "native", t = O(null);
  [a, t] = E(a, t, Q);
  let [f, b] = U(!a["aria-label"] && !a["aria-labelledby"]), [h, m] = T("input"), { labelProps: c, inputProps: $, descriptionProps: P, errorMessageProps: y, ...o } = ne({
    ...w(a),
    inputElementType: h,
    label: b,
    validationBehavior: s
  }, t), v = V((u) => {
    t.current = u, u && m(u instanceof HTMLTextAreaElement ? "textarea" : "input");
  }, [
    t
  ]), x = G({
    ...a,
    values: {
      isDisabled: a.isDisabled || !1,
      isInvalid: o.isInvalid,
      isReadOnly: a.isReadOnly || !1,
      isRequired: a.isRequired || !1
    },
    defaultClassName: "react-aria-TextField"
  }), p = D(a, {
    global: !0
  });
  return delete p.id, /* @__PURE__ */ g.createElement("div", {
    ...p,
    ...x,
    ref: i,
    slot: a.slot || void 0,
    "data-disabled": a.isDisabled || void 0,
    "data-invalid": o.isInvalid || void 0,
    "data-readonly": a.isReadOnly || void 0,
    "data-required": a.isRequired || void 0
  }, /* @__PURE__ */ g.createElement(J, {
    values: [
      [
        F,
        {
          ...c,
          ref: f
        }
      ],
      [
        W,
        {
          ...$,
          ref: v
        }
      ],
      [
        X,
        {
          ...$,
          ref: v
        }
      ],
      [
        oe,
        {
          role: "presentation",
          isInvalid: o.isInvalid,
          isDisabled: a.isDisabled || !1
        }
      ],
      [
        j,
        {
          slots: {
            description: P,
            errorMessage: y
          }
        }
      ],
      [
        z,
        o
      ]
    ]
  }, x.children));
});
var fe = "_1sr0d00", ce = "_1sr0d01";
const $e = _(
  ({
    label: e,
    hint: a,
    validationMessage: i,
    inputId: n,
    children: r,
    className: d,
    ...s
  }, t) => /* @__PURE__ */ R(
    se,
    {
      ref: t,
      className: de(fe, d),
      isInvalid: i?.variant === "invalid",
      ...n !== void 0 && { id: n },
      ...s,
      children: [
        (e || a) && /* @__PURE__ */ R("div", { className: ce, children: [
          e && /* @__PURE__ */ l(
            ie,
            {
              asChild: !0,
              ...e.required !== void 0 && {
                required: e.required
              },
              ...e.variant !== void 0 && { variant: e.variant },
              children: /* @__PURE__ */ l(L, { children: e.children })
            }
          ),
          a && /* @__PURE__ */ l(I, { elementType: "div", slot: "description", children: /* @__PURE__ */ l(te, { children: a.children }) })
        ] }),
        r,
        i && /* @__PURE__ */ l(I, { slot: "errorMessage", elementType: "div", children: /* @__PURE__ */ l(re, { ...i, children: i.children }) })
      ]
    }
  )
);
$e.displayName = "FormField";
export {
  $e as FormField
};
