// @angi/one-design-system/dist/components/es/OTPFieldset/index.js
import { jsxs as F, jsx as D } from "react/jsx-runtime";
import { useCallback as C, forwardRef as q, useRef as Y, useId as J, useMemo as A } from "react";
import { c as K } from "../_chunks/clsx-OuTLNxxd.js";
import { c as M } from "../_chunks/createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import { u as Q, i as W } from "../_chunks/useTranslations-Da7GR3Ot.js";
import { L as X } from "../_chunks/Legend-Cgz7gN8f.js";
import { H as x } from "../_chunks/Hint-CIDmUonJ.js";
import { V as tt } from "../_chunks/ValidationMessage-Dc42Hn25.js";
import { I as et } from "../_chunks/Input-Dyh2Ln22.js";
import '../assets/OTPFieldset.css';var rt = "_1b6nl550", st = "_1b6nl551", nt = "_1b6nl552", at = M({ defaultClassName: "_1b6nl553", variantClassNames: { size: { sm: "_1b6nl554", md: "_1b6nl555", lg: "_1b6nl556" } }, defaultVariants: { size: "md" }, compoundVariants: [] }), ot = M({ defaultClassName: "_1b6nl557", variantClassNames: { size: { sm: "_1b6nl558", md: "_1b6nl559", lg: "_1b6nl55a" } }, defaultVariants: { size: "md" }, compoundVariants: [] });
const Ot = ["sm", "md", "lg"], vt = [
  "none",
  "uppercase",
  "lowercase"
], ct = "md", lt = 6, j = {
  whitelistedChars: "[0-9]",
  transform: "none"
}, ut = {
  sm: "sm",
  md: "md",
  lg: "md"
}, k = "​";
function it(e, r) {
  const o = (e ?? "").split(""), s = [];
  for (let u = 0; u < r; u++) {
    const c = o[u] ?? "";
    s.push(c === k ? "" : c);
  }
  return s;
}
function H(e) {
  return e.map((r) => r === "" ? k : r).join("");
}
function O(e) {
  return e.replace(new RegExp(k, "g"), "");
}
function ft(e, r) {
  return O(e).length === r;
}
function dt(e) {
  return e === void 0 ? j : {
    ...j,
    ...e
  };
}
function pt(e, r) {
  if (r === "*")
    return !0;
  try {
    return new RegExp(r).test(e);
  } catch {
    return !0;
  }
}
function mt(e, r) {
  switch (r) {
    case "uppercase":
      return e.toUpperCase();
    case "lowercase":
      return e.toLowerCase();
    default:
      return e;
  }
}
function v(e, r) {
  return e === k ? e : pt(e, r.whitelistedChars) ? mt(e, r.transform ?? "none") : null;
}
function R(e, r) {
  return e.split("").map((o) => v(o, r)).filter((o) => o !== null).join("");
}
function bt({
  inputRefs: e,
  values: r,
  onChange: o,
  length: s,
  disabled: u,
  characterConfig: c
}) {
  const i = C(
    (t) => {
      const n = Math.max(0, Math.min(t, s - 1));
      e.current?.[n]?.focus();
    },
    [e, s]
  ), d = C(
    (t, n) => {
      const a = [...r], l = n.slice(0, 1), f = l ? v(l, c) : "";
      return f !== null ? (a[t] = f, o(a), !0) : !1;
    },
    [r, o, c]
  ), I = C(
    (t, n) => {
      if (u) return;
      const { key: a } = n;
      switch (a) {
        case "Backspace": {
          n.preventDefault(), r[t] ?? "" ? d(t, "") : t > 0 && (d(t - 1, ""), i(t - 1));
          break;
        }
        case "Delete": {
          n.preventDefault(), d(t, "");
          break;
        }
        case "ArrowLeft": {
          n.preventDefault(), t > 0 && i(t - 1);
          break;
        }
        case "ArrowRight": {
          n.preventDefault(), t < s - 1 && i(t + 1);
          break;
        }
        // Let Tab/Shift+Tab work naturally for accessibility
        case "Tab":
          break;
        default:
          a.length === 1 && !n.ctrlKey && !n.metaKey && !n.altKey && (r[t] ?? "" ? (n.preventDefault(), d(t, a) && t < s - 1 && i(t + 1)) : v(a, c) === null && n.preventDefault());
          break;
      }
    },
    [u, r, d, i, s, c]
  ), _ = C(
    (t, n) => {
      if (u) return;
      const a = n.target.value;
      if (a.length > 1) {
        const l = R(a, c);
        if (l.length === 0) return;
        const f = l.split(""), m = [...r];
        f.forEach((T, h) => {
          const b = t + h;
          b < s && (m[b] = T);
        }), o(m);
        const L = m.findIndex((T, h) => h >= t && !T), E = L !== -1 ? L : Math.min(t + f.length, s - 1);
        i(E);
        return;
      }
      a.length === 1 ? d(t, a) && t < s - 1 && i(t + 1) : a.length === 0 && d(t, "");
    },
    [
      u,
      r,
      o,
      s,
      d,
      i,
      c
    ]
  ), V = C(
    (t, n) => {
      if (u) return;
      n.preventDefault();
      const a = n.clipboardData.getData("text");
      if (!a) return;
      const l = R(a, c);
      if (l.length === 0) return;
      const f = l.split(""), m = [...r];
      f.forEach((h, b) => {
        const y = t + b;
        y < s && (m[y] = h);
      }), o(m);
      const L = Math.min(f.length, s - t), E = t + L, T = E < s ? E : s - 1;
      i(T);
    },
    [u, r, o, s, i, c]
  );
  return {
    handleKeyDown: I,
    handleInput: _,
    handlePaste: V
  };
}
const Tt = { digitLabel: "Zeichen {current} von {total}" }, ht = { digitLabel: "Character {current} of {total}" }, It = { digitLabel: "Caractère {current} sur {total}" }, _t = { digitLabel: "Carattere {current} di {total}" }, Lt = { digitLabel: "Teken {current} van {total}" }, Et = {
  de: Tt,
  en: ht,
  fr: It,
  it: _t,
  nl: Lt
}, wt = q(
  ({
    value: e,
    onChange: r,
    onComplete: o,
    length: s = lt,
    size: u = ct,
    validationMessage: c,
    disabled: i,
    name: d,
    legend: I,
    hint: _,
    characterConfig: V,
    className: t,
    ...n
  }, a) => {
    const l = Y([]), f = Q(Et), m = J(), L = _ ? `${m}-hint` : void 0, E = c ? `${m}-validation` : void 0, T = [L, E].filter(
      Boolean
    ), h = T.length > 0 ? T.join(" ") : void 0, b = A(
      () => dt(V),
      [V]
    ), y = A(
      () => R(e, b),
      [e, b]
    ), N = it(y, s), U = C(
      (P) => {
        const p = H(P);
        r(p), o && ft(p, s) && o(O(p));
      },
      [r, o, s]
    ), g = A(() => ["[0-9]", "[0-9]+", "\\d", "\\d+"].includes(b.whitelistedChars) ? "numeric" : "text", [b.whitelistedChars]), { handleKeyDown: z, handleInput: B, handlePaste: Z } = bt({
      inputRefs: l,
      values: N,
      onChange: U,
      length: s,
      disabled: i ?? !1,
      characterConfig: b
    }), $ = C(() => {
      l.current[0]?.focus();
    }, []);
    return /* @__PURE__ */ F(
      "fieldset",
      {
        ref: a,
        className: K(rt, t),
        ...h !== void 0 && {
          "aria-describedby": h
        },
        ...n,
        children: [
          (I || _) && /* @__PURE__ */ F("div", { className: nt, children: [
            I && /* @__PURE__ */ D(
              X,
              {
                ...I,
                onClick: $,
                className: K(st, I.className),
                children: I.children
              }
            ),
            _ && /* @__PURE__ */ D(x, { id: L, ..._, children: _.children })
          ] }),
          /* @__PURE__ */ F("div", { className: at({ size: u }), children: [
            d && /* @__PURE__ */ D(
              "input",
              {
                type: "hidden",
                name: d,
                value: O(H(N)),
                "aria-hidden": "true"
              }
            ),
            N.map((P, p) => {
              const S = `${m}-otp-${p}`, G = W(f.digitLabel, {
                current: p + 1,
                total: s
              });
              return /* @__PURE__ */ D(
                et,
                {
                  ref: (w) => {
                    l.current[p] = w;
                  },
                  id: S,
                  type: "text",
                  inputMode: g,
                  autoComplete: p === 0 ? "one-time-code" : "off",
                  maxLength: 1,
                  value: P,
                  disabled: i,
                  "aria-label": G,
                  size: ut[u],
                  validation: c?.variant,
                  className: ot({ size: u }),
                  onKeyDown: (w) => z(p, w),
                  onChange: (w) => B(p, w),
                  onPaste: (w) => Z(p, w)
                },
                S
              );
            })
          ] }),
          c && /* @__PURE__ */ D(tt, { id: E, ...c })
        ]
      }
    );
  }
);
wt.displayName = "OTPFieldset";
export {
  j as DEFAULT_OTP_FIELDSET_CHARACTER_CONFIG,
  lt as DEFAULT_OTP_FIELDSET_LENGTH,
  ct as DEFAULT_OTP_FIELDSET_SIZE,
  wt as OTPFieldset,
  vt as OTPFieldsetCharacterTransforms,
  Ot as OTPFieldsetSizes
};
