// @angi/one-design-system/dist/components/es/Textarea/index.js
import { jsxs as R, jsx as h } from "react/jsx-runtime";
import { forwardRef as w, useRef as E, useImperativeHandle as T, useLayoutEffect as B, useId as M, useState as S, useCallback as A } from "react";
import { $ as D } from "../_chunks/TextArea-DWRv-YP4.js";
import { c as F } from "../_chunks/createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import { c as W } from "../_chunks/clsx-OuTLNxxd.js";
import { H as q } from "../_chunks/Hint-CIDmUonJ.js";
import '../assets/Textarea.css';var z = "hak5910", G = F({ defaultClassName: "hak5912 hak5911", variantClassNames: { autoResize: { true: "hak5913", false: "hak5914" } }, defaultVariants: { autoResize: !1 }, compoundVariants: [] });
const J = w(
  ({
    validation: l,
    autoResize: r = !1,
    maxLength: t,
    characterCountLabel: o,
    className: m,
    id: g,
    onChange: d,
    "aria-describedby": C,
    ...a
  }, I) => {
    const i = E(null);
    T(
      I,
      () => i.current,
      []
    ), B(() => {
      const e = i.current;
      if (!e || !r) return;
      const s = () => {
        e.style.height = "auto", e.style.height = `${e.scrollHeight}px`;
      };
      return s(), e.addEventListener("input", s), () => e.removeEventListener("input", s);
    }, [r, a.value]);
    const y = M(), c = g ?? y, u = t !== void 0 ? `${c}-character-count` : void 0, b = [C, u].filter(Boolean).join(" ") || void 0, n = (e) => e == null ? 0 : String(e).length, v = a.value != null ? n(a.value) : n(a.defaultValue), k = t !== void 0 ? Math.min(v, t) : v, [$, p] = S(k), f = a.value != null ? n(a.value) : $, x = t !== void 0 ? Math.min(f, t) : f, H = t !== void 0 ? t - x : 0, N = A(
      (e) => {
        p(e.target.value.length), d?.(e);
      },
      [d]
    ), V = l === "invalid", j = W(
      G({
        autoResize: r
      }),
      m
    );
    return /* @__PURE__ */ R("div", { className: z, children: [
      /* @__PURE__ */ h(
        D,
        {
          ref: i,
          id: c,
          className: j,
          "data-validation": l ?? void 0,
          "aria-invalid": V || void 0,
          "aria-describedby": b,
          onChange: N,
          ...t !== void 0 && { maxLength: t },
          ...a
        }
      ),
      t !== void 0 && o !== void 0 && /* @__PURE__ */ h(q, { id: u, "aria-live": "polite", children: o(H) })
    ] });
  }
);
J.displayName = "Textarea";
const Y = ["invalid", "valid"];
export {
  J as Textarea,
  Y as TextareaValidations
};
