// @angi/one-design-system/dist/components/es/CTAGroup/index.js
import { jsx as f } from "react/jsx-runtime";
import { forwardRef as h, useRef as z, useState as G, useEffect as N } from "react";
import { c as v } from "../_chunks/clsx-OuTLNxxd.js";
import { c as C } from "../_chunks/createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import '../assets/CTAGroup.css';var R = C({ defaultClassName: "c1rmiu0", variantClassNames: { layout: { vertical: "c1rmiu1", horizontal: "c1rmiu2", auto: "c1rmiu3" } }, defaultVariants: { layout: "auto" }, compoundVariants: [] });
const E = ["auto", "horizontal", "vertical"], n = {
  horizontal: "orientation-horizontal",
  vertical: "orientation-vertical"
}, b = "auto", x = (t) => {
  if (t.children.length === 0) return 0;
  const i = n.horizontal, a = n.vertical, r = t.classList.contains(i), s = t.classList.contains(a), c = t.style.width;
  t.classList.remove(a), t.classList.add(i), t.style.width = "max-content";
  const e = t.offsetWidth;
  return r || t.classList.remove(i), s && t.classList.add(a), t.style.width = c, e;
}, V = ["default", "compact"], S = "default";
var g = C({ defaultClassName: "_1gax4u22 _1gax4u20", variantClassNames: { layout: { default: "_1gax4u23", compact: "_1gax4u24 _1gax4u21" } }, defaultVariants: { layout: "default" }, compoundVariants: [] });
const m = h(
  ({
    children: t,
    className: i,
    layout: a = S,
    ...r
  }, s) => /* @__PURE__ */ f(
    "div",
    {
      ref: s,
      className: v(g({ layout: a }), i),
      ...r,
      children: t
    }
  )
);
m.displayName = "CTAGroup.Section";
const p = h(
  ({ children: t, className: i, layout: a = b, ...r }, s) => {
    const c = z(null), [e, A] = G(
      n.vertical
    ), T = a === "auto" ? e : n[a];
    return N(() => {
      if (a !== "auto") return;
      const o = c.current;
      if (!o) return;
      const u = () => {
        const O = o.offsetWidth, _ = x(o), L = O >= _ ? "horizontal" : "vertical";
        A(n[L]);
      };
      u();
      const l = new ResizeObserver(u);
      l.observe(o);
      const d = new MutationObserver(u);
      return d.observe(o, {
        childList: !0,
        subtree: !0,
        characterData: !0
      }), () => {
        l.disconnect(), d.disconnect();
      };
    }, [a]), /* @__PURE__ */ f(
      "div",
      {
        ref: (o) => {
          c.current = o, typeof s == "function" ? s(o) : s && (s.current = o);
        },
        className: v(
          R({ layout: a }),
          i,
          T
        ),
        ...r,
        children: t
      }
    );
  }
);
p.displayName = "CTAGroup";
const D = Object.assign(p, {
  Section: m
});
export {
  D as CTAGroup,
  E as CTAGroupLayouts,
  V as CTAGroupSectionLayouts,
  b as DEFAULT_CTA_GROUP_LAYOUT,
  S as DEFAULT_CTA_GROUP_SECTION_LAYOUT
};
