// @angi/one-design-system/dist/components/es/RadioButtonFieldset/index.js
import { jsxs as p, jsx as e } from "react/jsx-runtime";
import { forwardRef as R, useId as I } from "react";
import { c as x } from "../_chunks/Label-DLRS1VtT.js";
import { f as C } from "../_chunks/Form-sh8c7lqF.js";
import { u as $, R as w } from "../_chunks/useRadioGroupContext-CuaGc53C.js";
import { L as y } from "../_chunks/Legend-Cgz7gN8f.js";
import { H as F } from "../_chunks/Hint-CIDmUonJ.js";
import { V as T } from "../_chunks/ValidationMessage-Dc42Hn25.js";
import { R as V } from "../_chunks/RadioButtonTile-DjBZCH0p.js";
import { R as D } from "../_chunks/RadioButton-Cl7_yIuj.js";
import { c as g } from "../_chunks/clsx-OuTLNxxd.js";
import { L } from "../_chunks/Label-BqoOfVfQ.js";
import { R as j } from "../_chunks/RadioItem-CkMea6Af.js";
import { c as q } from "../_chunks/createRuntimeFn-62c9670f.esm-5AHdoIVZ.js";
import '../assets/RadioButtonFieldset.css';var A = "jpvatp0";
const _ = R(
  ({ value: i, label: a, disabled: l = !1, hideFocusRing: t = !1, className: o }, d) => {
    const c = I(), n = `${c}-radio`, s = `${c}-label`, r = $(), u = r.selectedValue === i, m = r.isInvalid, v = l || r.isDisabled;
    return /* @__PURE__ */ p(
      "label",
      {
        className: g(A, o),
        "data-checked": u || void 0,
        "data-disabled": v || void 0,
        "data-invalid": m || void 0,
        children: [
          /* @__PURE__ */ e(
            j,
            {
              ref: d,
              value: i,
              id: n,
              "aria-labelledby": s,
              disabled: l,
              hideFocusRing: t
            }
          ),
          /* @__PURE__ */ e(L, { variant: "normal", ...a, id: s, asChild: !0, children: /* @__PURE__ */ e("span", { children: a.children }) })
        ]
      }
    );
  }
);
_.displayName = "_RadioButtonCompact";
var G = "_118w00b0", M = "_118w00b1", S = "_118w00b2", E = q({ defaultClassName: "_118w00b3", variantClassNames: { variant: { tile: "_118w00b4", text: "_118w00b5", compact: "_118w00b6" } }, defaultVariants: { variant: "tile" }, compoundVariants: [] });
const H = (i) => {
  switch (i.variant) {
    case "tile":
      return i.items.map((a) => /* @__PURE__ */ e(V, { ...a }, a.value));
    case "text":
      return i.items.map((a) => /* @__PURE__ */ e(
        D,
        {
          ...a,
          label: { variant: "normal", ...a.label }
        },
        a.value
      ));
    case "compact":
      return i.items.map((a) => /* @__PURE__ */ e(_, { ...a, hideFocusRing: !0 }, a.value));
  }
}, O = R((i, a) => {
  const {
    variant: l,
    legend: t,
    hint: o,
    validationMessage: d,
    className: c,
    required: n,
    disabled: s,
    readOnly: r,
    ...u
  } = i, m = I(), v = o ? `${m}-hint` : void 0, f = d ? `${m}-validation` : void 0, b = [v, f].filter(
    Boolean
  ), h = b.length > 0 ? b.join(" ") : void 0, B = d?.variant === "invalid", N = t?.required ?? n ?? !1;
  return /* @__PURE__ */ p(
    w,
    {
      ref: a,
      className: g(G, c),
      invalid: B,
      ...n !== void 0 && { required: n },
      ...s !== void 0 && { disabled: s },
      ...r !== void 0 && { readOnly: r },
      ...h !== void 0 && {
        "aria-describedby": h
      },
      ...u,
      children: [
        (t || o) && /* @__PURE__ */ p("div", { className: M, children: [
          t && /* @__PURE__ */ e(y, { ...t, asChild: !0, required: N, children: /* @__PURE__ */ e(x, { children: t.children }) }),
          o && /* @__PURE__ */ e(C, { elementType: "div", slot: "description", id: v, children: /* @__PURE__ */ e(F, { ...o }) })
        ] }),
        d && /* @__PURE__ */ e("div", { className: S, children: /* @__PURE__ */ e(T, { id: f, ...d, children: d.children }) }),
        /* @__PURE__ */ e("div", { className: E({ variant: l }), children: H(i) })
      ]
    }
  );
});
O.displayName = "RadioButtonFieldset";
const ta = ["tile", "text", "compact"], da = "tile";
export {
  da as DEFAULT_RADIO_BUTTON_FIELDSET_VARIANT,
  O as RadioButtonFieldset,
  ta as RadioButtonFieldsetVariants
};
